diff --git a/common/autoconf/generated-configure.sh b/common/autoconf/generated-configure.sh
--- openjdk///common/autoconf/generated-configure.sh
+++ openjdk///common/autoconf/generated-configure.sh
@@ -3999,7 +3999,7 @@
 #CUSTOM_AUTOCONF_INCLUDE
 
 # Do not change or remove the following line, it is needed for consistency checks:
-DATE_WHEN_GENERATED=1485433878
+DATE_WHEN_GENERATED=1494557051
 
 ###############################################################################
 #
@@ -6943,9 +6943,9 @@
       VAR_CPU_BITS=32
       VAR_CPU_ENDIAN=little
       ;;
-    arm*)
-      VAR_CPU=arm
-      VAR_CPU_ARCH=arm
+    arm*|aarch32)
+      VAR_CPU=aarch32
+      VAR_CPU_ARCH=aarch32
       VAR_CPU_BITS=32
       VAR_CPU_ENDIAN=little
       ;;
@@ -7088,9 +7088,9 @@
       VAR_CPU_BITS=32
       VAR_CPU_ENDIAN=little
       ;;
-    arm*)
-      VAR_CPU=arm
-      VAR_CPU_ARCH=arm
+    arm*|aarch32)
+      VAR_CPU=aarch32
+      VAR_CPU_ARCH=aarch32
       VAR_CPU_BITS=32
       VAR_CPU_ENDIAN=little
       ;;
@@ -7304,6 +7304,8 @@
   elif test "x$OPENJDK_TARGET_OS" != xmacosx && test "x$OPENJDK_TARGET_CPU" = xx86_64; then
     # On all platforms except macosx, we replace x86_64 with amd64.
     OPENJDK_TARGET_CPU_OSARCH="amd64"
+  elif test "x$OPENJDK_TARGET_CPU" = xaarch32; then
+    OPENJDK_TARGET_CPU_OSARCH="arm"
   fi
 
 
@@ -8058,7 +8060,11 @@
 
 
   if test "x$with_jvm_variants" = x; then
-    with_jvm_variants="server"
+    if test "x$OPENJDK_TARGET_CPU" = xaarch32; then
+      with_jvm_variants="client";
+    else
+      with_jvm_variants="server";
+    fi
   fi
 
   JVM_VARIANTS=",$with_jvm_variants,"
@@ -8122,6 +8128,9 @@
   if test "x$VAR_CPU" = xppc64 -o "x$VAR_CPU" = xppc64le ; then
     INCLUDE_SA=false
   fi
+  if test "x$OPENJDK_TARGET_CPU" = xaarch32; then
+    INCLUDE_SA=false
+  fi
   if test "x$OPENJDK_TARGET_CPU" = xaarch64; then
     INCLUDE_SA=false
   fi
diff --git a/common/autoconf/jdk-options.m4 b/common/autoconf/jdk-options.m4
--- openjdk///common/autoconf/jdk-options.m4
+++ openjdk///common/autoconf/jdk-options.m4
@@ -98,7 +98,11 @@
 	[JVM variants (separated by commas) to build (server, client, minimal1, kernel, zero, zeroshark, core) @<:@server@:>@])])
 
   if test "x$with_jvm_variants" = x; then
-    with_jvm_variants="server"
+    if test "x$OPENJDK_TARGET_CPU" = xaarch32; then
+      with_jvm_variants="client";
+    else
+      with_jvm_variants="server";
+    fi
   fi
 
   JVM_VARIANTS=",$with_jvm_variants,"
@@ -161,6 +165,9 @@
   if test "x$VAR_CPU" = xppc64 -o "x$VAR_CPU" = xppc64le ; then
     INCLUDE_SA=false
   fi
+  if test "x$OPENJDK_TARGET_CPU" = xaarch32; then
+    INCLUDE_SA=false
+  fi
   if test "x$OPENJDK_TARGET_CPU" = xaarch64; then
     INCLUDE_SA=false
   fi
diff --git a/common/autoconf/platform.m4 b/common/autoconf/platform.m4
--- openjdk///common/autoconf/platform.m4
+++ openjdk///common/autoconf/platform.m4
@@ -42,9 +42,9 @@
       VAR_CPU_BITS=32
       VAR_CPU_ENDIAN=little
       ;;
-    arm*)
-      VAR_CPU=arm
-      VAR_CPU_ARCH=arm
+    arm*|aarch32)
+      VAR_CPU=aarch32
+      VAR_CPU_ARCH=aarch32
       VAR_CPU_BITS=32
       VAR_CPU_ENDIAN=little
       ;;
@@ -322,6 +322,8 @@
   elif test "x$OPENJDK_TARGET_OS" != xmacosx && test "x$OPENJDK_TARGET_CPU" = xx86_64; then
     # On all platforms except macosx, we replace x86_64 with amd64.
     OPENJDK_TARGET_CPU_OSARCH="amd64"
+  elif test "x$OPENJDK_TARGET_CPU" = xaarch32; then
+    OPENJDK_TARGET_CPU_OSARCH="arm"
   fi
   AC_SUBST(OPENJDK_TARGET_CPU_OSARCH)
 
