diff -Nru openjdk.orig/hotspot/agent/src/os/linux/Makefile openjdk/hotspot/agent/src/os/linux/Makefile
--- openjdk.orig/hotspot/agent/src/os/linux/Makefile	2010-09-01 09:57:06.000000000 +0100
+++ openjdk/hotspot/agent/src/os/linux/Makefile	2010-09-01 10:27:09.592191147 +0100
@@ -40,7 +40,7 @@
 
 LIBS     = -lthread_db
 
-CFLAGS   = -c -fPIC -g -D_GNU_SOURCE -D$(ARCH) $(INCLUDES)
+CFLAGS   = -c -fPIC -g -D_GNU_SOURCE -D_$(ARCH)_ $(if $(filter $(ARCH),alpha),,-D$(ARCH)) $(INCLUDES)
 
 LIBSA = $(ARCH)/libsaproc.so
 
@@ -73,7 +73,7 @@
 	$(GCC) -shared $(LFLAGS_LIBSA) -o $(LIBSA) $(OBJS) $(LIBS)
 
 test.o: test.c
-	$(GCC) -c -o test.o -g -D_GNU_SOURCE -D$(ARCH) $(INCLUDES) test.c
+	$(GCC) -c -o test.o -g -D_GNU_SOURCE -D_$(ARCH)_ $(if $(filter $(ARCH),alpha),,-D$(ARCH)) $(INCLUDES) test.c
 
 test: test.o
 	$(GCC) -o test test.o -L$(ARCH) -lsaproc $(LIBS)
diff -Nru openjdk.orig/hotspot/src/share/vm/oops/generateOopMap.cpp openjdk/hotspot/src/share/vm/oops/generateOopMap.cpp
--- openjdk.orig/hotspot/src/share/vm/oops/generateOopMap.cpp	2010-08-27 19:15:27.000000000 +0100
+++ openjdk/hotspot/src/share/vm/oops/generateOopMap.cpp	2010-09-01 10:28:08.943025196 +0100
@@ -2104,7 +2104,7 @@
 void GenerateOopMap::report_error(const char *format, ...) {
   va_list ap;
   va_start(ap, format);
-  error_work(format, ap);
+  report_error(format, ap);
 }
 
 void GenerateOopMap::verify_error(const char *format, ...) {
