diff -Nru openjdk.orig/jdk/make/sun/awt/Makefile openjdk/jdk/make/sun/awt/Makefile
--- openjdk.orig/jdk/make/sun/awt/Makefile	2015-10-22 03:21:58.927759489 +0100
+++ openjdk/jdk/make/sun/awt/Makefile	2015-10-22 03:31:55.553678663 +0100
@@ -486,7 +486,8 @@
 	fontconfig.Fedora.9.properties			\
 	fontconfig.Fedora.10.properties			\
 	fontconfig.Fedora.11.properties			\
-	fontconfig.Fedora.12.properties
+	fontconfig.Fedora.12.properties			\
+	fontconfig.Gentoo.properties
 else
 
 FONTCONFIGS_SRC	= $(CLOSED_SRC)/solaris/classes/sun/awt/fontconfigs
diff -Nru openjdk.orig/jdk/src/solaris/classes/sun/awt/motif/MFontConfiguration.java openjdk/jdk/src/solaris/classes/sun/awt/motif/MFontConfiguration.java
--- openjdk.orig/jdk/src/solaris/classes/sun/awt/motif/MFontConfiguration.java	2015-07-23 09:51:31.000000000 +0100
+++ openjdk/jdk/src/solaris/classes/sun/awt/motif/MFontConfiguration.java	2015-10-22 03:31:36.429997969 +0100
@@ -178,6 +178,9 @@
                 } else if ((f = new File("/etc/SuSE-release")).canRead()) {
                     osName = "SuSE";
                     osVersion = getVersionString(f);
+                } else if ((f = new File("/etc/gentoo-release")).canRead()) {
+                    osName = "Gentoo";
+                    osVersion = getVersionString(f);
                 } else if ((f = new File("/etc/lsb-release")).canRead()) {
                     /* Ubuntu and (perhaps others) use only lsb-release.
                      * Syntax and encoding is compatible with java properties.
