diff -Nru openjdk.orig/jdk/make/sun/font/Makefile openjdk/jdk/make/sun/font/Makefile
--- openjdk.orig/jdk/make/sun/font/Makefile	2010-02-12 08:15:45.000000000 +0000
+++ openjdk/jdk/make/sun/font/Makefile	2010-02-19 18:26:43.000000000 +0000
@@ -129,7 +129,7 @@
   ifeq ($(USING_SYSTEM_FT_LIB), false)
     FREETYPE_LIB = $(LIB_LOCATION)/$(LIB_PREFIX)freetype.$(LIBRARY_SUFFIX).6
   endif
-  OTHER_LDLIBS += -L$(FREETYPE_LIB_PATH) -lfreetype
+  OTHER_LDLIBS += -L$(FREETYPE_LIB_PATH) $(FT2_LIB)
 endif 
 
 library:: $(FREETYPE_LIB)
@@ -159,11 +159,13 @@
 #
 CFLAGS       += -DHEADLESS
 CXXFLAGS     += -DHEADLESS
+FREETYPE_HEADERS=$(shell $(ECHO) \"$(FREETYPE2_HEADERS)\" | grep -o .I.*[a\-z])
 
 ifndef OPENJDK
   CPPFLAGS += -I$(CLOSED_SRC)/share/native/$(PKGDIR)/t2k
 else
-  CPPFLAGS += -I$(FREETYPE_HEADERS_PATH) -I$(FREETYPE_HEADERS_PATH)/freetype2
+  CPPFLAGS += $(FREETYPE_HEADERS) $(FREETYPE_HEADERS)/freetype2 \
+	-I$(FREETYPE_HEADERS_PATH) -I$(FREETYPE_HEADERS_PATH)/freetype2
 endif
 
 ifeq ($(PLATFORM), windows)
diff -Nru openjdk.orig/jdk/make/tools/freetypecheck/Makefile openjdk/jdk/make/tools/freetypecheck/Makefile
--- openjdk.orig/jdk/make/tools/freetypecheck/Makefile	2009-12-04 23:26:49.000000000 +0000
+++ openjdk/jdk/make/tools/freetypecheck/Makefile	2010-02-19 18:08:00.000000000 +0000
@@ -50,7 +50,7 @@
   ifeq ($(PLATFORM), solaris)
     FT_LD_OPTIONS += -R $(FREETYPE_LIB_PATH) -lfreetype
   else #linux
-      FT_LD_OPTIONS += -Wl,-rpath -Wl,$(FREETYPE_LIB_PATH) -lfreetype
+      FT_LD_OPTIONS += -Wl,-rpath -Wl,$(FREETYPE_LIB_PATH) $(FT2_LIB)
   endif
 endif
 FT_OPTIONS += -I$(FREETYPE_HEADERS_PATH)
diff -Nru openjdk.orig/hotspot/make/linux/makefiles/jvmti.make openjdk/hotspot/make/linux/makefiles/jvmti.make
--- openjdk.orig/hotspot/make/linux/makefiles/jvmti.make	2008-05-23 20:51:14.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/jvmti.make	2008-05-23 20:54:13.000000000 +0100
@@ -59,7 +59,7 @@
 
 JvmtiGeneratedFiles = $(JvmtiGeneratedNames:%=$(JvmtiOutDir)/%)
 
-XSLT = $(QUIETLY) $(REMOTE) $(RUN.JAVA) -classpath $(JvmtiOutDir) jvmtiGen
+XSLT = $(QUIETLY) $(REMOTE) $(RUN.JAVA) $(ENDORSED) -classpath $(JvmtiOutDir) jvmtiGen
 
 .PHONY: all jvmtidocs clean cleanall
 

