diff -Nru openjdk.orig/jdk/src/solaris/native/java/lang/java_props_md.c openjdk/jdk/src/solaris/native/java/lang/java_props_md.c
--- openjdk.orig/jdk/src/solaris/native/java/lang/java_props_md.c	2010-11-12 01:20:49.000000000 +0000
+++ openjdk/jdk/src/solaris/native/java/lang/java_props_md.c	2010-11-12 18:31:00.049493600 +0000
@@ -118,7 +118,7 @@
 #endif
 
 static int ParseLocale(int cat, char ** std_language, char ** std_country, char ** std_variant, char ** std_encoding) {
-    char temp[64];
+    char *temp;
     char *language = NULL, *country = NULL, *variant = NULL,
          *encoding = NULL;
     char *p, encoding_variant[64];
@@ -162,6 +162,7 @@
      * <country name>, <encoding name>, and <variant name> are optional.
      */
 
+    temp = (char*) malloc(strlen(lc)+1);
     strcpy(temp, lc);
 
     /* Parse the language, country, encoding, and variant from the
@@ -282,6 +283,10 @@
 #endif
     }
 
+	
+    /* Free temp */
+    free(temp);
+
     return 1;
 }
 
