<<<<<<< local
diff -Nru openjdk.orig/hotspot/src/share/vm/runtime/os.cpp openjdk/hotspot/src/share/vm/runtime/os.cpp
--- openjdk.orig/hotspot/src/share/vm/runtime/os.cpp	2008-10-23 08:41:05.000000000 +0100
+++ openjdk/hotspot/src/share/vm/runtime/os.cpp	2008-10-27 23:00:47.000000000 +0000
=======
--- openjdk/hotspot/src/share/vm/runtime/os.cpp.orig	2008-11-22 00:11:18.000000000 +0000
+++ openjdk/hotspot/src/share/vm/runtime/os.cpp	2008-12-03 21:16:14.000000000 +0000
>>>>>>> other
@@ -886,6 +886,7 @@
         "%/lib/jsse.jar:"
         "%/lib/jce.jar:"
         "%/lib/charsets.jar:"
<<<<<<< local
+	"%/lib/rhino.jar:"
=======
+        "%/lib/rhino.jar:"
>>>>>>> other
         "%/classes";
     char* sysclasspath = format_boot_path(classpath_format, home, home_len, fileSep, pathSep);
     if (sysclasspath == NULL) return false;
<<<<<<< local
diff -Nru openjdk.orig/jdk/make/com/sun/Makefile openjdk/jdk/make/com/sun/Makefile
--- openjdk.orig/jdk/make/com/sun/Makefile	2008-10-23 08:44:04.000000000 +0100
+++ openjdk/jdk/make/com/sun/Makefile	2008-10-27 23:00:47.000000000 +0000
=======
--- openjdk/jdk/make/com/sun/script/Makefile.orig	2008-11-25 09:01:10.000000000 +0000
+++ openjdk/jdk/make/com/sun/script/Makefile	2008-12-03 21:13:14.000000000 +0000
@@ -31,6 +31,8 @@
 
 AUTO_FILES_JAVA_DIRS = com/sun/script
 
+OTHER_JAVACFLAGS = -classpath $(RHINO_JAR)
+
 #
 # Files that need to be copied
 #
--- openjdk/jdk/make/com/sun/Makefile.orig	2008-11-25 09:01:09.000000000 +0000
+++ openjdk/jdk/make/com/sun/Makefile	2008-12-03 21:14:36.000000000 +0000
>>>>>>> other
@@ -31,15 +31,8 @@
 PRODUCT = sun
 include $(BUILDDIR)/common/Defs.gmk
 
-ifndef OPENJDK
-  ORG_EXISTS := $(call DirExists,$(CLOSED_SRC)/share/classes/sun/org,,)
-  ifneq ("$(ORG_EXISTS)", "") 
-    SCRIPT_SUBDIR = script
-  endif
-endif
-
 # Omit mirror since it's built with the apt tool.
-SUBDIRS = $(SCRIPT_SUBDIR) image security crypto/provider jndi jmx \
+SUBDIRS = script image security crypto/provider jndi jmx \
<<<<<<< local
     java inputmethods org xml rowset net/httpserver net/ssl demo \
     tools jarsigner tracing servicetag
=======
     java inputmethods org rowset net/httpserver net/ssl demo \
     tools jarsigner
>>>>>>> other
 
<<<<<<< local
diff -Nru openjdk.orig/jdk/make/com/sun/script/Makefile openjdk/jdk/make/com/sun/script/Makefile
--- openjdk.orig/jdk/make/com/sun/script/Makefile	2008-10-23 08:44:04.000000000 +0100
+++ openjdk/jdk/make/com/sun/script/Makefile	2008-10-27 23:00:47.000000000 +0000
@@ -31,6 +31,8 @@
 
 AUTO_FILES_JAVA_DIRS = com/sun/script
 
+OTHER_JAVACFLAGS = -classpath $(RHINO_JAR)
+
 #
 # Files that need to be copied
 #
diff -Nru openjdk.orig/jdk/make/common/Release.gmk openjdk/jdk/make/common/Release.gmk
--- openjdk.orig/jdk/make/common/Release.gmk	2008-10-27 23:00:09.000000000 +0000
+++ openjdk/jdk/make/common/Release.gmk	2008-10-27 23:00:47.000000000 +0000
@@ -774,6 +774,7 @@
=======
--- openjdk/jdk/make/common/Release.gmk.orig	2008-12-03 21:12:23.000000000 +0000
+++ openjdk/jdk/make/common/Release.gmk	2008-12-03 21:13:14.000000000 +0000
@@ -772,6 +772,7 @@
>>>>>>> other
 	$(CP) $(RT_JAR) $(JRE_IMAGE_DIR)/lib/rt.jar
 	$(CP) $(RESOURCES_JAR) $(JRE_IMAGE_DIR)/lib/resources.jar
 	$(CP) $(JSSE_JAR) $(JRE_IMAGE_DIR)/lib/jsse.jar
+	$(LN) -sf $(RHINO_JAR) $(JRE_IMAGE_DIR)/lib/rhino.jar
 	@# Generate meta-index to make boot and extension class loaders lazier
 	$(CD) $(JRE_IMAGE_DIR)/lib && \
 	    $(BOOT_JAVA_CMD) -jar $(BUILDMETAINDEX_JARFILE) \
<<<<<<< local
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/ExternalScriptable.java openjdk/jdk/src/share/classes/com/sun/script/javascript/ExternalScriptable.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/ExternalScriptable.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/ExternalScriptable.java	2008-10-27 23:00:47.000000000 +0000
=======
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoTopLevel.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoTopLevel.java	2008-12-03 21:13:13.000000000 +0000
@@ -25,7 +25,7 @@
 
 package com.sun.script.javascript;
 
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 import javax.script.*;
 
 /**
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngineFactory.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngineFactory.java	2008-12-03 21:13:05.000000000 +0000
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 import javax.script.*;
 import java.util.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 import com.sun.script.util.*;
 
 /**
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/ExternalScriptable.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/ExternalScriptable.java	2008-12-03 21:13:14.000000000 +0000
>>>>>>> other
@@ -24,7 +24,7 @@
  */
 
 package com.sun.script.javascript;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 import javax.script.*;
 import java.util.*;
 
<<<<<<< local
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/JavaAdapter.java openjdk/jdk/src/share/classes/com/sun/script/javascript/JavaAdapter.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/JavaAdapter.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/JavaAdapter.java	2008-10-27 23:00:47.000000000 +0000
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 
 import javax.script.Invocable;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * This class implements Rhino-like JavaAdapter to help implement a Java
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/JSAdapter.java openjdk/jdk/src/share/classes/com/sun/script/javascript/JSAdapter.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/JSAdapter.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/JSAdapter.java	2008-10-27 23:00:47.000000000 +0000
@@ -25,7 +25,7 @@
 
 package com.sun.script.javascript;
 
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 import java.util.*;
 
 /**
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoClassShutter.java openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoClassShutter.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoClassShutter.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoClassShutter.java	2008-10-27 23:00:47.000000000 +0000
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 
 import java.util.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * This class prevents script access to certain sensitive classes.
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoCompiledScript.java openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoCompiledScript.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoCompiledScript.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoCompiledScript.java	2008-10-27 23:00:47.000000000 +0000
@@ -25,7 +25,7 @@
 
 package com.sun.script.javascript;
 import javax.script.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * Represents compiled JavaScript code.
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngineFactory.java openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngineFactory.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngineFactory.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngineFactory.java	2008-10-27 23:00:47.000000000 +0000
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 import javax.script.*;
 import java.util.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 import com.sun.script.util.*;
 
 /**
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngine.java openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngine.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngine.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngine.java	2008-10-27 23:00:47.000000000 +0000
=======
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngine.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoScriptEngine.java	2008-12-03 21:13:05.000000000 +0000
>>>>>>> other
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 import com.sun.script.util.*;
 import javax.script.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 import java.lang.reflect.Method;
 import java.io.*;
 import java.util.*;
<<<<<<< local
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoTopLevel.java openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoTopLevel.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoTopLevel.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoTopLevel.java	2008-10-27 23:00:47.000000000 +0000
=======
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoClassShutter.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoClassShutter.java	2008-12-03 21:13:05.000000000 +0000
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 
 import java.util.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * This class prevents script access to certain sensitive classes.
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/JavaAdapter.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/JavaAdapter.java	2008-12-03 21:13:14.000000000 +0000
@@ -26,7 +26,7 @@
 package com.sun.script.javascript;
 
 import javax.script.Invocable;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * This class implements Rhino-like JavaAdapter to help implement a Java
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/JSAdapter.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/JSAdapter.java	2008-12-03 21:13:05.000000000 +0000
>>>>>>> other
@@ -25,7 +25,7 @@
 
 package com.sun.script.javascript;
 
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
<<<<<<< local
 import javax.script.*;
=======
 import java.util.*;
>>>>>>> other
 
 /**
<<<<<<< local
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoWrapFactory.java openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoWrapFactory.java
--- openjdk.orig/jdk/src/share/classes/com/sun/script/javascript/RhinoWrapFactory.java	2008-10-23 08:44:13.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoWrapFactory.java	2008-10-27 23:00:47.000000000 +0000
=======
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoWrapFactory.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoWrapFactory.java	2008-12-03 21:13:14.000000000 +0000
>>>>>>> other
@@ -27,7 +27,7 @@
 
 import java.lang.reflect.*;
 import static sun.security.util.SecurityConstants.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * This wrap factory is used for security reasons. JSR 223 script
<<<<<<< local
=======
--- openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoCompiledScript.java.orig	2008-11-25 09:02:56.000000000 +0000
+++ openjdk/jdk/src/share/classes/com/sun/script/javascript/RhinoCompiledScript.java	2008-12-03 21:13:05.000000000 +0000
@@ -25,7 +25,7 @@
 
 package com.sun.script.javascript;
 import javax.script.*;
-import sun.org.mozilla.javascript.internal.*;
+import org.mozilla.javascript.*;
 
 /**
  * Represents compiled JavaScript code.
>>>>>>> other
