--- openjdk/hotspot/src/share/vm/asm/codeBuffer.hpp.orig	2008-08-14 16:24:58.000000000 +0200
+++ openjdk/hotspot/src/share/vm/asm/codeBuffer.hpp	2008-08-14 16:25:20.000000000 +0200
@@ -84,7 +84,7 @@
   address     _locs_point;      // last relocated position (grows upward)
   bool        _locs_own;        // did I allocate the locs myself?
   bool        _frozen;          // no more expansion of this section
-  char        _index;           // my section number (SECT_INST, etc.)
+  signed char _index;           // my section number (SECT_INST, etc.)
   CodeBuffer* _outer;           // enclosing CodeBuffer
 
   // (Note:  _locs_point used to be called _last_reloc_offset.)
--- openjdk/hotspot/src/share/vm/ci/ciTypeFlow.cpp.orig	2008-08-14 16:21:47.000000000 +0200
+++ openjdk/hotspot/src/share/vm/ci/ciTypeFlow.cpp	2008-08-14 16:25:51.000000000 +0200
@@ -391,7 +391,7 @@
   // Set the rest of the locals to bottom.
   Cell cell = state->next_cell(state->tos());
   state->set_stack_size(0);
-  int limit = state->limit_cell();
+  Cell limit = state->limit_cell();
   for (; cell < limit; cell = state->next_cell(cell)) {
     state->set_type_at(cell, state->bottom_type());
   }
