diff -Nru openjdk.orig/jdk/test/sun/tools/native2ascii/NativeErrors.java openjdk/jdk/test/sun/tools/native2ascii/NativeErrors.java
--- openjdk.orig/jdk/test/sun/tools/native2ascii/NativeErrors.java	2009-05-12 23:26:50.000000000 +0100
+++ openjdk/jdk/test/sun/tools/native2ascii/NativeErrors.java	2009-05-12 23:28:57.000000000 +0100
@@ -89,8 +89,8 @@
         }
         //System.out.println("received: " + errorReceived);
         //System.out.println("expected: " + errorExpected);
-        if (!errorReceived.endsWith(errorExpected))
-            throw new RuntimeException("Native2ascii bad arg error broken.");
+        if (errorReceived == null || !errorReceived.endsWith(errorExpected))
+            throw new RuntimeException("Native2ascii " + errorExpected + " broken.");
     }
 
     private static String[] getComString(String arg2) {
