diff -r 6a85c2ba896a openjdk/hotspot/src/os/linux/vm/os_linux.cpp
--- openjdk/hotspot/src/os/linux/vm/os_linux.cpp	Fri Feb 29 10:10:42 2008 +0000
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	Fri Feb 29 10:17:34 2008 +0000
@@ -845,7 +845,12 @@ bool os::create_thread(Thread* thread, T
     }
 
     stack_size = MAX2(stack_size, os::Linux::min_stack_allowed);
+#ifdef PPC
+    // https://bugzilla.redhat.com/show_bug.cgi?id=435337
+    pthread_attr_setstacksize(&attr, stack_size + 64*K);
+#else
     pthread_attr_setstacksize(&attr, stack_size);
+#endif // PPC
   } else {
     // let pthread_create() pick the default value.
   }
