--- openjdk/jdk/src/share/classes/sun/applet/PluginAppletViewer.java.orig	2008-04-07 11:31:28.000000000 -0400
+++ openjdk/jdk/src/share/classes/sun/applet/PluginAppletViewer.java	2008-04-07 11:30:52.000000000 -0400
@@ -44,6 +44,7 @@
 import sun.awt.AppContext;
 import sun.awt.X11.*;
 import java.lang.ref.WeakReference;
+import net.sourceforge.jnlp.NetxPanel;
 
 /**
  * Lets us construct one using unix-style one shot behaviors
@@ -97,14 +98,21 @@
     /**
      * Create the applet viewer
      */
-    public PluginAppletViewer(long handle, int x, int y, URL doc,
-                              Hashtable atts, PrintStream statusMsgStream,
+    public PluginAppletViewer(long handle, int x, int y, final URL doc,
+                              final Hashtable atts, PrintStream statusMsgStream,
                               PluginAppletViewerFactory factory) {
         super(handle, true);
     	this.factory = factory;
 	this.statusMsgStream = statusMsgStream;
 
-	add("Center", panel = new AppletViewerPanel(doc, atts));
+	try {
+		panel = new NetxPanel(doc, atts);
+		AppletViewerPanel.debug("Using NetX panel");
+	} catch (Exception ex) {
+		AppletViewerPanel.debug("Unable to start NetX applet - defaulting to Sun applet", ex);
+		panel = new AppletViewerPanel(doc, atts);
+	}
+	add("Center", panel);
 	panel.init();
 	appletPanels.addElement(panel);
 
