diff -urN openjdk-ecj.orig/hotspot/build/linux/Makefile openjdk-ecj/hotspot/build/linux/Makefile
--- openjdk-ecj.orig/hotspot/build/linux/Makefile	2007-08-02 03:16:18.000000000 -0400
+++ openjdk-ecj/hotspot/build/linux/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -241,28 +241,24 @@
 
 $(TARGETS_C2):  $(SUBDIRS_C2)
 	cd $(OSNAME)_$(BUILDARCH)_compiler2/$@ && $(MAKE) $(MFLAGS)
-	cd $(OSNAME)_$(BUILDARCH)_compiler2/$@ && ./test_gamma
 ifdef INSTALL
 	cd $(OSNAME)_$(BUILDARCH)_compiler2/$@ && $(MAKE) $(MFLAGS) install
 endif
 
 $(TARGETS_TIERED):  $(SUBDIRS_TIERED)
 	cd $(OSNAME)_$(BUILDARCH)_tiered/$(patsubst %tiered,%,$@) && $(MAKE) $(MFLAGS)
-	cd $(OSNAME)_$(BUILDARCH)_tiered/$(patsubst %tiered,%,$@) && ./test_gamma
 ifdef INSTALL
 	cd $(OSNAME)_$(BUILDARCH)_tiered/$(patsubst %tiered,%,$@) && $(MAKE) $(MFLAGS) install
 endif
 
 $(TARGETS_C1):  $(SUBDIRS_C1)
 	cd $(OSNAME)_$(BUILDARCH)_compiler1/$(patsubst %1,%,$@) && $(MAKE) $(MFLAGS)
-	cd $(OSNAME)_$(BUILDARCH)_compiler1/$(patsubst %1,%,$@) && ./test_gamma
 ifdef INSTALL
 	cd $(OSNAME)_$(BUILDARCH)_compiler1/$(patsubst %1,%,$@) && $(MAKE) $(MFLAGS) install
 endif
 
 $(TARGETS_CORE):  $(SUBDIRS_CORE)
 	cd $(OSNAME)_$(BUILDARCH)_core/$(patsubst %core,%,$@) && $(MAKE) $(MFLAGS)
-	cd $(OSNAME)_$(BUILDARCH)_core/$(patsubst %core,%,$@) && ./test_gamma
 ifdef INSTALL
 	cd $(OSNAME)_$(BUILDARCH)_core/$(patsubst %core,%,$@) && $(MAKE) $(MFLAGS) install
 endif
diff -urN openjdk-ecj.orig/hotspot/build/linux/makefiles/jvmti.make openjdk-ecj/hotspot/build/linux/makefiles/jvmti.make
--- openjdk-ecj.orig/hotspot/build/linux/makefiles/jvmti.make	2007-08-02 03:16:18.000000000 -0400
+++ openjdk-ecj/hotspot/build/linux/makefiles/jvmti.make	2007-08-08 16:29:33.000000000 -0400
@@ -57,7 +57,7 @@
 
 JvmtiGeneratedFiles = $(JvmtiGeneratedNames:%=$(JvmtiOutDir)/%)
 
-XSLT = $(QUIETLY) $(REMOTE) $(RUN.JAVA) -classpath $(JvmtiOutDir) jvmtiGen
+XSLT = $(QUIETLY) $(REMOTE) $(RUN.JAVA) $(ENDORSED) -classpath $(JvmtiOutDir) jvmtiGen
 
 .PHONY: all jvmtidocs clean cleanall
 
diff -urN openjdk-ecj.orig/hotspot/build/linux/makefiles/rules.make openjdk-ecj/hotspot/build/linux/makefiles/rules.make
--- openjdk-ecj.orig/hotspot/build/linux/makefiles/rules.make	2007-08-02 03:16:18.000000000 -0400
+++ openjdk-ecj/hotspot/build/linux/makefiles/rules.make	2007-08-08 16:29:33.000000000 -0400
@@ -77,7 +77,7 @@
 RUN.JAVAP = $(ALT_BOOTDIR)/bin/javap
 RUN.JAVAH = $(ALT_BOOTDIR)/bin/javah
 RUN.JAR   = $(ALT_BOOTDIR)/bin/jar
-COMPILE.JAVAC = $(ALT_BOOTDIR)/bin/javac
+COMPILE.JAVAC = $(ALT_BOOTDIR)/bin/javac $(BOOTCLASSPATH_RT_LIBGCJ)
 COMPILE.RMIC = $(ALT_BOOTDIR)/bin/rmic
 BOOT_JAVA_HOME = $(ALT_BOOTDIR)
 
diff -urN openjdk-ecj.orig/j2se/make/common/Release.gmk openjdk-ecj/j2se/make/common/Release.gmk
--- openjdk-ecj.orig/j2se/make/common/Release.gmk	2007-08-02 03:17:53.000000000 -0400
+++ openjdk-ecj/j2se/make/common/Release.gmk	2007-08-08 16:29:33.000000000 -0400
@@ -979,17 +979,17 @@
 	        $(JAR_JFLAGS)
 	@$(java-vm-cleanup)
 	$(CP) $(LIBDIR)/tools.jar $(JDK_IMAGE_DIR)/lib/tools.jar
-	@#
-	@# lib/ct.sym
-	@#
-	$(JAVAC) -XDprocess.packages -proc:only \
-	    -processor com.sun.tools.javac.sym.CreateSymbols \
-	    -Acom.sun.tools.javac.sym.Jar=$(RT_JAR) \
-	    -Acom.sun.tools.javac.sym.Dest=$(OUTPUTDIR)/symbols/META-INF/sym/rt.jar \
-	    $(CORE_PKGS) $(NON_CORE_PKGS) $(EXCLUDE_PROPWARN_PKGS)
-	$(JAR) c0f $(LIBDIR)/ct.sym -C $(OUTPUTDIR)/symbols META-INF $(JAR_JFLAGS)
-	@$(java-vm-cleanup)
-	$(CP) $(LIBDIR)/ct.sym $(JDK_IMAGE_DIR)/lib/ct.sym
+# 	@#
+# 	@# lib/ct.sym
+# 	@#
+# 	$(JAVAC) -XDprocess.packages -proc:only \
+# 	    -processor com.sun.tools.javac.sym.CreateSymbols \
+# 	    -Acom.sun.tools.javac.sym.Jar=$(RT_JAR) \
+# 	    -Acom.sun.tools.javac.sym.Dest=$(OUTPUTDIR)/symbols/META-INF/sym/rt.jar \
+# 	    $(CORE_PKGS) $(NON_CORE_PKGS) $(EXCLUDE_PROPWARN_PKGS)
+# 	$(JAR) c0f $(LIBDIR)/ct.sym -C $(OUTPUTDIR)/symbols META-INF $(JAR_JFLAGS)
+# 	@$(java-vm-cleanup)
+# 	$(CP) $(LIBDIR)/ct.sym $(JDK_IMAGE_DIR)/lib/ct.sym
 	@#
 	@# CORBA supported orb.idl and ir.idl should be copied to lib
 	@#
diff -urN openjdk-ecj.orig/j2se/make/common/Rules.gmk openjdk-ecj/j2se/make/common/Rules.gmk
--- openjdk-ecj.orig/j2se/make/common/Rules.gmk	2007-08-02 03:17:53.000000000 -0400
+++ openjdk-ecj/j2se/make/common/Rules.gmk	2007-08-08 16:29:33.000000000 -0400
@@ -286,10 +286,12 @@
 $(CLASSHDR_DOTFILE): $(CLASSES_export)
 	$(prep-target)
 	@$(ECHO) "# Running javah:"
-	$(JAVAH_CMD) $(JAVAHFLAGS) -bootclasspath $(CLASSDESTDIR) \
+	$(JAVAH_CMD) $(JAVAHFLAGS) -bootclasspath $(CLASSDESTDIR):$(ICEDTEA_RT) \
 		-d $(CLASSHDRDIR)/ \
 		$(CLASSES.export) $(subst $$,\$$,$(EXPORTED_inner))
 	@$(java-vm-cleanup)
+	-mv $(CLASSHDRDIR)/java_lang_ClassLoader\$$NativeLibrary.h \
+	  $(CLASSHDRDIR)/java_lang_ClassLoader_NativeLibrary.h
 	@$(TOUCH) $@
 
 classheaders.clean:
diff -urN openjdk-ecj.orig/j2se/make/java/applet/Makefile openjdk-ecj/j2se/make/java/applet/Makefile
--- openjdk-ecj.orig/j2se/make/java/applet/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/java/applet/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -27,6 +27,7 @@
 PACKAGE = java.applet
 PRODUCT = sun
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files
diff -urN openjdk-ecj.orig/j2se/make/java/awt/Makefile openjdk-ecj/j2se/make/java/awt/Makefile
--- openjdk-ecj.orig/j2se/make/java/awt/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/java/awt/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -46,6 +46,7 @@
 
 endif
 
+JAVAC_BCP = $(ICEDTEA_RT)
 build: sources properties cursors
 
 #
diff -urN openjdk-ecj.orig/j2se/make/java/beans/Makefile openjdk-ecj/j2se/make/java/beans/Makefile
--- openjdk-ecj.orig/j2se/make/java/beans/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/java/beans/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -31,6 +31,7 @@
 PACKAGE = java.beans
 PRODUCT = java
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files to compile.
diff -urN openjdk-ecj.orig/j2se/make/java/management/Makefile openjdk-ecj/j2se/make/java/management/Makefile
--- openjdk-ecj.orig/j2se/make/java/management/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/java/management/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -28,6 +28,7 @@
 LIBRARY = management
 PRODUCT = java
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 MGMT_SRC  = $(SHARE_SRC)/classes/java/lang/management
 SMGMT_SRC = $(SHARE_SRC)/classes/sun/management
diff -urN openjdk-ecj.orig/j2se/make/java/text/Makefile openjdk-ecj/j2se/make/java/text/Makefile
--- openjdk-ecj.orig/j2se/make/java/text/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/java/text/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -80,19 +80,19 @@
 
 $(BUILDER_CLASS): $(BUILDER)
 	$(prep-target)
-	$(JAVAC_BOOT) -d $(GENBIDOUT) -sourcepath $(GENBIDSRC) $(BUILDER)
+	$(JAVAC_BOOT) -d $(GENBIDOUT) $(BOOTCLASSPATH_CLS_RT) -sourcepath $(GENBIDSRC) $(BUILDER)
 	@$(java-vm-cleanup)
 
 $(BIRULES_CLASS): $(BIRULES)
 	$(prep-target)
 	$(JAVAC_BOOT) -d $(GENBIDOUT) \
-		-sourcepath $(SHARE_SRC)/classes/sun/text/resources $(BIRULES)
+		-sourcepath $(SHARE_SRC)/classes/sun/text/resources $(BOOTCLASSPATH_CLS_RT) $(BIRULES)
 	@$(java-vm-cleanup)
 
 $(BIINFO_CLASS): $(BIINFO)
 	$(prep-target)
 	$(JAVAC_BOOT) -d $(GENBIDOUT) \
-		-sourcepath $(SHARE_SRC)/classes/sun/text/resources $(BIINFO)
+		-sourcepath $(SHARE_SRC)/classes/sun/text/resources $(BOOTCLASSPATH_CLS_RT) $(BIINFO)
 	@$(java-vm-cleanup)
 
 $(BIFILES): $(BUILDER_CLASS) $(BIRULES_CLASS) $(BIINFO_CLASS) $(UNICODE)
@@ -102,6 +102,7 @@
 		-o $(CLASSBINDIR)/sun/text/resources -spec $(UNICODE)
 	@$(java-vm-cleanup)
 
+JAVAC_BCP = $(ICEDTEA_RT)
 bifiles: $(BIFILES) $(CLASSBINDIR)/sun/text/resources/unorm.icu $(CLASSBINDIR)/sun/text/resources/uprops.icu
 
 build: bifiles
diff -urN openjdk-ecj.orig/j2se/make/javax/accessibility/Makefile openjdk-ecj/j2se/make/javax/accessibility/Makefile
--- openjdk-ecj.orig/j2se/make/javax/accessibility/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/accessibility/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -27,6 +27,7 @@
 PACKAGE = java.accessibility
 PRODUCT = java
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files
diff -urN openjdk-ecj.orig/j2se/make/javax/activation/Makefile openjdk-ecj/j2se/make/javax/activation/Makefile
--- openjdk-ecj.orig/j2se/make/javax/activation/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/activation/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -31,6 +31,7 @@
 PACKAGE = javax.activation
 PRODUCT = jaf
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files to compile
diff -urN openjdk-ecj.orig/j2se/make/javax/imageio/Makefile openjdk-ecj/j2se/make/javax/imageio/Makefile
--- openjdk-ecj.orig/j2se/make/javax/imageio/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/imageio/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -31,6 +31,7 @@
 PACKAGE = javax.imageio
 PRODUCT = jiio
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files to compile
diff -urN openjdk-ecj.orig/j2se/make/javax/naming/Makefile openjdk-ecj/j2se/make/javax/naming/Makefile
--- openjdk-ecj.orig/j2se/make/javax/naming/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/naming/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -31,6 +31,7 @@
 PACKAGE = javax.naming
 PRODUCT = sun
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_JCE):$(ICEDTEA_RT)
 
 #
 # Files to compile
diff -urN openjdk-ecj.orig/j2se/make/javax/print/Makefile openjdk-ecj/j2se/make/javax/print/Makefile
--- openjdk-ecj.orig/j2se/make/javax/print/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/print/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -31,6 +31,7 @@
 PACKAGE = javax.print
 PRODUCT = sun
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_JCE):$(ICEDTEA_RT)
 
 #
 # Files to compile
diff -urN openjdk-ecj.orig/j2se/make/javax/security/Makefile openjdk-ecj/j2se/make/javax/security/Makefile
--- openjdk-ecj.orig/j2se/make/javax/security/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/security/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -27,6 +27,7 @@
 PACKAGE = javax.security
 PRODUCT = sun
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_JCE):$(ICEDTEA_RT)
 
 #
 # Files to compile
diff -urN openjdk-ecj.orig/j2se/make/javax/swing/Makefile openjdk-ecj/j2se/make/javax/swing/Makefile
--- openjdk-ecj.orig/j2se/make/javax/swing/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/swing/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -28,6 +28,7 @@
 PRODUCT   = com
 SWING_SRC = $(SHARE_SRC)/classes/javax/swing
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files
diff -urN openjdk-ecj.orig/j2se/make/javax/swing/plaf/Makefile openjdk-ecj/j2se/make/javax/swing/plaf/Makefile
--- openjdk-ecj.orig/j2se/make/javax/swing/plaf/Makefile	2007-08-02 03:17:54.000000000 -0400
+++ openjdk-ecj/j2se/make/javax/swing/plaf/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -28,6 +28,7 @@
 PRODUCT   = com
 SWING_SRC = $(SHARE_SRC)/classes/javax/swing
 include $(BUILDDIR)/common/Defs.gmk
+JAVAC_BCP = $(ICEDTEA_RT)
 
 #
 # Files
diff -urN openjdk-ecj.orig/j2se/make/mkdemo/jfc/DemoSwing.gmk openjdk-ecj/j2se/make/mkdemo/jfc/DemoSwing.gmk
--- openjdk-ecj.orig/j2se/make/mkdemo/jfc/DemoSwing.gmk	2007-08-02 03:17:55.000000000 -0400
+++ openjdk-ecj/j2se/make/mkdemo/jfc/DemoSwing.gmk	2007-08-08 16:29:33.000000000 -0400
@@ -108,7 +108,7 @@
 .compile.classlist:
 	if [ -s $(TEMPDIR)/.classes.list ] ; \
 	then $(JAVAC) $(OTHER_JAVACFLAGS) \
-        $(DEMO_JAVAC_ARGS) $(LANGUAGE_VERSION)  -d $(DEMODST_CLASS) -classpath $(DEMO_CP) \
+        $(DEMO_JAVAC_ARGS) $(BOOTCLASSPATH_CLS_RT) $(LANGUAGE_VERSION)  -d $(DEMODST_CLASS) -classpath $(DEMO_CP) \
 	    $(shell if [ -s $(TEMPDIR)/.classes.list ] ; then $(CAT) $(TEMPDIR)/.classes.list; fi ) ; \
 	fi
 	@$(java-vm-cleanup)
diff -urN openjdk-ecj.orig/j2se/make/mkdemo/management/Demo.gmk openjdk-ecj/j2se/make/mkdemo/management/Demo.gmk
--- openjdk-ecj.orig/j2se/make/mkdemo/management/Demo.gmk	2007-08-02 03:17:55.000000000 -0400
+++ openjdk-ecj/j2se/make/mkdemo/management/Demo.gmk	2007-08-08 16:29:33.000000000 -0400
@@ -65,7 +65,7 @@
 	$(RM) $(JAR_FILE)
 	$(RM) -r $(CLASSES_DIR)
 	$(MKDIR) -p $(CLASSES_DIR)
-	$(JAVAC) -d $(CLASSES_DIR) $(JAVA_SOURCES)
+	$(JAVAC) $(BOOTCLASSPATH_CLS) -d $(CLASSES_DIR) $(JAVA_SOURCES)
 	$(ECHO) "Main-Class: $(MAIN_CLASS_NAME)" > $(CLASSES_DIR)/manifest
 
 $(CLASSES_DIR)/META-INF/services/%: $(SRCDIR)/META-INF/services/%
diff -urN openjdk-ecj.orig/j2se/make/sun/awt/Makefile openjdk-ecj/j2se/make/sun/awt/Makefile
--- openjdk-ecj.orig/j2se/make/sun/awt/Makefile	2007-08-02 03:17:56.000000000 -0400
+++ openjdk-ecj/j2se/make/sun/awt/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -468,7 +468,7 @@
 	$(prep-target)
 	$(RM) -r $(TEMPDIR)/CompileFontConfig
 	$(MKDIR) -p $(TEMPDIR)/CompileFontConfig
-	$(JAVAC) -d $(TEMPDIR)/CompileFontConfig $(BUILDDIR)/tools/CompileFontConfig/CompileFontConfig.java
+	$(JAVAC) -d $(TEMPDIR)/CompileFontConfig $(BOOTCLASSPATH_CLS) $(BUILDDIR)/tools/CompileFontConfig/CompileFontConfig.java
 	$(JAVA) -classpath $(TEMPDIR)/CompileFontConfig CompileFontConfig $< $(@)
 	$(CHMOD) 444 $(@)
 	@$(java-vm-cleanup)
diff -urN openjdk-ecj.orig/j2se/make/sun/text/Makefile openjdk-ecj/j2se/make/sun/text/Makefile
--- openjdk-ecj.orig/j2se/make/sun/text/Makefile	2007-08-02 03:17:58.000000000 -0400
+++ openjdk-ecj/j2se/make/sun/text/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -90,7 +90,7 @@
                 $(GENBIDOUT)/SupplementaryCharacterData.class
 
 $(BUILDER_CLASS): $(GENBIDOUT) $(BUILDER)
-	$(JAVAC_BOOT) -d $(GENBIDOUT) -sourcepath $(GENBIDSRC) $(BUILDER)
+	$(JAVAC_BOOT) $(BOOTCLASSPATH_CLS) -d $(GENBIDOUT) -sourcepath $(GENBIDSRC) $(BUILDER)
 	@$(java-vm-cleanup)
 
 $(GENBIDOUT):
@@ -98,13 +98,13 @@
 
 $(BIRULES_CLASS): $(GENBIDOUT)/sun/text/resources $(BIRULES)
 	$(RM) $@
-	$(JAVAC_BOOT) -d $(GENBIDOUT) \
+	$(JAVAC_BOOT) $(BOOTCLASSPATH_CLS) -d $(GENBIDOUT)  \
 		-sourcepath $(SHARE_SRC)/classes/sun/text/resources $(BIRULES)
 	@$(java-vm-cleanup)
 
 $(BIINFO_CLASS): $(GENBIDOUT)/sun/text/resources $(BIINFO)
 	$(RM) $@
-	$(JAVAC_BOOT) -d $(GENBIDOUT) \
+	$(JAVAC_BOOT) $(BOOTCLASSPATH_CLS) -d $(GENBIDOUT) \
 		-sourcepath $(SHARE_SRC)/classes/sun/text/resources $(BIINFO)
 	@$(java-vm-cleanup)

diff -urN openjdk-ecj.orig/j2se/make/sun/xawt/Makefile openjdk-ecj/j2se/make/sun/xawt/Makefile
--- openjdk-ecj.orig/j2se/make/sun/xawt/Makefile	2007-08-02 03:17:58.000000000 -0400
+++ openjdk-ecj/j2se/make/sun/xawt/Makefile	2007-08-08 16:29:33.000000000 -0400
@@ -279,13 +279,13 @@
 	    $(RM) $$classname; \
 	    $(ECHO) "package sun.awt.X11; public class XAWTIcon32_$$name {" >> $$classname; \
 	    $(ECHO) "public static int[] $$name = { " >> $$classname;  \
-	    $(CAT) $$i | $(JAVA_BOOT) -classpath ${CLASSBINDIR}  -Dawt.toolkit=sun.awt.motif.MToolkit -Djava.awt.headless=true sun.awt.X11.ToBin >> $$classname; \
+            $(ECHO) "0,0" >> $$classname ; \
 	    $(ECHO) "}; }" >> $$classname;  \
             classname=$(GEN_DIR)/XAWTIcon64_$$name.java; \
 	    $(RM) $$classname; \
 	    $(ECHO) "package sun.awt.X11; public class XAWTIcon64_$$name {" >> $$classname; \
 	    $(ECHO) "public static long[] $$name = { " >> $$classname;  \
-	    $(CAT) $$i | $(JAVA_BOOT) -classpath ${CLASSBINDIR}  -Dawt.toolkit=sun.awt.motif.MToolkit -Djava.awt.headless=true sun.awt.X11.ToBin >> $$classname; \
+            $(ECHO) "0,0" >> $$classname ; \
 	    $(ECHO) "}; }" >> $$classname;  \
 	done
 	$(TOUCH) $@
