--- defs.make	2008-02-17 16:25:58.000000000 -0500
+++ openjdk/hotspot/make/defs.make	2008-02-17 16:27:04.000000000 -0500
@@ -213,6 +213,7 @@
       SRCARCH = amd64
     endif
   endif
+SCARCH = zero
 
   # BUILDARCH is usually the same as SRCARCH, except for sparcv9 and zero
   BUILDARCH = $(SRCARCH)
--- oldopenjdk6/hotspot/build/linux/platform_amd64	2008-02-12 04:14:13.000000000 -0500
+++ openjdk/hotspot/build/linux/platform_amd64	2008-02-17 16:28:12.000000000 -0500
@@ -1,8 +1,8 @@
 os_family = linux
 
-arch = amd64
+arch = zero
 
-os_arch = linux_amd64
+os_arch = linux_zero
 
 lib_arch = amd64
 
@@ -10,4 +10,4 @@
 
 gnu_dis_arch = amd64
 
-sysdefs = -DLINUX -D_GNU_SOURCE -DAMD64
+sysdefs = -DLINUX -D_GNU_SOURCE -DAMD64 -DCC_INTERP -DZERO
--- oldopenjdk6/hotspot/build/linux/platform_i486	2008-02-12 04:14:13.000000000 -0500
+++ openjdk/hotspot/build/linux/platform_i486	2008-02-17 16:29:59.000000000 -0500
@@ -1,8 +1,8 @@
 os_family = linux
 
-arch = i486
+arch = zero
 
-os_arch = linux_i486
+os_arch = linux_zero
 
 lib_arch = i386
 
@@ -10,4 +10,4 @@
 
 gnu_dis_arch = i386
 
-sysdefs = -DLINUX -D_GNU_SOURCE -DIA32
+sysdefs = -DLINUX -D_GNU_SOURCE -DIA32 -DCC_INTERP -DZERO
--- defs.make	2008-02-17 16:30:44.000000000 -0500
+++ openjdk/hotspot/build/linux/makefiles/defs.make	2008-02-17 16:31:36.000000000 -0500
@@ -68,12 +68,12 @@
     MAKE_ARGS       += LP64=1
     PLATFORM        = linux-amd64
     VM_PLATFORM     = linux_amd64
-    HS_ARCH         = amd64
+    HS_ARCH         = zero
   else
     ARCH_DATA_MODEL = 32
     PLATFORM        = linux-i586
     VM_PLATFORM     = linux_i486
-    HS_ARCH         = i486
+    HS_ARCH         = zero
     # We have to reset ARCH to i686 since SRCARCH relies on it
     ARCH            = i686   
   endif
@@ -84,7 +84,7 @@
   ARCH_DATA_MODEL  = 32
   PLATFORM         = linux-i586
   VM_PLATFORM      = linux_i486
-  HS_ARCH          = i486
+  HS_ARCH          = zero
 endif
 
 # ppc
--- amd64.make	2008-02-17 16:32:11.000000000 -0500
+++ openjdk/hotspot/build/linux/makefiles/amd64.make	2008-02-17 16:32:16.000000000 -0500
@@ -22,9 +22,6 @@
 #  
 #
 
-# Not included in includeDB because it has no dependencies
-Obj_Files += linux_amd64.o
-
 # The copied fdlibm routines in sharedRuntimeTrig.o must not be optimized
 OPT_CFLAGS/sharedRuntimeTrig.o = $(OPT_CFLAGS/NOOPT)
 # The copied fdlibm routines in sharedRuntimeTrans.o must not be optimized
--- i486.make	2008-02-17 16:32:53.000000000 -0500
+++ openjdk/hotspot/build/linux/makefiles/i486.make	2008-02-17 16:32:58.000000000 -0500
@@ -22,9 +22,6 @@
 #  
 #
 
-# TLS helper, assembled from .s file
-# Not included in includeDB because it has no dependencies
-Obj_Files += linux_i486.o
 
 # The copied fdlibm routines in sharedRuntimeTrig.o must not be optimized
 OPT_CFLAGS/sharedRuntimeTrig.o = $(OPT_CFLAGS/NOOPT)
--- jvm.cfg	2008-02-17 16:34:15.000000000 -0500
+++ openjdk/jdk/src/solaris/bin/amd64/jvm.cfg	2008-02-17 16:34:47.000000000 -0500
@@ -31,8 +31,9 @@
 # "-XXaltjvm=<jvm_dir>" option, but that too is unsupported
 # and may not be available in a future release.
 #
--server KNOWN
+-core KNOWN
 -client IGNORE
+-server IGNORE
 -hotspot ERROR
 -classic WARN
 -native ERROR
--- oldopenjdk6/jdk/src/solaris/bin/i586/jvm.cfg	2008-02-12 04:08:27.000000000 -0500
+++ openjdk/jdk/src/solaris/bin/i586/jvm.cfg	2008-02-17 16:35:54.000000000 -0500
@@ -30,9 +30,10 @@
 # "-XXaltjvm=<jvm_dir>" option, but that too is unsupported
 # and may not be available in a future release.
 #
--client IF_SERVER_CLASS -server
--server KNOWN
--hotspot ALIASED_TO -client
+-core KNOWN
+-client IGNORE
+-server IGNORE
+-hotspot ERROR
 -classic WARN
 -native ERROR
 -green ERROR
