diff -r 114e54fc994e openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreter.cpp
--- openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreter.cpp	Fri Apr 04 09:33:18 2008 +0100
+++ openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreter.cpp	Fri Apr 04 09:34:11 2008 +0100
@@ -1281,12 +1281,7 @@ run:
           jfloat f;
           jdouble r;
           f = STACK_FLOAT(-1);
-#ifdef IA64
-          // IA64 gcc bug
-          r = ( f == 0.0f ) ? (jdouble) f : (jdouble) f + ia64_double_zero;
-#else
           r = (jdouble) f;
-#endif
           MORE_STACK(-1); // POP
           SET_STACK_DOUBLE(r, 1);
           UPDATE_PC_AND_TOS_AND_CONTINUE(1, 2);
