diff -r 48f64c7da34f j2se/src/share/classes/sun/misc/FloatingDecimal.java
--- openjdk/j2se/src/share/classes/sun/misc/FloatingDecimal.java	Thu Sep 27 12:56:46 2007 -0700
+++ openjdk/j2se/src/share/classes/sun/misc/FloatingDecimal.java	Wed Oct 03 15:53:31 2007 -0700
@@ -730,7 +730,7 @@ public class FloatingDecimal{
 		 * Thus we will need more than one digit if we're using
 		 * E-form
 		 */
-		if ( decExp <= -3 || decExp >= 8 ){
+		if ( decExp < -3 || decExp >= 8 ){
 		    high = low = false;
 		}
 		while( ! low && ! high ){
@@ -783,7 +783,7 @@ public class FloatingDecimal{
 		 * Thus we will need more than one digit if we're using
 		 * E-form
 		 */
-		if ( decExp <= -3 || decExp >= 8 ){
+		if ( decExp < -3 || decExp >= 8 ){
 		    high = low = false;
 		}
 		while( ! low && ! high ){
@@ -847,7 +847,7 @@ public class FloatingDecimal{
 	     * Thus we will need more than one digit if we're using
 	     * E-form
 	     */
-	    if ( decExp <= -3 || decExp >= 8 ){
+	    if ( decExp < -3 || decExp >= 8 ){
 		high = low = false;
 	    }
 	    while( ! low && ! high ){
