--- openjdk/jdk/make/common/Defs-linux.gmk.ark	2008-01-11 14:20:26.000000000 +0000
+++ openjdk/jdk/make/common/Defs-linux.gmk	2008-01-11 14:21:10.000000000 +0000
@@ -139,7 +139,9 @@
 # Treat compiler warnings as errors, if warnings not allowed
 #
 ifeq ($(COMPILER_WARNINGS_FATAL),true)
-  GCC_WARNINGS += -Werror
+  # gcc 4.3 introduces new warnings OpenJDK code isn't quite ready for, such as
+  # "cast from double to float may chance its value". Let's ignore this for now.
+  # GCC_WARNINGS += -Werror
 endif
 
 #
