--- openjdk/control/make/make/jdk-rules.gmk	2008-06-29 10:09:47.002591764 +0000
+++ openjdk/control/make/make/jdk-rules.gmk	2008-06-29 15:04:01.228901751 +0000
@@ -36,12 +36,7 @@
   IMAGES_TARGET = images
 endif
 
-# No DOCS build when JDK_UPDATE_VERSION set
-ifdef JDK_UPDATE_VERSION
-  DOCS_TARGET =
-else
-  DOCS_TARGET = docs
-endif
+DOCS_TARGET = docs
 
 # NO_DOCS may be set in conjunction with DEV_ONLY
 ifdef NO_DOCS
