diff -urN openjdk.orig/hotspot/build/linux/makefiles/gcc.make openjdk/hotspot/build/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/build/linux/makefiles/gcc.make	2007-10-12 03:46:25.000000000 -0400
+++ openjdk/hotspot/build/linux/makefiles/gcc.make	2007-10-12 17:41:02.000000000 -0400
@@ -98,7 +98,7 @@
 CFLAGS_WARN/BYFILE = $(CFLAGS_WARN/$@)$(CFLAGS_WARN/DEFAULT$(CFLAGS_WARN/$@)) 
 
 # The flags to use for an Optimized g++ build
-OPT_CFLAGS += -O3
+OPT_CFLAGS += -O0 -g
 
 # Hotspot uses very unstrict aliasing turn this optimization off
 OPT_CFLAGS += -fno-strict-aliasing
diff -urN openjdk.orig/jdk/make/common/Defs-linux.gmk openjdk/jdk/make/common/Defs-linux.gmk
--- openjdk.orig/jdk/make/common/Defs-linux.gmk	2007-10-12 03:54:05.000000000 -0400
+++ openjdk/jdk/make/common/Defs-linux.gmk	2007-10-12 17:41:02.000000000 -0400
@@ -97,6 +97,7 @@
     _OPT = $(CC_LOWER_OPT)
     CPPFLAGS_DBG    += -DLOGGING 
 endif
+_OPT = -O0 -g
 
 # For all platforms, do not omit the frame pointer register usage. 
 #    We need this frame pointer to make it easy to walk the stacks.
diff -urN openjdk.orig/jdk/make/Makefile openjdk/jdk/make/Makefile
--- openjdk.orig/jdk/make/Makefile	2007-10-12 03:54:03.000000000 -0400
+++ openjdk/jdk/make/Makefile	2007-10-12 17:41:02.000000000 -0400
@@ -249,9 +249,9 @@
 #
 # Docs
 #
-OTHERSUBDIRS = docs
-docs:: sanity-docs post-sanity-docs
-	$(OTHERSUBDIRS-loop)
+#OTHERSUBDIRS = docs
+#docs:: sanity-docs post-sanity-docs
+#	$(OTHERSUBDIRS-loop)
 
 #
 # Release engineering targets.
