--- openjdk/jdk/src/share/classes/sun/java2d/pisces/Dasher.java.orig	2009-01-13 12:14:53.000000000 -0500
+++ openjdk/jdk/src/share/classes/sun/java2d/pisces/Dasher.java	2009-01-13 12:15:09.000000000 -0500
@@ -120,7 +120,7 @@
 
         // Normalize so 0 <= phase < dash[0]
         int idx = 0;
-        dashOn = false;
+        dashOn = true;
         int d;
         while (phase >= (d = dash[idx])) {
             phase -= d;
