diff -r 8e41d25d7665 openjdk/control/make/make/hotspot-rules.gmk
--- openjdk/control/make/make/hotspot-rules.gmk	Fri Jun 20 13:26:10 2008 +0100
+++ openjdk/control/make/make/hotspot-rules.gmk	Fri Jun 20 13:26:20 2008 +0100
@@ -75,6 +75,9 @@ ifdef ICEDTEA_CORE_BUILD
 ifdef ICEDTEA_CORE_BUILD
   HOTSPOT_TARGET := $(HOTSPOT_TARGET)core
 endif
+ifdef ICEDTEA_SHARK_BUILD
+  HOTSPOT_TARGET := $(HOTSPOT_TARGET)shark
+endif
 
 HOTSPOT_BUILD_ARGUMENTS += $(COMMON_BUILD_ARGUMENTS)
 HOTSPOT_BUILD_ARGUMENTS += ALT_OUTPUTDIR=$(HOTSPOT_OUTPUTDIR)
iff -r 8e41d25d7665 openjdk/jdk/make/java/redist/Makefile
--- openjdk/jdk/make/java/redist/Makefile	Fri Jun 20 13:26:10 2008 +0100
+++ openjdk/jdk/make/java/redist/Makefile	Fri Jun 20 13:28:19 2008 +0100
@@ -96,9 +96,11 @@ endif # INCLUDE_SA
 
 # Hotspot client is only available on 32-bit builds
 ifndef ICEDTEA_CORE_BUILD
+ifndef ICEDTEA_SHARK_BUILD
 ifeq ($(ARCH_DATA_MODEL), 32)
   IMPORT_LIST += $(LIB_LOCATION)/$(CLIENT_LOCATION)/$(JVM_NAME) \
                  $(LIB_LOCATION)/$(CLIENT_LOCATION)/Xusage.txt
+endif
 endif
 endif
 
@@ -164,6 +166,7 @@ endif
 endif 
 
 ifndef ICEDTEA_CORE_BUILD
+ifndef ICEDTEA_SHARK_BUILD
 ifeq ($(ARCH_DATA_MODEL), 32)
 
 IMPORT_LIST += $(LIB_LOCATION)/$(CLIENT_LOCATION)/$(LIBJSIG_NAME)
@@ -193,6 +196,8 @@ endif # 32bit solaris
 endif # 32bit solaris
 
 endif # 32bit
+
+endif # ICEDTEA_SHARK_BUILD
 
 endif # ICEDTEA_CORE_BUILD
 
