diff -Nru openjdk.orig/jdk/make/common/shared/Defs-versions.gmk openjdk/jdk/make/common/shared/Defs-versions.gmk
--- openjdk.orig/jdk/make/common/shared/Defs-versions.gmk	2011-03-14 22:10:30.000000000 +0000
+++ openjdk/jdk/make/common/shared/Defs-versions.gmk	2011-03-15 01:45:56.009913021 +0000
@@ -207,7 +207,7 @@
 # Generic
 REQUIRED_ANT_VER          = 1.6.3
 REQUIRED_BOOT_VER         = 1.6
-REQUIRED_FREETYPE_VERSION = 2.3.0
+REQUIRED_FREETYPE_VERSION = 2.2.1
 REQUIRED_MAKE_VER         = 3.81
 REQUIRED_UNZIP_VER        = 5.12
 REQUIRED_ZIP_VER          = 2.2
