--- ../openjdk/j2se/src/share/lib/security/java.security	2007-06-21 03:47:46.000000000 -0400
+++ openjdk/j2se/src/share/lib/security/java.security	2007-07-05 12:49:09.000000000 -0400
@@ -43,14 +43,19 @@
 #
 # List of providers and their preference orders (see above):
 #
-security.provider.1=sun.security.provider.Sun
-security.provider.2=sun.security.rsa.SunRsaSign
-security.provider.3=com.sun.net.ssl.internal.ssl.Provider
-security.provider.4=com.sun.crypto.provider.SunJCE
-security.provider.5=sun.security.jgss.SunProvider
-security.provider.6=com.sun.security.sasl.Provider
-security.provider.7=org.jcp.xml.dsig.internal.dom.XMLDSigRI
-security.provider.8=sun.security.smartcardio.SunPCSC
+security.provider.1=gnu.java.security.icedtea.IcedTls
+security.provider.2=sun.security.provider.Sun
+security.provider.3=sun.security.rsa.SunRsaSign
+security.provider.4=com.sun.net.ssl.internal.ssl.Provider
+security.provider.5=com.sun.crypto.provider.SunJCE
+security.provider.6=sun.security.jgss.SunProvider
+security.provider.7=com.sun.security.sasl.Provider
+security.provider.8=org.jcp.xml.dsig.internal.dom.XMLDSigRI
+security.provider.9=sun.security.smartcardio.SunPCSC
+security.provider.10=gnu.java.security.provider.Gnu
+security.provider.11=gnu.javax.crypto.jce.GnuCrypto
+security.provider.12=gnu.javax.crypto.jce.GnuSasl
+security.provider.13=gnu.javax.security.auth.callback.GnuCallbacks
 
 #
 # Select the source of seed data for SecureRandom. By default an
