--- openjdk/j2se/src/solaris/native/sun/awt/gtk2_interface.c.orig	2007-09-27 04:08:04.000000000 -0400
+++ openjdk/j2se/src/solaris/native/sun/awt/gtk2_interface.c	2007-10-04 16:31:04.000000000 -0400
@@ -1410,20 +1410,6 @@
             (*fp_gtk_menu_item_set_submenu)((GtkMenuItem*)root_menu, result);
             (*fp_gtk_menu_shell_append)((GtkMenuShell *)menu_bar, root_menu);
         }
-        else if (widget_type == COMBO_BOX_ARROW_BUTTON ||
-                 widget_type == COMBO_BOX_TEXT_FIELD)
-        {
-            /*
-             * We add a regular GtkButton/GtkEntry to a GtkComboBoxEntry
-             * in order to trick engines into thinking it's a real combobox
-             * arrow button/text field.
-             */
-            GtkWidget *combo = (*fp_gtk_combo_box_entry_new)();
-            (*fp_gtk_container_add)((GtkContainer *)combo, result);
-            (*fp_gtk_container_add)((GtkContainer *)gtk2_fixed, combo);
-            (*fp_gtk_widget_realize)(result);
-            return result;
-        }
         else if (widget_type != TOOL_TIP &&
                  widget_type != INTERNAL_FRAME &&
                  widget_type != OPTION_PANE)
