--- AccessibleContext.java.orig	2008-12-24 11:33:22.000000000 -0500
+++ openjdk/jdk/src/share/classes/javax/accessibility/AccessibleContext.java	2008-12-24 11:35:29.000000000 -0500
@@ -736,6 +736,14 @@
                                    Object oldValue,
                                    Object newValue) {
         if (accessibleChangeSupport != null) {
+
+            if (oldValue == newValue) {
+                return;
+            }
+            if (oldValue != null && newValue != null && oldValue.equals(newValue)) {
+                return;
+            }
+
             if (newValue instanceof PropertyChangeEvent) {
                 PropertyChangeEvent pce = (PropertyChangeEvent)newValue;
                 accessibleChangeSupport.firePropertyChange(pce);
