--- EncodingInfo.java.orig	2009-02-19 14:46:50.000000000 -0500
+++ openjdk/jaxp/src/share/classes/com/sun/org/apache/xml/internal/serializer/EncodingInfo.java	2009-02-20 10:31:37.000000000 -0500
@@ -326,9 +326,11 @@
             m_last = last;
 
             // Set the range of unicode values that this object
-            // explicitly manages
-            m_explFirst = codePoint;
-            m_explLast = codePoint + (RANGE-1);
+            // explicitly manages. Align the explicitly managed values
+            // to RANGE so multiple EncodingImpl objects dont manage the same 
+            // values.
+            m_explFirst = codePoint / RANGE * RANGE;
+            m_explLast = m_explFirst + (RANGE-1);
 
             m_encoding = encoding;
 

