Index: j2se/src/share/bin/java.c
===================================================================
--- openjdk/j2se/src/share/bin/java.c	(revision 256)
+++ j2se/src/share/bin/java.c	(working copy)
@@ -1955,7 +1955,8 @@ ContinueInNewThread(InvocationFunctions*
       args.classname = classname;
       args.ifn = *ifn;
 
-      rslt = ContinueInNewThread0(JavaMain, threadStackSize, (void*)&args);
+/*       rslt = ContinueInNewThread0(JavaMain, threadStackSize, (void*)&args); */
+      rslt = JavaMain((void*)&args);
       /* If the caller has deemed there is an error we
        * simply return that, otherwise we return the value of
        * the callee
