diff -Nru openjdk.orig/jdk/src/solaris/native/java/lang/java_props_md.c openjdk/jdk/src/solaris/native/java/lang/java_props_md.c
--- openjdk.orig/jdk/src/solaris/native/java/lang/java_props_md.c	2011-03-14 22:10:33.000000000 +0000
+++ openjdk/jdk/src/solaris/native/java/lang/java_props_md.c	2011-03-15 23:40:55.950143541 +0000
@@ -119,7 +119,7 @@
 
 static int ParseLocale(int cat, char ** std_language, char ** std_script,
                        char ** std_country, char ** std_variant, char ** std_encoding) {
-    char temp[64];
+    char *temp;
     char *language = NULL, *country = NULL, *variant = NULL,
          *encoding = NULL;
     char *p, encoding_variant[64];
@@ -163,6 +163,7 @@
      * <country name>, <encoding name>, and <variant name> are optional.
      */
 
+    temp = (char*) malloc(strlen(lc)+1);
     strcpy(temp, lc);
 
     /* Parse the language, country, encoding, and variant from the
@@ -301,6 +302,10 @@
 #endif
     }
 
+	
+    /* Free temp */
+    free(temp);
+
     return 1;
 }
 
