Index: j2se/make/java/instrument/Makefile
===================================================================
--- j2se/make/java/instrument/Makefile	(revision 250)
+++ openjdk/j2se/make/java/instrument/Makefile	(working copy)
@@ -109,7 +109,7 @@ else
     LDFLAGS += -R \$$ORIGIN/jli
   endif
   ifeq ($(PLATFORM), linux)
-    LDFLAGS += -z origin
+    LDFLAGS += -Xlinker -z -Xlinker origin
     LDFLAGS += -Wl,--allow-shlib-undefined
     LDFLAGS += -Wl,-rpath -Wl,\$$ORIGIN/jli
   endif
