--- orig	2007-06-29 13:17:17.000000000 -0400
+++ openjdk/j2se/src/share/lib/security/java.security	2007-06-29 13:17:33.000000000 -0400
@@ -54,6 +54,11 @@
 security.provider.6=com.sun.security.sasl.Provider
 security.provider.7=org.jcp.xml.dsig.internal.dom.XMLDSigRI
 security.provider.8=sun.security.smartcardio.SunPCSC
+security.provider.9=gnu.java.security.provider.Gnu
+security.provider.10=gnu.javax.crypto.jce.GnuCrypto
+security.provider.11=gnu.javax.crypto.jce.GnuSasl
+security.provider.12=gnu.javax.net.ssl.provider.Jessie
+security.provider.13=gnu.javax.security.auth.callback.GnuCallbacks
 
 #
 # Select the source of seed data for SecureRandom. By default an
