diff -r 99a06bc7fdb5 make/launchers/Makefile
--- a/make/launchers/Makefile	Tue Dec 04 16:28:33 2007 -0800
+++ openjdk/jdk/make/launchers/Makefile	Fri Dec 21 17:26:29 2007 -0500
@@ -87,6 +87,7 @@ define make-all-launchers
 $(call make-launcher, jstatd, sun.tools.jstatd.Jstatd, , )
 $(call make-launcher, keytool, sun.security.tools.KeyTool, , )
 $(call make-launcher, native2ascii, sun.tools.native2ascii.Main, , )
+$(call make-launcher, javaws, netx.jnlp.runtime.Boot13, , )
 $(call make-launcher, orbd, com.sun.corba.se.impl.activation.ORBD, \
   -J-Dcom.sun.CORBA.activation.DbDir=./orb.db \
   -J-Dcom.sun.CORBA.activation.Port=1049 \
--- openjdk/jdk/make/common/Release.gmk.orig	2008-01-02 15:10:01.000000000 -0500
+++ openjdk/jdk/make/common/Release.gmk	2008-01-02 15:10:29.000000000 -0500
@@ -469,7 +469,8 @@
 	wsimport$(EXE_SUFFIX) \
         schemagen$(EXE_SUFFIX) \
 	jsadebugd$(EXE_SUFFIX) \
-	jhat$(EXE_SUFFIX)
+	jhat$(EXE_SUFFIX) \
+	javaws$(EXE_SUFFIX)
 
 # The following get removed from the JRE after the bulk-copy of LIBDIR...
 NOTJRELIBS = tools.jar \
