diff -Nru openjdk.orig/jdk/make/common/internal/BinaryPlugs.gmk openjdk/jdk/make/common/internal/BinaryPlugs.gmk
--- openjdk.orig/jdk/make/common/internal/BinaryPlugs.gmk	2008-08-14 08:42:49.000000000 +0100
+++ openjdk/jdk/make/common/internal/BinaryPlugs.gmk	2008-08-20 22:56:55.000000000 +0100
@@ -147,6 +147,21 @@
 	@for i in $(PLUG_JMF_CLASS_NAMES) ; do \
 	  $(ECHO) "$$i" >> $@; \
 	done
+$(PLUG_TEMPDIR)/gnu.clist:
+	@$(prep-target)
+	@for i in $(PLUG_GNU_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
+$(PLUG_TEMPDIR)/javax.clist:
+	@$(prep-target)
+	@for i in $(PLUG_JAVAX_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
+$(PLUG_TEMPDIR)/netx.clist:
+	@$(prep-target)
+	@for i in $(PLUG_NETX_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
 $(PLUG_TEMPDIR)/sound.clist:
 	@$(prep-target)
 	@for i in $(PLUG_SOUND_CLASS_NAMES) ; do \
@@ -159,6 +174,12 @@
 	$(plug-create-jargs)
 $(PLUG_TEMPDIR)/sound.jargs: $(PLUG_TEMPDIR)/sound.clist
 	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/gnu.jargs: $(PLUG_TEMPDIR)/gnu.clist
+	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/javax.jargs: $(PLUG_TEMPDIR)/javax.clist
+	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/netx.jargs: $(PLUG_TEMPDIR)/netx.clist
+	$(plug-create-jargs)
 $(PLUG_TEMPDIR)/all.jargs: $(PLUG_TEMPDIR)/all.clist
 	$(plug-create-jargs)
 
@@ -183,9 +204,9 @@
 # Import classes command
 
 define import-binary-plug-classes
-@$(MKDIR) -p $(CLASSDESTDIR)
+$(MKDIR) -p $(CLASSDESTDIR)
 @$(CAT) $1 | $(SED) -e 's/^/PLUG IMPORT: /'
-($(CD) $(CLASSDESTDIR) && $(BOOT_JAR_CMD) xf $(PLUG_IMPORT_JARFILE) @$1 $(BOOT_JAR_JFLAGS) )
+($(CD) $(CLASSDESTDIR) && $(BOOT_JAR_CMD) xf $(PLUG_IMPORT_JARFILE) `$(CAT) $1`)
 ($(CD) $(CLASSDESTDIR) && $(java-vm-cleanup) )
 endef # import-binary-plug-classes
 
diff -Nru openjdk.orig/jdk/src/share/classes/java/beans/MetaData.java openjdk/jdk/src/share/classes/java/beans/MetaData.java
--- openjdk.orig/jdk/src/share/classes/java/beans/MetaData.java	2008-08-14 08:43:00.000000000 +0100
+++ openjdk/jdk/src/share/classes/java/beans/MetaData.java	2008-08-20 22:55:35.000000000 +0100
@@ -1463,7 +1463,7 @@
     }
 
     private static String[] getAnnotationValue(Constructor<?> constructor) {
-        ConstructorProperties annotation = constructor.getAnnotation(ConstructorProperties.class);
+        ConstructorProperties annotation = ((ConstructorProperties) constructor.getAnnotation(ConstructorProperties.class));
         return (annotation != null)
                 ? annotation.value()
                 : null;
