--- openjdk/jdk/make/common/shared/Platform.gmk.orig	2008-07-26 17:21:17.441295678 +0200
+++ openjdk/jdk/make/common/shared/Platform.gmk	2008-07-26 17:30:24.589549285 +0200
@@ -191,6 +191,9 @@
   OS_VERSION := $(shell uname -r)
   # Arch and OS name/version
   mach := $(shell uname -m)
+  ifneq (,$(filter $(mach),mips s390 s390x))
+    mach := $(shell dpkg-architecture -qDEB_BUILD_ARCH_CPU 2>/dev/null || echo $(mach))
+  endif
   archExpr = case "$(mach)" in \
                 i[3-9]86) \
                     echo i586 \
@@ -204,6 +207,9 @@
                 sparc*) \
                     echo sparc \
                     ;; \
+                arm*) \
+                    echo arm \
+                    ;; \
                 *) \
                     echo $(mach) \
                     ;; \
--- openjdk/corba/make/common/shared/Platform.gmk.orig	2008-07-26 17:21:17.441295678 +0200
+++ openjdk/corba/make/common/shared/Platform.gmk	2008-07-26 17:30:22.059511120 +0200
@@ -181,6 +181,9 @@
   OS_VERSION := $(shell uname -r)
   # Arch and OS name/version
   mach := $(shell uname -m)
+  ifneq (,$(filter $(mach),mips s390 s390x))
+    mach := $(shell dpkg-architecture -qDEB_BUILD_ARCH_CPU 2>/dev/null || echo $(mach))
+  endif
   archExpr = case "$(mach)" in \
                 i[3-9]86) \
                     echo i586 \
@@ -194,6 +197,9 @@
                 sparc*) \
                     echo sparc \
                     ;; \
+                arm*) \
+                    echo arm \
+                    ;; \
                 *) \
                     echo $(mach) \
                     ;; \
