--- origopenjdk/hotspot/src/os/linux/vm/os_linux.cpp	2008-01-04 18:11:53.000000000 -0500
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2008-01-11 11:15:24.000000000 -0500
@@ -268,7 +268,12 @@
  *        ...
  *        7: The default directories, normally /lib and /usr/lib.
  */
+
+#if defined(AMD64) || defined(_LP64) && (defined(PPC) || defined(S390))
+#define DEFAULT_LIBPATH "/lib:/usr/lib:/usr/lib64:/lib64"
+#else
 #define DEFAULT_LIBPATH "/lib:/usr/lib"
+#endif
 
 #define EXTENSIONS_DIR  "/lib/ext"
 #define ENDORSED_DIR    "/lib/endorsed"
