diff -ruN openjdk-ecj/jdk/make/tools/sharing/classlist.linux openjdk/jdk/make/tools/sharing/classlist.linux
--- openjdk-ecj/jdk/make/tools/sharing/classlist.linux	2007-08-30 03:19:44.000000000 -0400
+++ openjdk/jdk/make/tools/sharing/classlist.linux	2007-09-04 13:33:52.000000000 -0400
@@ -330,7 +330,6 @@
 java/beans/PropertyChangeEvent
 java/util/EventObject
 java/awt/Component$AWTTreeLock
-sun/awt/DebugHelper
 sun/awt/NativeLibLoader
 sun/security/action/LoadLibraryAction
 java/awt/GraphicsEnvironment
@@ -341,7 +340,6 @@
 java/lang/ProcessEnvironment$Value
 java/lang/ProcessEnvironment$StringEnvironment
 java/util/Collections$UnmodifiableMap
-sun/awt/DebugHelperStub
 java/awt/Toolkit
 java/awt/Toolkit$3
 sun/util/CoreResourceBundleControl
diff -ruN openjdk-ecj/jdk/make/tools/sharing/classlist.solaris openjdk/jdk/make/tools/sharing/classlist.solaris
--- openjdk-ecj/jdk/make/tools/sharing/classlist.solaris	2007-08-30 03:19:44.000000000 -0400
+++ openjdk/jdk/make/tools/sharing/classlist.solaris	2007-09-04 13:34:17.000000000 -0400
@@ -332,7 +332,6 @@
 java/beans/PropertyChangeEvent
 java/util/EventObject
 java/awt/Component$AWTTreeLock
-sun/awt/DebugHelper
 sun/awt/NativeLibLoader
 sun/security/action/LoadLibraryAction
 java/awt/GraphicsEnvironment
@@ -343,7 +342,6 @@
 java/lang/ProcessEnvironment$Value
 java/lang/ProcessEnvironment$StringEnvironment
 java/util/Collections$UnmodifiableMap
-sun/awt/DebugHelperStub
 java/awt/Toolkit
 java/awt/Toolkit$3
 sun/util/CoreResourceBundleControl
diff -ruN openjdk-ecj/jdk/make/tools/sharing/classlist.windows openjdk/jdk/make/tools/sharing/classlist.windows
--- openjdk-ecj/jdk/make/tools/sharing/classlist.windows	2007-08-30 03:19:44.000000000 -0400
+++ openjdk/jdk/make/tools/sharing/classlist.windows	2007-09-04 13:34:05.000000000 -0400
@@ -333,10 +333,8 @@
 java/beans/PropertyChangeEvent
 java/util/EventObject
 java/awt/Component$AWTTreeLock
-sun/awt/DebugHelper
 sun/awt/NativeLibLoader
 sun/security/action/LoadLibraryAction
-sun/awt/DebugHelperStub
 java/awt/Toolkit
 java/awt/Toolkit$3
 sun/util/CoreResourceBundleControl
--- openjdk.orig/jdk/make/java/awt/Makefile	2008-02-28 04:39:42.000000000 -0500
+++ openjdk/jdk/make/java/awt/Makefile	2008-02-29 14:58:50.000000000 -0500
@@ -34,7 +34,7 @@
 #
 AUTO_FILES_JAVA_DIRS = java/awt sun/awt/geom
 
-build: properties cursors
+build: sources properties cursors
 
 #
 # Resources
@@ -47,9 +47,11 @@
 #
 include $(BUILDDIR)/common/Classes.gmk
 
+sources:
 
 clean:: properties.clean sun.geom.clean cursors.clean
 
+
 _LIBPROPS = \
 	flavormap.properties
 
@@ -102,6 +104,6 @@
 cursors.clean :
 	$(RM) -r $(CURSORDIR)
 
-.PHONY: properties properties.clean \
+.PHONY: sources properties properties.clean \
 	cursors sun.geom.clean cursors.clean
 
