--- openjdk.old/hotspot/src/share/vm/interpreter/bytecodeInterpreterWithChecks.xsl      2007-10-12 03:46:08.000000000 -0400
+++ openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreterWithChecks.xsl      2007-10-29 11:57:33.000000000 -0400
@@ -8,6 +8,7 @@

 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">

+<xsl:output method="text" indent="no" omit-xml-declaration="yes"/>
 <xsl:template match="processcode">
 <xsl:text>
 #define VM_JVMTI
@@ -17,7 +18,6 @@

 </xsl:text>

-<xsl:output method="text" indent="no" omit-xml-declaration="yes"/>
 </xsl:template>

 </xsl:stylesheet>
