diff -u openjdk.orig/jdk/src/share/classes/sun/applet/PluginAppletViewer.java openjdk/jdk/src/share/classes/sun/applet/PluginAppletViewer.java
--- openjdk.orig/jdk/src/share/classes/sun/applet/PluginAppletViewer.java	2008-02-27 08:45:37.000000000 -0500
+++ openjdk/jdk/src/share/classes/sun/applet/PluginAppletViewer.java	2008-02-27 08:51:14.000000000 -0500
@@ -40,6 +40,8 @@
 import java.security.PrivilegedAction;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.lang.reflect.Constructor;
+import java.lang.ClassNotFoundException;
 import sun.awt.SunToolkit;
 import sun.awt.AppContext;
 import sun.awt.X11.*;
@@ -97,14 +99,33 @@
     /**
      * Create the applet viewer
      */
-    public PluginAppletViewer(long handle, int x, int y, URL doc,
-                              Hashtable atts, PrintStream statusMsgStream,
+    public PluginAppletViewer(long handle, int x, int y, final URL doc,
+                              final Hashtable atts, PrintStream statusMsgStream,
                               PluginAppletViewerFactory factory) {
         super(handle, true);
     	this.factory = factory;
 	this.statusMsgStream = statusMsgStream;
 
-	add("Center", panel = new AppletViewerPanel(doc, atts));
+	try
+	{
+		Class c = Thread.currentThread().getContextClassLoader().loadClass("net.sourceforge.jnlp.NetxPanel");
+		Class[] constructClasses = new Class[] {doc.getClass(), atts.getClass()};
+		Constructor constructor = c.getConstructor(constructClasses);
+		Object[] constructArgs = new Object[] {doc, atts};
+		panel = (AppletViewerPanel) constructor.newInstance(constructArgs);
+		AppletViewerPanel.debug("Using Netx panel");
+	}
+	catch (ClassNotFoundException ex)
+	{
+		AppletViewerPanel.debug("Netx Panel not found - using AppletViewerPanel", ex);
+		panel = new AppletViewerPanel(doc, atts);
+	}
+	catch (Exception ex)
+	{
+		AppletViewerPanel.debug("Unable to start NetX applet - defaulting to Sun applet", ex);
+		panel = new AppletViewerPanel(doc, atts);
+	}
+	add("Center", panel);
 	panel.init();
 	appletPanels.addElement(panel);
 
