diff -Nru openjdk.orig/hotspot/src/share/vm/asm/codeBuffer.hpp openjdk/hotspot/src/share/vm/asm/codeBuffer.hpp
--- openjdk.orig/hotspot/src/share/vm/asm/codeBuffer.hpp	2008-05-27 21:18:54.000000000 +0100
+++ openjdk/hotspot/src/share/vm/asm/codeBuffer.hpp	2008-05-27 21:56:42.000000000 +0100
@@ -81,7 +81,7 @@
   address     _locs_point;      // last relocated position (grows upward)
   bool        _locs_own;        // did I allocate the locs myself?
   bool        _frozen;          // no more expansion of this section
-  char        _index;           // my section number (SECT_INST, etc.)
+  signed char _index;           // my section number (SECT_INST, etc.)
   CodeBuffer* _outer;           // enclosing CodeBuffer
 
   // (Note:  _locs_point used to be called _last_reloc_offset.)
diff -Nru openjdk.orig/hotspot/src/share/vm/utilities/ostream.cpp openjdk/hotspot/src/share/vm/utilities/ostream.cpp
--- openjdk.orig/hotspot/src/share/vm/utilities/ostream.cpp	2008-05-27 21:18:54.000000000 +0100
+++ openjdk/hotspot/src/share/vm/utilities/ostream.cpp	2008-05-27 21:59:01.000000000 +0100
@@ -829,7 +829,7 @@
   server.sin_port = htons(port);
 
   server.sin_addr.s_addr = inet_addr(ip);
-  if (server.sin_addr.s_addr == (uint32_t)-1) {
+  if (server.sin_addr.s_addr == (in_addr_t)-1) {
 #ifdef _WINDOWS
     struct hostent* host = hpi::get_host_by_name((char*)ip);
 #else
diff -r ca98ea212429 openjdk/hotspot/src/share/vm/ci/ciTypeFlow.cpp
--- openjdk/hotspot/src/share/vm/ci/ciTypeFlow.cpp	Fri Aug 08 16:06:52 2008 +0100
+++ openjdk/hotspot/src/share/vm/ci/ciTypeFlow.cpp	Fri Aug 08 16:15:45 2008 +0100
@@ -388,7 +388,7 @@ const ciTypeFlow::StateVector* ciTypeFlo
   // Set the rest of the locals to bottom.
   Cell cell = state->next_cell(state->tos());
   state->set_stack_size(0);
-  int limit = state->limit_cell();
+  Cell limit = state->limit_cell();
   for (; cell < limit; cell = state->next_cell(cell)) {
     state->set_type_at(cell, state->bottom_type());
   }
