diff -urN openjdk.orig/j2se/make/common/Defs-linux.gmk openjdk/j2se/make/common/Defs-linux.gmk
--- openjdk.orig/j2se/make/common/Defs-linux.gmk	2007-05-24 03:33:20.000000000 -0400
+++ openjdk/j2se/make/common/Defs-linux.gmk	2007-06-18 16:02:48.000000000 -0400
@@ -106,6 +106,7 @@
     _OPT = $(CC_LOWER_OPT)
     CPPFLAGS_DBG    += -DLOGGING 
 endif
+_OPT = -O0 -g
 
 # For all platforms, do not omit the frame pointer register usage. 
 #    We need this frame pointer to make it easy to walk the stacks.
diff -urN openjdk.orig/j2se/make/Makefile openjdk/j2se/make/Makefile
--- openjdk.orig/j2se/make/Makefile	2007-05-24 03:33:16.000000000 -0400
+++ openjdk/j2se/make/Makefile	2007-06-18 16:03:40.000000000 -0400
@@ -245,9 +245,9 @@
 #
 # Docs
 #
-OTHERSUBDIRS = docs
-docs:: sanity-docs post-sanity-docs
-	$(OTHERSUBDIRS-loop)
+#OTHERSUBDIRS = docs
+#docs:: sanity-docs post-sanity-docs
+#	$(OTHERSUBDIRS-loop)
 
 #
 # Release engineering targets.
diff -urN openjdk.orig/hotspot/build/linux/makefiles/gcc.make openjdk/hotspot/build/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/build/linux/makefiles/gcc.make	2007-05-24 03:31:18.000000000 -0400
+++ openjdk/hotspot/build/linux/makefiles/gcc.make	2007-06-18 18:02:39.000000000 -0400
@@ -96,7 +94,7 @@
 CFLAGS_WARN/BYFILE = $(CFLAGS_WARN/$@)$(CFLAGS_WARN/DEFAULT$(CFLAGS_WARN/$@)) 
 
 # The flags to use for an Optimized g++ build
-OPT_CFLAGS += -O3
+OPT_CFLAGS += -O0 -g
 
 # Hotspot uses very unstrict aliasing turn this optimization off
 OPT_CFLAGS += -fno-strict-aliasing
