diff -Nru openjdk.orig/jdk/make/docs/Makefile openjdk/jdk/make/docs/Makefile
--- openjdk.orig/jdk/make/docs/Makefile	2009-04-14 15:55:52.000000000 +0100
+++ openjdk/jdk/make/docs/Makefile	2009-04-15 23:13:07.000000000 +0100
@@ -78,34 +78,19 @@
 
 TAGS = $(IGNORED_TAGS:%=-tag %:X) $(TAG_JLS3)
 
-ifeq ($(MILESTONE), fcs)
-    DOCTITLE_SWITCH = $(JAVADOCTITLE)
-    WINDOWTITLE_SWITCH = $(JAVADOCWINDOWTITLE)
-    HEADER_SWITCH = $(JAVADOCHEADER)
-    TOPOPTION=
-    JAVADOCBOTTOM_SWITCH= $(JAVADOCBOTTOM)
-    OVERVIEW_OPTION = -overview $(JAVADOCOVERVIEW)
-else
-    DOCTITLE_SWITCH = $(JAVADOCTITLE_EARLYACCESS)$(DRAFT)
-    WINDOWTITLE_SWITCH = $(JAVADOCWINDOWTITLE)" $(BUILD_NUMBER)"
-    HEADER_SWITCH = $(JAVADOCHEADER)$(DRAFT)
-    JAVADOCBOTTOM_SWITCH= $(JAVADOCBOTTOM_EARLYACCESS)
-    TOPOPTION= -top $(JAVADOCTOP_EARLYACCESS)
-    OVERVIEW_OPTION =
-endif
+DOCTITLE_SWITCH = $(JAVADOCTITLE)
+WINDOWTITLE_SWITCH = $(JAVADOCWINDOWTITLE)" $(BUILD_NUMBER)"
+HEADER_SWITCH = $(JAVADOCHEADER)
+TOPOPTION=
+JAVADOCBOTTOM_SWITCH= $(JAVADOCBOTTOM)
+OVERVIEW_OPTION = -overview $(JAVADOCOVERVIEW)
 
 JAVADOCTITLE  = 'Java$(TRADEMARK) Platform, Standard Edition $(JDK_MINOR_VERSION)<br>API Specification'
 JAVADOCWINDOWTITLE  = 'Java Platform SE $(JDK_MINOR_VERSION)'
 JAVADOCHEADER = '<strong>Java$(TRADEMARK)&nbsp;Platform<br>Standard&nbsp;Ed.&nbsp;$(JDK_MINOR_VERSION)</strong>'
-JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Submit a bug or feature</a><br>For further API reference and developer documentation, see <a href="{@docroot}/../../webnotes/devdocs-vs-specs.html">Java SE Developer Documentation</a>. That documentation contains more detailed, developer-targeted descriptions, with conceptual overviews, definitions of terms, workarounds, and working code examples. <p>Copyright $(THIS_YEAR) Sun Microsystems, Inc. All Rights Reserved. Use is subject to <a href="{@docroot}/../legal/license.html">license terms</a>. Also see the <a href="http://java.sun.com/docs/redist.html">documentation redistribution policy</a>.</font>'
+JAVADOCBOTTOM = '<font size="-1"><a href="http://icedtea.classpath.org/bugzilla/">Submit a bug or feature</a> <p>Copyright $(THIS_YEAR) Sun Microsystems, Inc. All rights reserved. Use is subject to the terms of the <a href="http://www.gnu.org/copyleft/gpl.html">GNU General Public License</a>.</font>'
 JAVADOCOVERVIEW = $(SHARE_SRC)/classes/overview-core.html
 
-#
-# Early access top and bottom text (for snapshots, beta and rc)
-#
-JAVADOCTOP_EARLYACCESS = '<div style="background-color: \#EEEEEE"> <div style="padding: 6px; margin-top: 2px; margin-bottom: 6px; margin-left: 6px; margin-right: 6px; text-align: justify; font-size: 80%; font-family: Helvetica, Arial, sans-serif; font-weight: normal;"> Please note that this documentation is not final and is subject to change. </div> </div>'
-JAVADOCBOTTOM_EARLYACCESS = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Submit a bug or feature</a> <p>Copyright $(THIS_YEAR) Sun Microsystems, Inc. All Rights Reserved. Use is subject to <a href="{@docroot}/../legal/license.html">license terms</a>.</font>'
-JAVADOCTITLE_EARLYACCESS = $(subst Specification,Documentation,$(JAVADOCTITLE))
 
 #
 # Variables used by domapidocs target
@@ -117,12 +102,11 @@
                     -doctitle $(DOMAPI_JAVADOCTITLE)          \
                     -windowtitle $(DOMAPI_JAVADOCWINDOWTITLE) \
                     -header $(DOMAPI_JAVADOCHEADER)           \
-                    -bottom $(DOMAPI_JAVADOCBOTTOM)           \
+                    -bottom $(JAVADOCBOTTOM)           \
                     -group $(DOMAPI_GROUPNAME) $(DOMAPI_REGEXP)
 DOMAPI_JAVADOCTITLE  = 'Common DOM API'
 DOMAPI_JAVADOCWINDOWTITLE  = 'Common DOM API'
 DOMAPI_JAVADOCHEADER = '<strong>Common DOM API</strong>'
-DOMAPI_JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Submit a bug or feature</a><br>Java is a trademark or registered trademark of Sun Microsystems, Inc. in the US and other countries.<br>Copyright $(THIS_YEAR) Sun Microsystems, Inc. 4150 Network Circle<br>Santa Clara, California, 95054, U.S.A.  All Rights Reserved.</font>'
 DOMAPI_GROUPNAME = "Packages"
 DOMAPI_REGEXP = "com.sun.java.browser.dom:org.w3c.dom*"
 # DOMAPI_PKGS is located in NON_CORE_PKGS.gmk
@@ -136,13 +120,12 @@
                     -doctitle $(MIRROR_JAVADOCTITLE)            \
                     -windowtitle $(MIRROR_JAVADOCWINDOWTITLE)   \
                     -header $(MIRROR_JAVADOCHEADER)             \
-                    -bottom $(MIRROR_JAVADOCBOTTOM)             \
+                    -bottom $(JAVADOCBOTTOM)             	\
                     -group $(MIRROR_GROUPNAME) $(MIRROR_REGEXP) \
                     -overview $(MIRROR_OVERVIEW)
 MIRROR_JAVADOCTITLE = 'Mirror API'
 MIRROR_JAVADOCWINDOWTITLE = 'Mirror API'
 MIRROR_JAVADOCHEADER = '<strong>Mirror API</strong>'
-MIRROR_JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Report a bug or request a feature.</a><br>Copyright $(THIS_YEAR) Sun Microsystems, Inc. All Rights Reserved. Use is subject to license terms.</font>'
 MIRROR_GROUPNAME = "Packages"
 MIRROR_OVERVIEW = $(IMPORTSRCDIR)/com/sun/mirror/overview.html
 MIRROR_REGEXP = "com.sun.mirror.*"
@@ -160,12 +143,11 @@
                     -doctitle $(DOCLETAPI_JAVADOCTITLE)          \
                     -windowtitle $(DOCLETAPI_JAVADOCWINDOWTITLE) \
                     -header $(DOCLETAPI_JAVADOCHEADER)           \
-                    -bottom $(DOCLETAPI_JAVADOCBOTTOM)           \
+                    -bottom $(JAVADOCBOTTOM)           \
                     -group $(DOCLETAPI_GROUPNAME) $(DOCLETAPI_REGEXP)
 DOCLETAPI_JAVADOCTITLE  = 'Doclet API'
 DOCLETAPI_JAVADOCWINDOWTITLE  = 'Doclet API'
 DOCLETAPI_JAVADOCHEADER = '<strong>Doclet API</strong>'
-DOCLETAPI_JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Submit a bug or feature</a><br>Java is a trademark or registered trademark of Sun Microsystems, Inc. in the US and other countries.<br>Copyright 1993-$(THIS_YEAR) Sun Microsystems, Inc. 4150 Network Circle<br>Santa Clara, California, 95054, U.S.A.  All Rights Reserved.</font>'
 DOCLETAPI_GROUPNAME = "Packages"
 DOCLETAPI_REGEXP = "com.sun.javadoc"
 DOCLETAPI_LINKOPT = -linkoffline ../../../../api $(DOCSDIR)/api/
@@ -179,8 +161,7 @@
 		    -encoding ascii			        \
                     -nonavbar                                   \
                     -noindex                                    \
-                    -bottom $(TAGLETAPI_JAVADOCBOTTOM)
-TAGLETAPI_JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Submit a bug or feature</a><br>Java is a trademark or registered trademark of Sun Microsystems, Inc. in the US and other countries.<br>Copyright 1993-$(THIS_YEAR) Sun Microsystems, Inc. 4150 Network Circle<br>Santa Clara, California, 95054, U.S.A.  All Rights Reserved.</font>'
+                    -bottom $(JAVADOCBOTTOM)
 # TAGLETAPI_FILE is located in NON_CORE_PKGS.gmk
 
 #
@@ -197,6 +178,7 @@
                    -windowtitle $(JDI_WINDOWTITLE)               \
                    -doctitle $(JDI_DOCTITLE)                     \
                    -header $(JDI_HEADER)                         \
+                   -bottom $(JAVADOCBOTTOM)             	\
                    -linkoffline ../../../../api $(DOCSDIR)/api/     \
                    -overview $(JPDA_SOURCEPATH)/jdi-overview.html
 JDI_WINDOWTITLE = "Java Debug Interface"
@@ -204,9 +186,6 @@
 JDI_HEADER      = "Java Debug Interface"
 # JDI_PKGS is located in NON_CORE_PKGS.gmk
 
-# Variables used by security components
-SECURITYAPI_JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Report a bug or request a feature.</a><br>Copyright $(THIS_YEAR) Sun Microsystems, Inc. All Rights Reserved. Use is subject to license terms.</font>'
-
 #
 # Variables used by JAAS target
 #
@@ -224,7 +203,7 @@
                    -windowtitle $(JAAS_WINDOWTITLE)              \
                    -doctitle $(JAAS_DOCTITLE)                    \
                    -header $(JAAS_JAVADOCHEADER)                 \
-                   -bottom $(SECURITYAPI_JAVADOCBOTTOM)                      \
+                   -bottom $(JAVADOCBOTTOM)                      \
                    -linkoffline ../../../../../api $(DOCSDIR)/api/	 \
                    -overview $(TOPDIR)/src/share/classes/com/sun/security/auth/jaas-overview.html
 JAAS_WINDOWTITLE = "Java Authentication and Authorization Service "
@@ -247,7 +226,7 @@
                    -windowtitle $(JGSS_WINDOWTITLE)             \
                    -doctitle $(JGSS_DOCTITLE)                   \
                    -header $(JGSS_JAVADOCHEADER)                \
-                   -bottom $(SECURITYAPI_JAVADOCBOTTOM)                      \
+                   -bottom $(JAVADOCBOTTOM)                      \
                    -linkoffline ../../../../../api $(DOCSDIR)/api/ \
                    -overview $(JGSS_SOURCEPATH)/com/sun/security/jgss/jgss-overview.html
 
@@ -271,7 +250,7 @@
                    -windowtitle $(SMARTCARDIO_WINDOWTITLE)             \
                    -doctitle $(SMARTCARDIO_DOCTITLE)                   \
                    -header $(SMARTCARDIO_JAVADOCHEADER)                \
-                   -bottom $(SECURITYAPI_JAVADOCBOTTOM)                      \
+                   -bottom $(JAVADOCBOTTOM)                      \
                    -linkoffline ../../../../../api $(DOCSDIR)/api/
 
 SMARTCARDIO_WINDOWTITLE = "Java Smart Card I/O"
@@ -316,6 +295,7 @@
                    -windowtitle $(HTTPSERVER_WINDOWTITLE)             \
                    -doctitle $(HTTPSERVER_DOCTITLE)                   \
                    -header $(HTTPSERVER_JAVADOCHEADER)                \
+                   -bottom $(JAVADOCBOTTOM)             	\
                    -linkoffline ../../../../../api $(DOCSDIR)/api/
 
 HTTPSERVER_WINDOWTITLE = "Java HTTP Server"
@@ -351,6 +331,7 @@
                      -windowtitle $(MGMT_WINDOWTITLE)              \
                      -doctitle $(MGMT_DOCTITLE)                    \
                      -header $(MGMT_HEADER)                        \
+                     -bottom $(JAVADOCBOTTOM)             	\
 		     -linkoffline ../../../../api $(DOCSDIR)/api/  \
                      -overview $(MGMT_SOURCEPATH)/com/sun/management/mgmt-overview.html
 MGMT_WINDOWTITLE = "Monitoring and Management Interface for the Java Platform"
@@ -372,6 +353,7 @@
                    -windowtitle $(ATTACH_WINDOWTITLE)               	\
                    -doctitle $(ATTACH_DOCTITLE)                     	\
                    -header $(ATTACH_HEADER)                         	\
+                   -bottom $(JAVADOCBOTTOM)             	\
 		   -linkoffline ../../../../api $(DOCSDIR)/api/
 ATTACH_WINDOWTITLE = "Attach API"
 ATTACH_DOCTITLE    = "Attach API"
@@ -392,6 +374,7 @@
                    -windowtitle $(JCONSOLE_WINDOWTITLE)               	\
                    -doctitle $(JCONSOLE_DOCTITLE)                     	\
                    -header $(JCONSOLE_HEADER)                         	\
+                   -bottom $(JAVADOCBOTTOM)             	\
 		   -linkoffline ../../../../api $(DOCSDIR)/api/
 JCONSOLE_WINDOWTITLE = "JConsole API"
 JCONSOLE_DOCTITLE    = "JConsole API"
@@ -407,7 +390,7 @@
                     -doctitle $(TREEAPI_JAVADOCTITLE)            \
                     -windowtitle $(TREEAPI_JAVADOCWINDOWTITLE)   \
                     -header $(TREEAPI_JAVADOCHEADER)             \
-                    -bottom $(TREEAPI_JAVADOCBOTTOM)             \
+                    -bottom $(JAVADOCBOTTOM)                     \
                     -group $(TREEAPI_GROUPNAME) $(TREEAPI_REGEXP) 
 #
 #	            -overview $(TREEAPI_OVERVIEW)
@@ -415,7 +398,6 @@
 TREEAPI_JAVADOCTITLE = 'Compiler Tree API'
 TREEAPI_JAVADOCWINDOWTITLE = 'Compiler Tree API'
 TREEAPI_JAVADOCHEADER = '<strong>Compiler Tree API</strong>'
-TREEAPI_JAVADOCBOTTOM = '<font size="-1"><a href="http://bugs.sun.com/services/bugreport/index.jsp">Report a bug or request a feature.</a><br>Copyright $(THIS_YEAR) Sun Microsystems, Inc. All Rights Reserved. Use is subject to license terms.</font>'
 TREEAPI_GROUPNAME = "Packages"
 TREEAPI_OVERVIEW = $(SHARE_SRC)/classes/com/sun/source/overview.html
 TREEAPI_REGEXP = "com.sun.source.*"
