--- openjdk-b20/openjdk/j2se/make/common/shared/Defs.gmk	2007-09-13 03:52:42.000000000 -0400
+++ openjdk/j2se/make/common/shared/Defs.gmk	2007-09-18 16:24:59.000000000 -0400
@@ -186,13 +186,13 @@
 
 # Default names
 LAUNCHER_NAME = java
-PRODUCT_NAME = Java(TM)
+PRODUCT_NAME = IcedTea
 PRODUCT_SUFFIX = SE Runtime Environment
 COMPANY_NAME = Sun Microsystems, Inc.
 
 ifdef OPENJDK
-  LAUNCHER_NAME = openjdk
-  PRODUCT_NAME = OpenJDK
+  LAUNCHER_NAME = java
+  PRODUCT_NAME = IcedTea
   PRODUCT_SUFFIX = Runtime Environment
   COMPANY_NAME =
 endif
@@ -250,12 +250,7 @@
 JDK_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_VERSION))
 JDK_MKTG_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_MKTG_VERSION))
 
-# RELEASE is JDK_VERSION and -MILESTONE if MILESTONE is set
-ifneq ($(MILESTONE),fcs)
-  RELEASE      = $(JDK_VERSION)-$(MILESTONE)$(BUILD_VARIANT_RELEASE)
-else
-  RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
-endif
+RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
 
 # FULL_VERSION is RELEASE and -BUILD_NUMBER if BUILD_NUMBER is set
 ifdef BUILD_NUMBER
--- ../openjdk-b21/openjdk/hotspot/build/hotspot_distro	2007-09-27 22:30:19.000000000 -0400
+++ openjdk/hotspot/build/hotspot_distro	2007-09-28 11:07:28.000000000 -0400
@@ -27,6 +27,6 @@
 #
 
 # Don't put quotes (fail windows build).
-HOTSPOT_VM_DISTRO=OpenJDK
+HOTSPOT_VM_DISTRO=IcedTea
 COMPANY_NAME=
-PRODUCT_NAME=OpenJDK
+PRODUCT_NAME=IcedTea
--- ../openjdk-b21/openjdk/hotspot/build/windows/projectfiles/common/Makefile	2007-09-27 22:30:19.000000000 -0400
+++ openjdk/hotspot/build/windows/projectfiles/common/Makefile	2007-09-28 11:07:50.000000000 -0400
@@ -125,7 +125,7 @@
 !if exists($(HOTSPOTWORKSPACE)\src\closed)
 HOTSPOT_VM_DISTRO="Java HotSpot(TM)"
 !else
-HOTSPOT_VM_DISTRO="OpenJDK"
+HOTSPOT_VM_DISTRO="IcedTea"
 !endif
 !endif
 
