diff -Nru openjdk.orig/jdk/make/common/Release.gmk openjdk/jdk/make/common/Release.gmk
--- openjdk.orig/jdk/make/common/Release.gmk	2010-08-27 19:16:10.000000000 +0100
+++ openjdk/jdk/make/common/Release.gmk	2010-08-31 18:26:34.688231634 +0100
@@ -345,8 +345,6 @@
 # Bug 5008685 - exclude jconsole from sun/tools
 #
 TOOLS = \
-	META-INF/services/com.sun.jdi.connect.Connector \
-	META-INF/services/com.sun.jdi.connect.spi.TransportService \
 	sun/tools/asm		\
 	sun/tools/jar		\
 	sun/tools/java		\
@@ -389,8 +387,6 @@
 	com/sun/tools/corba     \
 	com/sun/tools/internal/xjc       \
 	com/sun/tools/internal/ws       \
-	META-INF/services/com.sun.mirror.apt.AnnotationProcessorFactory \
-	META-INF/services/com.sun.tools.xjc.Plugin \
 	com/sun/istack/internal/tools       \
 	com/sun/istack/internal/ws       \
         com/sun/codemodel       \
@@ -401,7 +397,6 @@
 	com/sun/xml/internal/dtdparser \
 	com/sun/tools/jdi	\
 	com/sun/tools/script/shell	\
-	META-INF/services/com.sun.tools.attach.spi.AttachProvider \
 	com/sun/tools/attach	\
 	sun/tools/attach	\
 	sun/tools/jstack        \
@@ -565,10 +560,6 @@
 	$(ECHO) "com/sun/source/" >> $@
 	$(ECHO) "com/sun/istack/internal/tools/" >> $@
 	$(ECHO) "com/sun/istack/internal/ws/" >> $@
-	$(ECHO) "META-INF/services/com.sun.jdi.connect.Connector" >> $@
-	$(ECHO) "META-INF/services/com.sun.jdi.connect.spi.TransportService" >> $@
-	$(ECHO) "META-INF/services/com.sun.mirror.apt.AnnotationProcessorFactory" >> $@
-	$(ECHO) "META-INF/services/com.sun.tools.xjc.Plugin" >> $@
 	$(ECHO) "com/sun/tools/" >> $@
 	$(ECHO) "sun/jvmstat/" >> $@
 	$(ECHO) "sun/nio/cs/ext/" >> $@
@@ -611,7 +602,6 @@
 	$(ECHO) "javax/crypto/" >> $@
 	$(ECHO) "sun/security/internal/" >> $@
 	$(ECHO) "com/sun/crypto/provider/" >> $@
-	$(ECHO) "META-INF/services/com.sun.tools.attach.spi.AttachProvider" >> $@
 	$(ECHO) "com/sun/tools/attach/" >> $@
 	$(ECHO) "org/relaxng/datatype/" >> $@
 	$(ECHO) "com/sun/codemodel/" >> $@
@@ -645,6 +635,10 @@
 	    $(BOOT_JAVA_CMD) -jar $(JARREORDER_JARFILE) \
 		-o  $@.temp $(ABS_LIBDIR)/classlist $(NOT_RT_JAR_LIST) . )
 	$(MV) $@.temp $@
+	$(CAT) $@ | $(GREP) "^sun" | $(GREP) "class" | $(GREP) -v '\$$' \
+	  > $(ABS_TEMPDIR)/sun_classes_list
+	sed 's/\.class/\.java/' < $(ABS_TEMPDIR)/sun_classes_list \
+	  > $(ABS_TEMPDIR)/sun_sources_list
 	@($(CD) $(CLASSBINDIR) && $(java-vm-cleanup))
 
 # Create the rt.jar file list & non-class files list
