diff -urN openjdk.orig/jdk/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java openjdk/jdk/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java
--- openjdk.orig/jdk/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java	2007-10-12 04:01:55.000000000 -0400
+++ openjdk/jdk/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java	2007-10-12 17:17:28.000000000 -0400
@@ -1154,7 +1154,7 @@
             Set<BitSet> getterIndexSets = newSet();
             for (Constructor constr : annotatedConstrList) {
                 String[] propertyNames =
-                    constr.getAnnotation(propertyNamesClass).value();
+                    ((ConstructorProperties)constr.getAnnotation(propertyNamesClass)).value();
 
                 Type[] paramTypes = constr.getGenericParameterTypes();
                 if (paramTypes.length != propertyNames.length) {
diff -urN openjdk.orig/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java openjdk/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java
--- openjdk.orig/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java	2007-10-12 04:01:56.000000000 -0400
+++ openjdk/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java	2007-10-12 17:17:28.000000000 -0400
@@ -78,6 +78,9 @@
      */
     public long            timeStamp ;
 
+    // TODO: IcedTea: I am a stub.
+    static public int trapAuthenticationFailure = 0;
+
 
 
     /**
--- openjdk.orig/j2se/src/share/classes/java/beans/MetaData.java        2007-10-12 04:02:14.000000000 -0400
+++ openjdk/jdk/src/share/classes/java/beans/MetaData.java     2007-10-12 17:17:28.000000000 -0400
@@ -1564,7 +1564,7 @@
     }

     private static String[] getAnnotationValue(Constructor constructor) {
-        ConstructorProperties annotation = constructor.getAnnotation(ConstructorProperties.class);
+        ConstructorProperties annotation = ((ConstructorProperties) constructor.getAnnotation(ConstructorProperties.class));
         return (annotation != null)
                 ? annotation.value()
                 : null;

--- openjdk/jdk/make/common/internal/BinaryPlugs.gmk.orig	2008-02-21 00:02:59.000000000 -0500
+++ openjdk/jdk/make/common/internal/BinaryPlugs.gmk	2008-02-21 11:53:48.000000000 -0500
@@ -51,8 +51,6 @@
 com/sun/jmx/snmp/SnmpDataTypeEnums.class \
 com/sun/jmx/snmp/SnmpDefinitions.class \
 com/sun/jmx/snmp/SnmpOid.class \
-com/sun/jmx/snmp/SnmpOidDatabase.class \
-com/sun/jmx/snmp/SnmpOidDatabaseSupport.class \
 com/sun/jmx/snmp/SnmpOidRecord.class \
 com/sun/jmx/snmp/SnmpOidTable.class \
 com/sun/jmx/snmp/SnmpOidTableSupport.class \
@@ -62,17 +60,9 @@
 com/sun/jmx/snmp/SnmpTimeticks.class \
 com/sun/jmx/snmp/SnmpVarBind.class \
 com/sun/jmx/snmp/SnmpVarBindList.class \
-com/sun/jmx/snmp/Timestamp.class \
 com/sun/jmx/snmp/daemon/SendQ.class \
 com/sun/jmx/snmp/daemon/SnmpInformRequest.class \
-com/sun/jmx/snmp/daemon/SnmpQManager.class \
-com/sun/jmx/snmp/daemon/SnmpRequestCounter.class \
-com/sun/jmx/snmp/daemon/SnmpResponseHandler.class \
-com/sun/jmx/snmp/daemon/SnmpSendServer.class \
-com/sun/jmx/snmp/daemon/SnmpSession.class \
-com/sun/jmx/snmp/daemon/SnmpSocket.class \
-com/sun/jmx/snmp/daemon/SnmpTimerServer.class \
-com/sun/jmx/snmp/daemon/WaitQ.class
+com/sun/jmx/snmp/daemon/SnmpSession.class
 
 PLUG_SOUND_CLASS_NAMES = \
 com/sun/media/sound/AbstractPlayer.class \
@@ -130,9 +120,6 @@
 java/awt/color/CMMException.class \
 java/awt/color/ColorSpace.class \
 java/awt/color/ICC_ColorSpace.class \
-java/awt/color/ICC_Profile\$$1.class \
-java/awt/color/ICC_Profile\$$2.class \
-java/awt/color/ICC_Profile\$$3.class \
 java/awt/color/ICC_Profile.class \
 java/awt/color/ICC_ProfileGray.class \
 java/awt/color/ICC_ProfileRGB.class \
@@ -160,10 +147,16 @@
 java/awt/image/renderable/RenderableImageProducer.class \
 java/awt/image/renderable/RenderedImageFactory.class
 
+PLUG_GNU_CLASS_NAMES = gnu
+
+PLUG_JAVAX_CLASS_NAMES = javax/jnlp
+
+PLUG_NETX_CLASS_NAMES = net
+
 # Class list temp files (used by both import and export of plugs)
 
 PLUG_TEMPDIR=$(ABS_TEMPDIR)/plugs
-PLUG_CLASS_AREAS = jmf sound awt
+PLUG_CLASS_AREAS = jmf sound awt gnu javax netx
 PLUG_CLISTS = $(PLUG_CLASS_AREAS:%=$(PLUG_TEMPDIR)/%.clist)
 
 # Create jargs file command
@@ -191,6 +184,21 @@
 	@for i in $(PLUG_AWT_CLASS_NAMES) ; do \
 	  $(ECHO) "$$i" >> $@ ; \
 	done
+$(PLUG_TEMPDIR)/gnu.clist:
+	@$(prep-target)
+	@for i in $(PLUG_GNU_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
+$(PLUG_TEMPDIR)/javax.clist:
+	@$(prep-target)
+	@for i in $(PLUG_JAVAX_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
+$(PLUG_TEMPDIR)/netx.clist:
+	@$(prep-target)
+	@for i in $(PLUG_NETX_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
 $(PLUG_TEMPDIR)/all.clist: $(PLUG_CLISTS)
 	@$(prep-target)
 	$(CAT) $(PLUG_CLISTS) > $@
@@ -200,6 +208,12 @@
 	$(plug-create-jargs)
 $(PLUG_TEMPDIR)/awt.jargs: $(PLUG_TEMPDIR)/awt.clist
 	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/gnu.jargs: $(PLUG_TEMPDIR)/gnu.clist
+	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/javax.jargs: $(PLUG_TEMPDIR)/javax.clist
+	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/netx.jargs: $(PLUG_TEMPDIR)/netx.clist
+	$(plug-create-jargs)
 $(PLUG_TEMPDIR)/all.jargs: $(PLUG_TEMPDIR)/all.clist
 	$(plug-create-jargs)
 
@@ -224,9 +238,9 @@
 # Import classes command
 
 define import-binary-plug-classes
-@$(MKDIR) -p $(CLASSDESTDIR)
+$(MKDIR) -p $(CLASSDESTDIR)
 @$(CAT) $1 | $(SED) -e 's/^/PLUG IMPORT: /'
-($(CD) $(CLASSDESTDIR) && $(BOOT_JAR_CMD) xf $(PLUG_IMPORT_JARFILE) @$1)
+($(CD) $(CLASSDESTDIR) && $(BOOT_JAR_CMD) xf $(PLUG_IMPORT_JARFILE) `$(CAT) $1`)
 endef # import-binary-plug-classes
 
 # Import specific area classes (the classes are always created)
@@ -237,13 +251,22 @@
 	$(call import-binary-plug-classes,$(PLUG_TEMPDIR)/sound.clist)
 import-binary-plug-awt-classes: $(PLUG_IMPORT_JARFILE) $(PLUG_TEMPDIR)/awt.clist
 	$(call import-binary-plug-classes,$(PLUG_TEMPDIR)/awt.clist)
+import-binary-plug-gnu-classes: $(PLUG_IMPORT_JARFILE) $(PLUG_TEMPDIR)/gnu.clist
+	$(call import-binary-plug-classes,$(PLUG_TEMPDIR)/gnu.clist)
+import-binary-plug-javax-classes: $(PLUG_IMPORT_JARFILE) $(PLUG_TEMPDIR)/javax.clist
+	$(call import-binary-plug-classes,$(PLUG_TEMPDIR)/javax.clist)
+import-binary-plug-netx-classes: $(PLUG_IMPORT_JARFILE) $(PLUG_TEMPDIR)/netx.clist
+	$(call import-binary-plug-classes,$(PLUG_TEMPDIR)/netx.clist)
 
 # Import all classes from the jar file
 
 import-binary-plug-jar: \
 	     import-binary-plug-jmf-classes \
 	     import-binary-plug-sound-classes \
-	     import-binary-plug-awt-classes
+	     import-binary-plug-awt-classes \
+	     import-binary-plug-gnu-classes \
+	     import-binary-plug-javax-classes \
+	     import-binary-plug-netx-classes
 
 # Import native libraries
 
