diff -Nru openjdk.orig/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java openjdk/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java
--- openjdk.orig/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java	2008-09-25 08:44:19.000000000 +0100
+++ openjdk/jdk/src/solaris/classes/sun/awt/X11/XWindowPeer.java	2008-09-26 21:13:18.000000000 +0100
@@ -1118,6 +1118,8 @@
 
     boolean isOverrideRedirect() {
         return (XWM.getWMID() == XWM.OPENLOOK_WM ? true : false) ||
+            (XWM.getWMID() == XWM.METACITY_WM ? true : false) ||
+            target.getName().equals("###overrideRedirect###") ||
             ((XToolkit)Toolkit.getDefaultToolkit()).isOverrideRedirect((Window)target) ||
             XTrayIconPeer.isTrayIconStuffWindow((Window)target);
     }
