--- openjdk.orig/j2se/make/common/Release.gmk	2007-10-26 16:46:41.000000000 -0400
+++ openjdk/j2se/make/common/Release.gmk	2007-10-28 18:28:19.000000000 -0400
@@ -625,6 +625,10 @@
 	    $(BOOT_JAVA_CMD) -jar $(JARREORDER_JARFILE) \
 		-o  $@.temp $(ABS_LIBDIR)/classlist $(NOT_RT_JAR_LIST) . )
 	$(MV) $@.temp $@
+	$(CAT) $@ | $(GREP) "^sun" | $(GREP) "class" | $(GREP) -v '\$$' \
+	    > $(ABS_TEMPDIR)/sun_classes_list
+	sed 's/\.class/\.java/' < $(ABS_TEMPDIR)/sun_classes_list \
+	    > $(ABS_TEMPDIR)/sun_sources_list
 	@($(CD) $(CLASSBINDIR) && $(java-vm-cleanup))
 
 # Create the rt.jar file list & non-class files list
@@ -1001,6 +1005,8 @@
 	    ($(CD) $$d && $(TAR) cf - -T $(ABS_TEMPDIR)/src-files.list ) \
 	       | ($(CD) $(JDK_IMAGE_DIR)/src && $(TAR) xf -); \
 	  fi; \
+	  (($(CD) $$d && $(TAR) cf - -T $(ABS_TEMPDIR)/sun_sources_list ) \
+	    | ($(CD) $(JDK_IMAGE_DIR)/src && $(TAR) xf -)) > /dev/null 2>&1 ; \
 	done
   else
 	for d in $(SOURCE_DIRS); do \
