--- openjdk/jdk/src/share/native/sun/java2d/cmm/lcms/LCMS.c.orig	2007-11-17 11:07:47.000000000 -0500
+++ openjdk/jdk/src/share/native/sun/java2d/cmm/lcms/LCMS.c	2007-11-17 11:12:47.000000000 -0500
@@ -182,6 +182,8 @@
 
     sProf.pf = cmsOpenProfileFromMem((LPVOID)dataArray, (DWORD) dataSize);
 
+    (*env)->ReleaseByteArrayElements (env, data, dataArray, 0);
+
     if (sProf.pf == NULL) {
         JNU_ThrowIllegalArgumentException(env, "Invalid profile data");
     }
