diff -Nru openjdk.orig/jdk/make/javax/swing/plaf/Makefile openjdk/jdk/make/javax/swing/plaf/Makefile
--- openjdk.orig/jdk/make/javax/swing/plaf/Makefile	2009-05-21 16:40:04.000000000 +0100
+++ openjdk/jdk/make/javax/swing/plaf/Makefile	2009-05-21 16:40:19.000000000 +0100
@@ -34,9 +34,7 @@
 #
 include FILES.gmk
 AUTO_FILES_JAVA_DIRS = javax/swing/plaf sun/swing com/sun/java/swing/plaf
-ifndef DISABLE_NIMBUS
-  SUBDIRS = nimbus
-endif
+SUBDIRS = nimbus
 
 # Nimbus is handled in its own directory
 AUTO_JAVA_PRUNE = nimbus
diff -Nru openjdk.orig/jdk/make/javax/swing/plaf/nimbus/Makefile openjdk/jdk/make/javax/swing/plaf/nimbus/Makefile
--- openjdk.orig/jdk/make/javax/swing/plaf/nimbus/Makefile	2009-05-08 16:10:44.000000000 +0100
+++ openjdk/jdk/make/javax/swing/plaf/nimbus/Makefile	2009-05-21 17:12:39.000000000 +0100
@@ -44,8 +44,6 @@
 #
 # Rules
 #
-CLASSES_INIT = $(NIMBUS_GENSRC_DIR)
-
 include $(BUILDDIR)/common/Classes.gmk
 
 $(NIMBUS_GENSRC_DIR): $(NIMBUS_SKIN_FILE) $(NIMBUS_GENERATOR_JAR)
