diff -Nru openjdk.orig/jdk/make/common/internal/BinaryPlugs.gmk openjdk/jdk/make/common/internal/BinaryPlugs.gmk
--- openjdk.orig/jdk/make/common/internal/BinaryPlugs.gmk	2008-07-13 03:42:22.000000000 +0100
+++ openjdk/jdk/make/common/internal/BinaryPlugs.gmk	2008-07-13 03:41:56.000000000 +0100
@@ -27,16 +27,6 @@
 
 # Definitions for openjdk plugs (used by both import and export)
 
-# Names of native shared libraries
-
-PLUG_JSOUND_LIBRARY=$(LIB_PREFIX)jsoundhs.$(LIBRARY_SUFFIX)
-PLUG_LIBRARY_NAMES = \
-    $(PLUG_JSOUND_LIBRARY)
-
-# Sub-directory where native shared libraries are located (e.g. jre/bin or...)
-
-PLUG_LOCATION_SUBDIR=$(ARCH_VM_SUBDIR)
-
 # Explicit classfile lists
 
 # WARNING: These classlists will not work with pattern rules, only used in 
@@ -222,17 +212,6 @@
 	     import-binary-plug-jmf-classes \
 	     import-binary-plug-sound-classes
 
-# Import native libraries
-
-$(LIB_LOCATION)/$(PLUG_JSOUND_LIBRARY): \
-    $(PLUG_IMPORT_DIR)/$(PLUG_LOCATION_SUBDIR)/$(PLUG_JSOUND_LIBRARY)
-	$(import-binary-plug-file)
-
-# Rules only used by lower level makefiles
-
-import-binary-plug-jsound-library: \
-    $(LIB_LOCATION)/$(PLUG_JSOUND_LIBRARY)
-
 # Binary plug start/complete messages
 
 import-binary-plugs-started:
diff -Nru openjdk.orig/jdk/make/javax/sound/Makefile openjdk/jdk/make/javax/sound/Makefile
--- openjdk.orig/jdk/make/javax/sound/Makefile	2008-07-13 03:42:40.000000000 +0100
+++ openjdk/jdk/make/javax/sound/Makefile	2008-07-13 03:41:22.000000000 +0100
@@ -96,11 +96,6 @@
 
 
 #
-# add "closed" library
-#
-SUBDIRS += jsoundhs
-
-#
 # system dependent flags
 #
 ifeq ($(PLATFORM), windows)
diff -uNr openjdk-orig/jdk/src/share/classes/com/sun/media/sound/Platform.java openjdk/jdk/src/share/classes/com/sun/media/sound/Platform.java
--- openjdk-orig/jdk/src/share/classes/com/sun/media/sound/Platform.java	2008-11-07 10:38:15.000000000 -0500
+++ openjdk/jdk/src/share/classes/com/sun/media/sound/Platform.java	2008-11-07 10:43:21.000000000 -0500
@@ -42,7 +42,6 @@
 
     // native library we need to load
     private static final String libNameMain     = "jsound";
-    private static final String libNameMain2    = "jsoundhs";
 
     private static final String libNameALSA     = "jsoundalsa";
     private static final String libNameDSound   = "jsoundds";
@@ -160,7 +159,6 @@
         try {
             // load the main libraries
             JSSecurityManager.loadLibrary(libNameMain);
-            JSSecurityManager.loadLibrary(libNameMain2);
             // just for the heck of it...
             loadedLibs |= LIB_MAIN;
         } catch (SecurityException e) {

