--- openjdk.orig/jdk/src/share/classes/sun/java2d/SurfaceData.java	2007-12-10 15:47:39.000000000 -0500
+++ openjdk/jdk/src/share/classes/sun/java2d/SurfaceData.java	2007-12-20 13:44:53.000000000 -0500
@@ -509,7 +509,7 @@
                 // assert(sg2d.surfaceData == this);
             }
         } else if (sg2d.compositeState == sg2d.COMP_CUSTOM) {
-            if (sg2d.antialiasHint == SunHints.INTVAL_ANTIALIAS_ON) {
+            if (sg2d.textAntialiasHint == SunHints.INTVAL_TEXT_ANTIALIAS_ON) {
                 if (sg2d.clipState == sg2d.CLIP_SHAPE) {
                     sg2d.drawpipe = AAClipCompViaShape;
                     sg2d.fillpipe = AAClipCompViaShape;
@@ -531,7 +531,7 @@
                     sg2d.textpipe = compText;
                 }
             }
-        } else if (sg2d.antialiasHint == SunHints.INTVAL_ANTIALIAS_ON) {
+        } else if (sg2d.textAntialiasHint == SunHints.INTVAL_TEXT_ANTIALIAS_ON) {
             sg2d.alphafill = getMaskFill(sg2d);
             // assert(sg2d.surfaceData == this);
             if (sg2d.alphafill != null) {
