diff -r 29e7597d75c4 openjdk/jdk/make/common/shared/Platform.gmk
--- openjdk/jdk/make/common/shared/Platform.gmk	Fri Mar 28 14:19:36 2008 +0000
+++ openjdk/jdk/make/common/shared/Platform.gmk	Tue Apr 01 09:02:17 2008 +0100
@@ -205,6 +205,9 @@ ifeq ($(SYSTEM_UNAME), Linux)
                 sparc*) \
                     echo sparc \
                     ;; \
+                arm*) \
+                    echo arm \
+                    ;; \
                 *) \
                     echo $(mach) \
                     ;; \
diff -r 29e7597d75c4 openjdk/corba/make/common/shared/Platform.gmk
--- openjdk/corba/make/common/shared/Platform.gmk	Fri Mar 28 14:19:36 2008 +0000
+++ openjdk/corba/make/common/shared/Platform.gmk	Tue Apr 01 09:02:17 2008 +0100
@@ -194,6 +194,9 @@ ifeq ($(SYSTEM_UNAME), Linux)
                 sparc*) \
                     echo sparc \
                     ;; \
+                arm*) \
+                    echo arm \
+                    ;; \
                 *) \
                     echo $(mach) \
                     ;; \
