diff -Nru openjdk.orig/corba/make/common/shared/Defs-java.gmk openjdk/corba/make/common/shared/Defs-java.gmk
--- openjdk.orig/corba/make/common/shared/Defs-java.gmk	2008-10-20 18:06:13.000000000 +0100
+++ openjdk/corba/make/common/shared/Defs-java.gmk	2008-10-20 18:07:05.000000000 +0100
@@ -91,15 +91,8 @@
 #  -- Use JAVAC_CMD if you want to take the defaults given to you.
 #
 
-ifndef DEBUG_CLASSFILES
-  ifeq ($(VARIANT), DBG)
-    DEBUG_CLASSFILES = true
-  endif
-endif
-JAVACFLAGS =
-ifeq ($(DEBUG_CLASSFILES),true)
-  JAVACFLAGS += -g
-endif
+# Always include debug info
+JAVACFLAGS += -g
 ifeq ($(COMPILER_WARNINGS_FATAL), true)
   JAVACFLAGS  += -Werror
 endif
diff -Nru openjdk.orig/jaxp/make/build.xml openjdk/jaxp/make/build.xml
--- openjdk.orig/jaxp/make/build.xml	2008-10-20 18:41:39.000000000 +0100
+++ openjdk/jaxp/make/build.xml	2008-10-20 18:42:12.000000000 +0100
@@ -85,7 +85,8 @@
              destdir="${build.classes.dir}"
              memoryInitialSize="${javac.memoryInitialSize}"
              memoryMaximumSize="${javac.memoryMaximumSize}"
-             target="${javac.target}">
+             target="${javac.target}"
+	     debug="${javac.debug}">
          <compilerarg value="-J-Xbootclasspath/p:${bootstrap.dir}/lib/javac.jar"/>
          <compilerarg line="${javac.version.opt}"/>
       </javac>
diff -Nru openjdk.orig/jaxp/make/Makefile openjdk/jaxp/make/Makefile
--- openjdk.orig/jaxp/make/Makefile	2008-10-20 17:58:09.000000000 +0100
+++ openjdk/jaxp/make/Makefile	2008-10-20 17:59:19.000000000 +0100
@@ -61,13 +61,8 @@
   endif
 endif
 
-ifeq ($(VARIANT), DBG)
-  ANT_OPTIONS += -Djavac.debug=true
-else
-  ifeq ($(VARIANT), OPT)
-    ANT_OPTIONS += -Djavac.debug=false
-  endif
-endif
+# Always include debug info
+ANT_OPTIONS += -Djavac.debug=true
 
 # Note: jdk/make/common/Defs.gmk uses LANGUAGE_VERSION (-source NN)
 # and the somewhat misnamed CLASS_VERSION (-target NN)
diff -Nru openjdk.orig/jaxws/make/build.xml openjdk/jaxws/make/build.xml
--- openjdk.orig/jaxws/make/build.xml	2008-10-20 18:41:44.000000000 +0100
+++ openjdk/jaxws/make/build.xml	2008-10-20 18:42:46.000000000 +0100
@@ -99,6 +99,7 @@
              memoryInitialSize="${javac.memoryInitialSize}"
              memoryMaximumSize="${javac.memoryMaximumSize}"
              target="${javac.target}"
+	     debug="${javac.debug}"
              excludes="com/sun/tools/internal/txw2/**">
 	 <compilerarg line="-bootclasspath ${build.dir}/../../jaxp/build/classes:${build.classes.dir}:../../../generated:${plugs}/jre/lib/tools.jar:${plugs}/jre/lib/rt-closed.jar"/>
          <compilerarg value="-J-Xbootclasspath/p:${bootstrap.dir}/lib/javac.jar"/>
diff -Nru openjdk.orig/jaxws/make/Makefile openjdk/jaxws/make/Makefile
--- openjdk.orig/jaxws/make/Makefile	2008-10-20 17:58:18.000000000 +0100
+++ openjdk/jaxws/make/Makefile	2008-10-20 17:59:03.000000000 +0100
@@ -61,13 +61,8 @@
   endif
 endif
 
-ifeq ($(VARIANT), DBG)
-  ANT_OPTIONS += -Djavac.debug=true
-else
-  ifeq ($(VARIANT), OPT)
-    ANT_OPTIONS += -Djavac.debug=false
-  endif
-endif
+# Always include debug info
+ANT_OPTIONS += -Djavac.debug=true
 
 # Note: jdk/make/common/Defs.gmk uses LANGUAGE_VERSION (-source NN)
 # and the somewhat misnamed CLASS_VERSION (-target NN)
diff -Nru openjdk.orig/langtools/make/Makefile openjdk/langtools/make/Makefile
--- openjdk.orig/langtools/make/Makefile	2008-10-20 18:06:46.000000000 +0100
+++ openjdk/langtools/make/Makefile	2008-10-20 17:59:54.000000000 +0100
@@ -93,13 +93,8 @@
   endif
 endif
 
-ifeq ($(VARIANT), DBG)
-  ANT_OPTIONS += -Djavac.debug=true
-else
-  ifeq ($(VARIANT), OPT)
-    ANT_OPTIONS += -Djavac.debug=false
-  endif
-endif
+# Always include debug info
+ANT_OPTIONS += -Djavac.debug=true
 
 # Note: jdk/make/common/Defs.gmk uses LANGUAGE_VERSION (-source NN)
 # and the somewhat misnamed CLASS_VERSION (-target NN)
