diff -Nru openjdk.orig/jdk/make/common/shared/Compiler-gcc.gmk openjdk/jdk/make/common/shared/Compiler-gcc.gmk
--- openjdk.orig/jdk/make/common/shared/Compiler-gcc.gmk	2009-04-14 18:31:09.000000000 +0100
+++ openjdk/jdk/make/common/shared/Compiler-gcc.gmk	2009-04-14 20:32:47.000000000 +0100
@@ -66,6 +66,11 @@
   else
     CXX            = $(COMPILER_PATH)g++$(GCC_SUFFIX)
   endif
+  ifdef ZERO_BUILD
+    # zero
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
+  endif
   # Option used to create a shared library
   SHARED_LIBRARY_FLAG = -shared -mimpure-text
   SUN_COMP_VER := $(shell $(CC) --verbose 2>&1 )
