diff -Nru openjdk.orig/hotspot/src/share/vm/runtime/vm_version.cpp openjdk/hotspot/src/share/vm/runtime/vm_version.cpp
--- openjdk.orig/hotspot/src/share/vm/runtime/vm_version.cpp	2008-05-27 22:41:12.000000000 +0100
+++ openjdk/hotspot/src/share/vm/runtime/vm_version.cpp	2008-05-28 11:58:57.000000000 +0100
@@ -88,9 +88,6 @@
   #define VMLP ""
 #endif
 
-#ifdef KERNEL
-  #define VMTYPE "Kernel"
-#else // KERNEL
 #ifdef TIERED
   #define VMTYPE "Server"
 #else
@@ -101,7 +98,6 @@
   #define VMTYPE "Core"
 #endif // COMPILER1 || COMPILER2
 #endif // TIERED
-#endif // KERNEL
 
 #ifndef HOTSPOT_VM_DISTRO
   #error HOTSPOT_VM_DISTRO must be defined
