diff -Nru openjdk.orig/jdk/make/common/Defs.gmk openjdk/jdk/make/common/Defs.gmk
--- openjdk.orig/jdk/make/common/Defs.gmk	2008-05-23 20:51:33.000000000 +0100
+++ openjdk/jdk/make/common/Defs.gmk	2008-05-23 20:58:12.000000000 +0100
@@ -219,7 +219,7 @@
   ifeq ($(PLATFORM), windows)
     FREETYPE_LIB = $(FREETYPE_LIB_PATH)/freetype.lib
   else
-    FREETYPE_LIB = -L$(FREETYPE_LIB_PATH) -lfreetype
+    FREETYPE_LIB = -L$(FREETYPE_LIB_PATH) $(FT2_LIB)
   endif
 
   ifdef ALT_FREETYPE_HEADERS_PATH
diff -Nru openjdk.orig/jdk/make/sun/font/Makefile openjdk/jdk/make/sun/font/Makefile
--- openjdk.orig/jdk/make/sun/font/Makefile	2008-05-23 20:51:44.000000000 +0100
+++ openjdk/jdk/make/sun/font/Makefile	2008-05-23 20:56:56.000000000 +0100
@@ -160,11 +160,13 @@
 #
 CFLAGS       += -DHEADLESS
 CXXFLAGS     += -DHEADLESS
+FREETYPE_HEADERS=$(shell $(ECHO) \"$(FREETYPE2_HEADERS)\" | grep -o .I.*[a\-z])
 
 ifndef OPENJDK
   CPPFLAGS += -I$(CLOSED_SRC)/share/native/$(PKGDIR)/t2k
 else
-  CPPFLAGS += -I$(FREETYPE_HEADERS_PATH) -I$(FREETYPE_HEADERS_PATH)/freetype2
+  CPPFLAGS += $(FREETYPE_HEADERS) $(FREETYPE_HEADERS)/freetype2 \
+	-I$(FREETYPE_HEADERS_PATH) -I$(FREETYPE_HEADERS_PATH)/freetype2
   OTHER_LDLIBS += $(FREETYPE_LIB)
 endif
 
diff -Nru openjdk.orig/jdk/make/tools/freetypecheck/Makefile openjdk/jdk/make/tools/freetypecheck/Makefile
--- openjdk.orig/jdk/make/tools/freetypecheck/Makefile	2008-05-23 20:51:57.000000000 +0100
+++ openjdk/jdk/make/tools/freetypecheck/Makefile	2008-05-23 20:54:53.000000000 +0100
@@ -45,7 +45,7 @@
   ifeq ($(PLATFORM), solaris)
     FT_LD_OPTIONS += -R $(FREETYPE_LIB_PATH) -lfreetype
   else #linux
-      FT_LD_OPTIONS += -Wl,-rpath -Wl,$(FREETYPE_LIB_PATH) -lfreetype
+      FT_LD_OPTIONS += -Wl,-rpath -Wl,$(FREETYPE_LIB_PATH) $(FT2_LIB)
   endif
 endif
 FT_OPTIONS += -I$(FREETYPE_HEADERS_PATH)
