diff -Nru openjdk.orig/hotspot/make/linux/makefiles/vm.make openjdk/hotspot/make/linux/makefiles/vm.make
--- openjdk.orig/hotspot/make/linux/makefiles/vm.make	2008-10-09 08:42:26.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/vm.make	2008-10-24 10:08:32.000000000 +0100
@@ -86,6 +86,10 @@
   ${JRE_VERSION}     \
   ${VM_DISTRO}
 
+ifdef DISTRIBUTION_ID
+CPPFLAGS += -DDISTRIBUTION_ID="\"$(DISTRIBUTION_ID)\""
+endif
+
 # CFLAGS_WARN holds compiler options to suppress/enable warnings.
 CFLAGS += $(CFLAGS_WARN/BYFILE)
 
diff -Nru openjdk.orig/hotspot/src/share/vm/runtime/arguments.cpp openjdk/hotspot/src/share/vm/runtime/arguments.cpp
--- openjdk.orig/hotspot/src/share/vm/runtime/arguments.cpp	2008-10-09 08:42:33.000000000 +0100
+++ openjdk/hotspot/src/share/vm/runtime/arguments.cpp	2008-10-24 10:08:32.000000000 +0100
@@ -25,7 +25,7 @@
 #include "incls/_precompiled.incl"
 #include "incls/_arguments.cpp.incl"
 
-#define DEFAULT_VENDOR_URL_BUG "http://java.sun.com/webapps/bugreport/crash.jsp"
+#define DEFAULT_VENDOR_URL_BUG "http://icedtea.classpath.org/bugzilla"
 #define DEFAULT_JAVA_LAUNCHER  "generic"
 
 char**  Arguments::_jvm_flags_array             = NULL;
diff -Nru openjdk.orig/hotspot/src/share/vm/utilities/vmError.cpp openjdk/hotspot/src/share/vm/utilities/vmError.cpp
--- openjdk.orig/hotspot/src/share/vm/utilities/vmError.cpp	2008-10-09 08:42:34.000000000 +0100
+++ openjdk/hotspot/src/share/vm/utilities/vmError.cpp	2008-10-24 10:08:32.000000000 +0100
@@ -165,7 +165,8 @@
 
 static void print_bug_submit_message(outputStream *out, Thread *thread) {
   if (out == NULL) return;
-  out->print_raw_cr("# If you would like to submit a bug report, please visit:");
+  out->print_raw_cr("# If you would like to submit a bug report, please include");
+  out->print_raw_cr("# instructions how to reproduce the bug and visit:");
   out->print_raw   ("#   ");
   out->print_raw_cr(Arguments::java_vendor_url_bug());
   // If the crash is in native code, encourage user to submit a bug to the
@@ -339,6 +340,9 @@
                    Abstract_VM_Version::vm_platform_string(),
                    UseCompressedOops ? "compressed oops" : ""
                  );
+#ifdef DISTRIBUTION_ID
+     st->print_cr("# Distribution: %s", DISTRIBUTION_ID);
+#endif
 
   STEP(60, "(printing problematic frame)")
 
diff -Nru openjdk.orig/jdk/make/common/shared/Defs.gmk openjdk/jdk/make/common/shared/Defs.gmk
--- openjdk.orig/jdk/make/common/shared/Defs.gmk	2008-10-09 08:45:42.000000000 +0100
+++ openjdk/jdk/make/common/shared/Defs.gmk	2008-10-24 10:08:32.000000000 +0100
@@ -189,8 +189,8 @@
 
 # Default names
 ifdef OPENJDK
-  LAUNCHER_NAME = openjdk
-  PRODUCT_NAME = OpenJDK
+  LAUNCHER_NAME = java
+  PRODUCT_NAME = IcedTea
   PRODUCT_SUFFIX = Runtime Environment
   JDK_RC_PLATFORM_NAME = Platform
   COMPANY_NAME = N/A
@@ -255,12 +255,7 @@
 JDK_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_VERSION))
 JDK_MKTG_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_MKTG_VERSION))
 
-# RELEASE is JDK_VERSION and -MILESTONE if MILESTONE is set
-ifneq ($(MILESTONE),fcs)
-  RELEASE      = $(JDK_VERSION)-$(MILESTONE)$(BUILD_VARIANT_RELEASE)
-else
-  RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
-endif
+RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
 
 # FULL_VERSION is RELEASE and -BUILD_NUMBER if BUILD_NUMBER is set
 ifdef BUILD_NUMBER
