--- ../openjdk/control/make/j2se-rules.gmk	2007-10-12 03:45:09.000000000 -0400
+++ openjdk/control/make/j2se-rules.gmk	2007-11-07 17:05:36.000000000 -0500
@@ -36,37 +36,9 @@
   IMAGES_TARGET = images
 endif
 
-# No DOCS build when JDK_UPDATE_VERSION set
-ifdef JDK_UPDATE_VERSION
-  DOCS_TARGET =
-else
-  DOCS_TARGET = docs
-endif
-
-# NO_DOCS may be set in conjunction with DEV_ONLY
-ifdef NO_DOCS
-  DOCS_TARGET =
-endif
+DOCS_TARGET = docs
 
-# If langtools not being built, full jdk javadoc is not possible
-ifneq ($(BUILD_LANGTOOLS), true)
-  DOCS_TARGET =
-endif
-ifneq ($(BUILD_CORBA), true)
-  DOCS_TARGET =
-endif
-ifneq ($(BUILD_JAXP), true)
-  DOCS_TARGET =
-endif
-ifneq ($(BUILD_JAXWS), true)
-  DOCS_TARGET =
-endif
-
-ifndef DEV_ONLY
-  J2SE_BUILD_TARGETS = sanity all $(DOCS_TARGET) images
-else
-  J2SE_BUILD_TARGETS = sanity all $(IMAGES_TARGET)
-endif
+J2SE_BUILD_TARGETS = sanity all $(DOCS_TARGET) images
 
 J2SE_CLOBBER_TARGETS = clobber
 
