diff -Nru openjdk-ecj.orig/jdk/src/share/classes/javax/management/AttributeValueExp.java openjdk-ecj/jdk/src/share/classes/javax/management/AttributeValueExp.java
--- openjdk.orig/jdk/src/share/classes/javax/management/AttributeValueExp.java	2008-05-26 19:27:49.000000000 +0100
+++ openjdk/jdk/src/share/classes/javax/management/AttributeValueExp.java	2008-05-26 19:28:38.000000000 +0100
@@ -113,7 +113,6 @@
      * @exception BadBinaryOpValueExpException
      *
      */
-    @Override
     public ValueExp apply(ObjectName name) throws BadStringOperationException, BadBinaryOpValueExpException,
         BadAttributeValueExpException, InvalidApplicationException {
         Object result = getAttribute(name);
@@ -150,7 +149,6 @@
        evaluted an AttributeValueExp can only appear inside a QueryExp,
        and that QueryExp will itself have done setMBeanServer.  */
     @Deprecated
-    @Override
     public void setMBeanServer(MBeanServer s)  {
     }
 
diff -Nru openjdk.orig/jdk/src/share/classes/sun/java2d/SunGraphics2D.java openjdk/jdk/src/share/classes/sun/java2d/SunGraphics2D.java
--- openjdk.orig/jdk/src/share/classes/sun/java2d/SunGraphics2D.java	2008-08-20 23:50:02.000000000 +0100
+++ openjdk/jdk/src/share/classes/sun/java2d/SunGraphics2D.java	2008-08-20 23:50:17.000000000 +0100
@@ -3301,7 +3301,6 @@
      *
      * @see sun.java2d.DestSurfaceProvider#getDestSurface
      */
-    @Override
     public Surface getDestSurface() {
         return surfaceData;
     }
diff -Nru openjdk.orig/jdk/src/solaris/classes/sun/awt/X11/XComponentPeer.java openjdk/jdk/src/solaris/classes/sun/awt/X11/XComponentPeer.java
--- openjdk.orig/jdk/src/solaris/classes/sun/awt/X11/XComponentPeer.java	2008-08-14 08:43:27.000000000 +0100
+++ openjdk/jdk/src/solaris/classes/sun/awt/X11/XComponentPeer.java	2008-08-20 23:56:06.000000000 +0100
@@ -1292,7 +1292,6 @@
                                                            backBuffer);
     }
 
-    @Override
     public BufferCapabilities getBackBufferCaps() {
         return backBufferCaps;
     }
diff -Nru openjdk.orig/jdk/src/share/classes/sun/awt/image/SunVolatileImage.java openjdk/jdk/src/share/classes/sun/awt/image/SunVolatileImage.java
--- openjdk.orig/jdk/src/share/classes/sun/awt/image/SunVolatileImage.java	2008-08-14 08:43:10.000000000 +0100
+++ openjdk/jdk/src/share/classes/sun/awt/image/SunVolatileImage.java	2008-08-21 00:07:02.000000000 +0100
@@ -263,7 +263,6 @@
      *
      * @see sun.java2d.DestSurfaceProvider#getDestSurface
      */
-    @Override
     public Surface getDestSurface() {
         return volSurfaceManager.getPrimarySurfaceData();
     }
diff -Nru openjdk.orig/jdk/src/solaris/classes/sun/java2d/opengl/GLXGraphicsConfig.java openjdk/jdk/src/solaris/classes/sun/java2d/opengl/GLXGraphicsConfig.java
--- openjdk.orig/jdk/src/solaris/classes/sun/java2d/opengl/GLXGraphicsConfig.java	2008-08-14 08:43:27.000000000 +0100
+++ openjdk/jdk/src/solaris/classes/sun/java2d/opengl/GLXGraphicsConfig.java	2008-08-21 01:26:22.000000000 +0100
@@ -92,7 +92,6 @@
         return this;
     }
 
-    @Override
     public SurfaceData createManagedSurface(int w, int h, int transparency) {
         return GLXSurfaceData.createData(this, w, h,
                                          getColorModel(transparency),
@@ -163,12 +162,10 @@
      * Returns true if the provided capability bit is present for this config.
      * See OGLContext.java for a list of supported capabilities.
      */
-    @Override
     public final boolean isCapPresent(int cap) {
         return ((oglCaps.getCaps() & cap) != 0);
     }
 
-    @Override
     public final long getNativeConfigInfo() {
         return pConfigInfo;
     }
@@ -178,12 +175,10 @@
      *
      * @see sun.java2d.pipe.hw.BufferedContextProvider#getContext
      */
-    @Override
     public final OGLContext getContext() {
         return context;
     }
 
-    @Override
     public BufferedImage createCompatibleImage(int width, int height) {
         ColorModel model = new DirectColorModel(24, 0xff0000, 0xff00, 0xff);
         WritableRaster
@@ -392,7 +387,6 @@
      *
      * @see sun.java2d.pipe.hw.AccelGraphicsConfig#createCompatibleVolatileImage
      */
-    @Override
     public VolatileImage
         createCompatibleVolatileImage(int width, int height,
                                       int transparency, int type)
@@ -432,17 +426,14 @@
      *
      * @see sun.java2d.pipe.hw.AccelGraphicsConfig#getContextCapabilities
      */
-    @Override
     public ContextCapabilities getContextCapabilities() {
         return oglCaps;
     }
 
-    @Override
     public void addDeviceEventListener(AccelDeviceEventListener l) {
         AccelDeviceEventNotifier.addListener(l, screen.getScreen());
     }
 
-    @Override
     public void removeDeviceEventListener(AccelDeviceEventListener l) {
         AccelDeviceEventNotifier.removeListener(l);
     }
