--- old/src/share/vm/opto/parse3.cpp    Wed Apr  9 12:49:10 2008
+++ openjdk/hotspot/src/share/vm/opto/parse3.cpp        Wed Apr  9 12:49:08 2008
@@ -398,7 +398,7 @@
   // fill the lowest dimensions with unknown sizes
   for (int index = 0; index < mdimensions - ndimensions; index++) {
     const TypeAry* arr0 = TypeAry::make(prev_type, TypeInt::POS);
-    prev_type = TypeAryPtr::make(TypePtr::BotPTR, arr0, prev_array, true, 0);
+    prev_type = TypeAryPtr::make(TypePtr::BotPTR, arr0, prev_array, false, 0);
     prev_array = NULL; // array klasses can be lazy, except the first
   }

