diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/jmx/mbeanserver/MBeanIntrospector.java openjdk/jdk/src/share/classes/com/sun/jmx/mbeanserver/MBeanIntrospector.java
--- openjdk.orig/jdk/src/share/classes/com/sun/jmx/mbeanserver/MBeanIntrospector.java	2008-08-09 03:59:20.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/jmx/mbeanserver/MBeanIntrospector.java	2008-08-09 04:00:56.000000000 +0100
@@ -627,7 +627,7 @@
         MBeanConstructorInfo[] mbc = new MBeanConstructorInfo[cons.length];
         for (int i = 0; i < cons.length; i++) {
             String descr = "Public constructor of the MBean";
-            Description d = cons[i].getAnnotation(Description.class);
+            Description d = (Description) cons[i].getAnnotation(Description.class);
             if (d != null)
                 descr = d.value();
             mbc[i] = new MBeanConstructorInfo(descr, cons[i]);
