--- openjdk/hotspot/src/share/vm/prims/jvmtiManageCapabilities.cpp	2008-12-24 12:01:18.000000000 +0000
+++ openjdk/hotspot/src/share/vm/prims/jvmtiManageCapabilities.cpp	2008-12-24 12:00:29.000000000 +0000
@@ -84,7 +84,9 @@
 
   memset(&jc, 0, sizeof(jc));
   jc.can_get_bytecodes = 1;
+#ifndef CC_INTERP
   jc.can_signal_thread = 1;
+#endif // !CC_INTERP
   jc.can_get_source_file_name = 1;
   jc.can_get_line_numbers = 1;
   jc.can_get_synthetic_attribute = 1;
@@ -115,8 +117,10 @@
   jvmtiCapabilities jc;
 
   memset(&jc, 0, sizeof(jc));
+#ifndef CC_INTERP
   jc.can_pop_frame = 1;
   jc.can_force_early_return = 1;
+#endif // !CC_INTERP
   jc.can_get_source_debug_extension = 1;
   jc.can_access_local_variables = 1;
   jc.can_maintain_original_method_order = 1;
