diff -Nru openjdk.orig/hotspot/src/share/vm/runtime/arguments.cpp openjdk/hotspot/src/share/vm/runtime/arguments.cpp
--- openjdk.orig/hotspot/src/share/vm/runtime/arguments.cpp	2008-08-04 08:40:23.000000000 +0100
+++ openjdk/hotspot/src/share/vm/runtime/arguments.cpp	2008-08-09 01:10:55.000000000 +0100
@@ -25,7 +25,7 @@
 #include "incls/_precompiled.incl"
 #include "incls/_arguments.cpp.incl"
 
-#define DEFAULT_VENDOR_URL_BUG "http://java.sun.com/webapps/bugreport/crash.jsp"
+#define DEFAULT_VENDOR_URL_BUG "http://icedtea.classpath.org/bugzilla"
 #define DEFAULT_JAVA_LAUNCHER  "generic"
 
 char**  Arguments::_jvm_flags_array             = NULL;
diff -Nru openjdk.orig/jdk/make/common/shared/Defs.gmk openjdk/jdk/make/common/shared/Defs.gmk
--- openjdk.orig/jdk/make/common/shared/Defs.gmk	2008-08-04 08:43:01.000000000 +0100
+++ openjdk/jdk/make/common/shared/Defs.gmk	2008-08-09 01:13:25.000000000 +0100
@@ -189,8 +189,8 @@
 
 # Default names
 ifdef OPENJDK
-  LAUNCHER_NAME = openjdk
-  PRODUCT_NAME = OpenJDK
+  LAUNCHER_NAME = java
+  PRODUCT_NAME = IcedTea
   PRODUCT_SUFFIX = Runtime Environment
   JDK_RC_PLATFORM_NAME = Platform
   COMPANY_NAME = N/A
@@ -255,12 +255,7 @@
 JDK_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_VERSION))
 JDK_MKTG_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_MKTG_VERSION))
 
-# RELEASE is JDK_VERSION and -MILESTONE if MILESTONE is set
-ifneq ($(MILESTONE),fcs)
-  RELEASE      = $(JDK_VERSION)-$(MILESTONE)$(BUILD_VARIANT_RELEASE)
-else
-  RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
-endif
+RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
 
 # FULL_VERSION is RELEASE and -BUILD_NUMBER if BUILD_NUMBER is set
 ifdef BUILD_NUMBER
