diff -urN openjdk.orig/hotspot/src/cpu/x86/vm/c2_globals_x86.hpp openjdk/hotspot/src/cpu/x86/vm/c2_globals_x86.hpp
--- openjdk.orig/hotspot/src/cpu/x86/vm/c2_globals_x86.hpp	2007-10-12 03:45:58.000000000 -0400
+++ openjdk/hotspot/src/cpu/x86/vm/c2_globals_x86.hpp	2007-10-12 17:32:54.000000000 -0400
@@ -101,7 +101,7 @@
 
 // Heap related flags
 define_pd_global(uintx, PermSize,    ScaleForWordSize(16*M));
-define_pd_global(uintx, MaxPermSize, ScaleForWordSize(64*M));
+define_pd_global(uintx, MaxPermSize, ScaleForWordSize(128*M));
 
 // Ergonomics related flags
 define_pd_global(bool, NeverActAsServerClassMachine, false);
diff -urN openjdk.orig/hotspot/src/share/vm/runtime/globals.hpp openjdk/hotspot/src/share/vm/runtime/globals.hpp
--- openjdk.orig/hotspot/src/share/vm/runtime/globals.hpp	2007-10-12 03:46:16.000000000 -0400
+++ openjdk/hotspot/src/share/vm/runtime/globals.hpp	2007-10-12 17:32:54.000000000 -0400
@@ -2615,7 +2615,7 @@
           "an OS lock")                                                     \
                                                                             \
   /* gc parameters */                                                       \
-  product(uintx, MaxHeapSize, ScaleForWordSize(64*M),                       \
+  product(uintx, MaxHeapSize, ScaleForWordSize(512*M),                       \
           "Default maximum size for object heap (in bytes)")                \
                                                                             \
   product_pd(uintx, NewSize, 						    \
