<<<<<<< local
diff -Nru openjdk.orig/hotspot/src/share/vm/runtime/vm_version.cpp openjdk/hotspot/src/share/vm/runtime/vm_version.cpp
--- openjdk.orig/hotspot/src/share/vm/runtime/vm_version.cpp	2008-10-30 17:10:16.000000000 +0000
+++ openjdk/hotspot/src/share/vm/runtime/vm_version.cpp	2008-10-30 17:11:34.000000000 +0000
@@ -88,9 +88,6 @@
=======
--- openjdk/hotspot/src/share/vm/runtime/vm_version.cpp~	2008-12-05 16:02:43.000000000 +0000
+++ openjdk/hotspot/src/share/vm/runtime/vm_version.cpp	2008-12-05 16:13:04.000000000 +0000
@@ -86,16 +86,12 @@
>>>>>>> other
   #define VMLP ""
 #endif
 
-#ifdef KERNEL
-  #define VMTYPE "Kernel"
-#else // KERNEL
 #ifdef TIERED
   #define VMTYPE "Server"
 #else
<<<<<<< local
@@ -101,7 +98,6 @@
   #define VMTYPE "Core"
 #endif // COMPILER1 || COMPILER2
=======
   #define VMTYPE COMPILER1_PRESENT("Client")   \
                  COMPILER2_PRESENT("Server")   
>>>>>>> other
 #endif // TIERED
-#endif // KERNEL
 
 #ifndef HOTSPOT_VM_DISTRO
   #error HOTSPOT_VM_DISTRO must be defined
