diff -Nru openjdk.orig/jdk/make/common/internal/BinaryPlugs.gmk openjdk/jdk/make/common/internal/BinaryPlugs.gmk
--- openjdk.orig/jdk/make/common/internal/BinaryPlugs.gmk	2008-06-22 02:47:03.000000000 +0100
+++ openjdk/jdk/make/common/internal/BinaryPlugs.gmk	2008-06-22 02:51:22.000000000 +0100
@@ -147,6 +147,21 @@
 	@for i in $(PLUG_JMF_CLASS_NAMES) ; do \
 	  $(ECHO) "$$i" >> $@; \
 	done
+$(PLUG_TEMPDIR)/gnu.clist:
+	@$(prep-target)
+	@for i in $(PLUG_GNU_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
+$(PLUG_TEMPDIR)/javax.clist:
+	@$(prep-target)
+	@for i in $(PLUG_JAVAX_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
+$(PLUG_TEMPDIR)/netx.clist:
+	@$(prep-target)
+	@for i in $(PLUG_NETX_CLASS_NAMES) ; do \
+	  $(ECHO) "$$i" >> $@ ; \
+	done
 $(PLUG_TEMPDIR)/sound.clist:
 	@$(prep-target)
 	@for i in $(PLUG_SOUND_CLASS_NAMES) ; do \
@@ -159,6 +174,12 @@
 	$(plug-create-jargs)
 $(PLUG_TEMPDIR)/sound.jargs: $(PLUG_TEMPDIR)/sound.clist
 	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/gnu.jargs: $(PLUG_TEMPDIR)/gnu.clist
+	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/javax.jargs: $(PLUG_TEMPDIR)/javax.clist
+	$(plug-create-jargs)
+$(PLUG_TEMPDIR)/netx.jargs: $(PLUG_TEMPDIR)/netx.clist
+	$(plug-create-jargs)
 $(PLUG_TEMPDIR)/all.jargs: $(PLUG_TEMPDIR)/all.clist
 	$(plug-create-jargs)
 
@@ -183,9 +204,9 @@
 # Import classes command
 
 define import-binary-plug-classes
-@$(MKDIR) -p $(CLASSDESTDIR)
+$(MKDIR) -p $(CLASSDESTDIR)
 @$(CAT) $1 | $(SED) -e 's/^/PLUG IMPORT: /'
-($(CD) $(CLASSDESTDIR) && $(BOOT_JAR_CMD) xf $(PLUG_IMPORT_JARFILE) @$1)
+($(CD) $(CLASSDESTDIR) && $(BOOT_JAR_CMD) xf $(PLUG_IMPORT_JARFILE) `$(CAT) $1`)
 endef # import-binary-plug-classes
 
 # Import specific area classes (the classes are always created)
diff -Nru openjdk.orig/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java openjdk/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java
--- openjdk.orig/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java	2008-05-27 22:30:15.000000000 +0100
+++ openjdk/jdk/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java	2008-06-22 02:48:42.000000000 +0100
@@ -78,6 +78,9 @@
      */
     public long            timeStamp ;
 
+    // TODO: IcedTea: I am a stub.
+    static public int trapAuthenticationFailure = 0;
+
 
 
     /**
diff -Nru openjdk.orig/jdk/src/share/classes/java/beans/MetaData.java openjdk/jdk/src/share/classes/java/beans/MetaData.java
--- openjdk.orig/jdk/src/share/classes/java/beans/MetaData.java	2008-05-27 22:30:16.000000000 +0100
+++ openjdk/jdk/src/share/classes/java/beans/MetaData.java	2008-06-22 02:48:42.000000000 +0100
@@ -1564,7 +1564,7 @@
     }
 
     private static String[] getAnnotationValue(Constructor<?> constructor) {
-        ConstructorProperties annotation = constructor.getAnnotation(ConstructorProperties.class);
+        ConstructorProperties annotation = ((ConstructorProperties) constructor.getAnnotation(ConstructorProperties.class));
         return (annotation != null)
                 ? annotation.value()
                 : null;
