diff -Nru openjdk.orig/hotspot/make/linux/makefiles/gcc.make openjdk/hotspot/make/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/make/linux/makefiles/gcc.make	2008-08-20 22:58:22.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2008-08-20 23:01:45.000000000 +0100
@@ -62,6 +62,10 @@
 CFLAGS += -D_REENTRANT
 CFLAGS += -fcheck-new
 
+# Always generate full debuginfo on Linux.  It'll be in a separate
+# debuginfo package when building RPMs.
+CFLAGS += -g
+
 ARCHFLAG = $(ARCHFLAG/$(BUILDARCH))
 ARCHFLAG/i486    = -m32 -march=i586
 ARCHFLAG/amd64   = -m64
diff -Nru openjdk.orig/jdk/make/common/Defs-linux.gmk openjdk/jdk/make/common/Defs-linux.gmk
--- openjdk.orig/jdk/make/common/Defs-linux.gmk	2008-08-14 08:42:49.000000000 +0100
+++ openjdk/jdk/make/common/Defs-linux.gmk	2008-08-20 23:01:45.000000000 +0100
@@ -146,13 +146,11 @@
  CFLAGS_COMMON += -pipe
 endif
 
-# Linux 64bit machines use Dwarf2, which can be HUGE, have fastdebug use -g1
 DEBUG_FLAG = -g
-ifeq ($(FASTDEBUG), true)
-  ifeq ($(ARCH_DATA_MODEL), 64)
-    DEBUG_FLAG = -g1
-  endif
-endif
+
+# Always generate full debuginfo on Linux.  It'll be in a separate
+# debuginfo package when building RPMs.
+CFLAGS_REQUIRED += -g
 
 CFLAGS_OPT      = $(CC_OPT)
 CFLAGS_DBG      = $(DEBUG_FLAG)
@@ -230,7 +228,7 @@
 ifeq ($(VARIANT), OPT)
   ifneq ($(NO_STRIP), true)
     # Debug 'strip -g' leaves local function Elf symbols (better stack traces)
-    POST_STRIP_PROCESS = $(STRIP) -g
+    # POST_STRIP_PROCESS = $(STRIP) -g
   endif
 endif
 
diff -Nru openjdk.orig/jdk/make/common/shared/Defs-java.gmk openjdk/jdk/make/common/shared/Defs-java.gmk
--- openjdk.orig/jdk/make/common/shared/Defs-java.gmk	2008-08-14 08:42:49.000000000 +0100
+++ openjdk/jdk/make/common/shared/Defs-java.gmk	2008-08-20 23:01:45.000000000 +0100
@@ -98,15 +98,9 @@
 #  -- Use JAVAC_CMD if you want to take the defaults given to you.
 #
 
-ifndef DEBUG_CLASSFILES
-  ifeq ($(VARIANT), DBG)
-    DEBUG_CLASSFILES = true
-  endif
-endif
-JAVACFLAGS =
-ifeq ($(DEBUG_CLASSFILES),true)
-  JAVACFLAGS += -g
-endif
+DEBUG_CLASSFILES = true
+JAVACFLAGS += -g
+
 ifeq ($(JAVAC_MAX_WARNINGS), true)
   JAVACFLAGS  += -Xlint:all
 endif
diff -Nru openjdk.orig/jdk/make/sun/awt/mawt.gmk openjdk/jdk/make/sun/awt/mawt.gmk
--- openjdk.orig/jdk/make/sun/awt/mawt.gmk	2008-08-14 08:42:50.000000000 +0100
+++ openjdk/jdk/make/sun/awt/mawt.gmk	2008-08-20 23:01:45.000000000 +0100
@@ -129,7 +129,7 @@
 #
 
 
-#CFLAGS += -g
+CFLAGS += -g
 ifeq ($(HEADLESS),true)
 CFLAGS += -DHEADLESS=$(HEADLESS)
 CPPFLAGS += -DHEADLESS=$(HEADLESS)
