diff -urN openjdk.orig/hotspot/make/linux/makefiles/gcc.make openjdk/hotspot/make/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/make/linux/makefiles/gcc.make	2007-10-12 18:11:59.000000000 -0400
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2007-10-12 18:12:22.000000000 -0400
@@ -63,6 +63,10 @@
 CFLAGS += -D_REENTRANT
 CFLAGS += -fcheck-new
 
+# Always generate full debuginfo on Linux.  It'll be in a separate
+# debuginfo package when building RPMs.
+CFLAGS += -g
+
 ARCHFLAG = $(ARCHFLAG/$(BUILDARCH))
 ARCHFLAG/i486    = -m32 -march=i586
 ARCHFLAG/amd64   = -m64
diff -urN openjdk.orig/jdk/make/common/Defs-linux.gmk openjdk/jdk/make/common/Defs-linux.gmk
--- openjdk.orig/jdk/make/common/Defs-linux.gmk	2007-10-12 03:54:05.000000000 -0400
+++ openjdk/jdk/make/common/Defs-linux.gmk	2007-10-12 18:12:22.000000000 -0400
@@ -155,13 +155,11 @@
    CFLAGS_COMMON   += $(GLOBAL_KPIC) $(GCC_WARNINGS)
 endif
 
-# Linux 64bit machines use Dwarf2, which can be HUGE, have fastdebug use -g1
 DEBUG_FLAG = -g
-ifeq ($(FASTDEBUG), true)
-  ifeq ($(ARCH_DATA_MODEL), 64)
-    DEBUG_FLAG = -g1
-  endif
-endif
+
+# Always generate full debuginfo on Linux.  It'll be in a separate
+# debuginfo package when building RPMs.
+CFLAGS_REQUIRED += -g
 
 CFLAGS_OPT      = $(POPT)
 CFLAGS_DBG      = $(DEBUG_FLAG)
@@ -236,7 +234,7 @@
 ifeq ($(VARIANT), OPT)
   ifneq ($(NO_STRIP), true)
     # Debug 'strip -g' leaves local function Elf symbols (better stack traces)
-    POST_STRIP_PROCESS = $(STRIP) -g
+    # POST_STRIP_PROCESS = $(STRIP) -g
   endif
 endif
 
diff -urN openjdk.orig/jdk/make/common/shared/Defs-java.gmk openjdk/jdk/make/common/shared/Defs-java.gmk
--- openjdk.orig/jdk/make/common/shared/Defs-java.gmk	2007-10-12 03:54:05.000000000 -0400
+++ openjdk/jdk/make/common/shared/Defs-java.gmk	2007-10-12 18:12:22.000000000 -0400
@@ -98,15 +98,9 @@
 #  -- Use JAVAC_CMD if you want to take the defaults given to you.
 #
 
-ifndef DEBUG_CLASSFILES
-  ifeq ($(VARIANT), DBG)
-    DEBUG_CLASSFILES = true
-  endif
-endif
-JAVACFLAGS =
-ifeq ($(DEBUG_CLASSFILES),true)
-  JAVACFLAGS += -g
-endif
+DEBUG_CLASSFILES = true
+JAVACFLAGS += -g
+
 ifeq ($(COMPILER_WARNINGS_FATAL), true)
   JAVACFLAGS  += -Werror
 endif
diff -urN openjdk.orig/jdk/make/sun/awt/mawt.gmk openjdk/jdk/make/sun/awt/mawt.gmk
--- openjdk.orig/jdk/make/sun/awt/mawt.gmk	2007-10-12 03:54:07.000000000 -0400
+++ openjdk/jdk/make/sun/awt/mawt.gmk	2007-10-12 18:12:22.000000000 -0400
@@ -129,7 +129,7 @@
 #
 
 
-#CFLAGS += -g
+CFLAGS += -g
 ifeq ($(HEADLESS),true)
 CFLAGS += -DHEADLESS=$(HEADLESS)
 CPPFLAGS += -DHEADLESS=$(HEADLESS)
