diff -Nru openjdk.orig/jdk/make/common/shared/Defs.gmk openjdk/jdk/make/common/shared/Defs.gmk
--- openjdk.orig/jdk/make/common/shared/Defs.gmk	2008-10-15 15:36:07.000000000 +0100
+++ openjdk/jdk/make/common/shared/Defs.gmk	2008-10-15 15:36:53.000000000 +0100
@@ -194,8 +194,8 @@
 
 # Default names
 ifdef OPENJDK
-  LAUNCHER_NAME = openjdk
-  PRODUCT_NAME = OpenJDK
+  LAUNCHER_NAME = java
+  PRODUCT_NAME = IcedTea6
   PRODUCT_SUFFIX = Runtime Environment
   JDK_RC_PLATFORM_NAME = Platform
   COMPANY_NAME = N/A
@@ -260,12 +260,7 @@
 JDK_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_VERSION))
 JDK_MKTG_UNDERSCORE_VERSION =  $(subst .,_,$(JDK_MKTG_VERSION))
 
-# RELEASE is JDK_VERSION and -MILESTONE if MILESTONE is set
-ifneq ($(MILESTONE),fcs)
-  RELEASE      = $(JDK_VERSION)-$(MILESTONE)$(BUILD_VARIANT_RELEASE)
-else
-  RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
-endif
+RELEASE      = $(JDK_VERSION)$(BUILD_VARIANT_RELEASE)
 
 # FULL_VERSION is RELEASE and -BUILD_NUMBER if BUILD_NUMBER is set
 ifdef BUILD_NUMBER
--- openjdk/hotspot/src/share/vm/utilities/vmError.cpp~	2008-08-28 10:23:18.000000000 +0200
+++ openjdk/hotspot/src/share/vm/utilities/vmError.cpp	2008-10-21 14:05:12.000000000 +0200
@@ -168,7 +168,8 @@
 
 static void print_bug_submit_message(outputStream *out, Thread *thread) {
   if (out == NULL) return;
-  out->print_raw_cr("# If you would like to submit a bug report, please visit:");
+  out->print_raw_cr("# If you would like to submit a bug report, please include");
+  out->print_raw_cr("# instructions how to reproduce the bug and visit:");
   out->print_raw   ("#   ");
   out->print_raw_cr(Arguments::java_vendor_url_bug());
   // If the crash is in native code, encourage user to submit a bug to the
@@ -340,6 +341,9 @@
                    Abstract_VM_Version::vm_info_string(),
                    Abstract_VM_Version::vm_platform_string()
                  );
+#ifdef DISTRIBUTION_ID
+     st->print_cr("# Distribution: %s", DISTRIBUTION_ID);
+#endif
 
   STEP(60, "(printing problematic frame)")
 
--- openjdk/hotspot/build/linux/makefiles/vm.make~	2008-08-28 10:22:55.000000000 +0200
+++ openjdk/hotspot/build/linux/makefiles/vm.make	2008-10-21 14:03:36.000000000 +0200
@@ -111,6 +111,10 @@
   ${JRE_VERSION}     \
   ${VM_DISTRO}
 
+ifdef DISTRIBUTION_ID
+CPPFLAGS += -DDISTRIBUTION_ID="\"$(DISTRIBUTION_ID)\""
+endif
+
 # CFLAGS_WARN holds compiler options to suppress/enable warnings.
 CFLAGS += $(CFLAGS_WARN/BYFILE)
 
