diff -ru openjdk.orig/jdk/make/common/Defs-linux.gmk openjdk/jdk/make/common/Defs-linux.gmk
--- openjdk.orig/jdk/make/common/Defs-linux.gmk	2007-11-08 11:35:35.000000000 +0000
+++ openjdk/jdk/make/common/Defs-linux.gmk	2007-11-08 11:36:15.000000000 +0000
@@ -109,6 +109,10 @@
 LDFLAGS_COMMON_sparcv9  += -m64 -mcpu=v9
 CFLAGS_REQUIRED_sparc   += -m32 -mcpu=v9
 LDFLAGS_COMMON_sparc    += -m32 -mcpu=v9
+CFLAGS_REQUIRED_ppc     += -m32
+LDFLAGS_COMMON_ppc      += -m32
+CFLAGS_REQUIRED_ppc64   += -m64
+LDFLAGS_COMMON_ppc64    += -m64
 CFLAGS_REQUIRED         =  $(CFLAGS_REQUIRED_$(ARCH))
 LDFLAGS_COMMON          += $(LDFLAGS_COMMON_$(ARCH))
 
diff -r 37a05a11f281 make/common/shared/Compiler-gcc.gmk
--- a/jdk/make/common/shared/Compiler-gcc.gmk       Sat Dec 01 00:00:00 2007 +0000
+++ openjdk/jdk/make/common/shared/Compiler-gcc.gmk       Wed Jan 23 14:02:23 2008 +0100
@@ -70,28 +70,51 @@ ifeq ($(PLATFORM), linux)
   else
     CXX            = $(COMPILER_PATH)g++$(GCC_SUFFIX)
   endif
-  ifneq ("$(findstring sparc,$(ARCH))", "")
-    # sparc or sparcv9
-    REQUIRED_CC_VER = 4.0
-    REQUIRED_GCC_VER = 4.0.*
-  else
-  ifeq ($(ARCH_DATA_MODEL), 32)
+  ifeq ($(ARCH), alpha)
+    # alpha
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
+  endif
+  ifeq ($(ARCH), amd64)
+    # amd64
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
+  endif
+  ifeq ($(ARCH), arm)
+    # arm
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
+  endif
+  ifeq ($(ARCH), i586)
     # i586
     REQUIRED_CC_VER = 3.2
     REQUIRED_GCC_VER = 3.2.1*
     REQUIRED_GCC_VER_INT = 3.2.1-7a
-  else
-  ifeq ($(ARCH), amd64)
-    # amd64
-    REQUIRED_CC_VER = 3.2
-    REQUIRED_GCC_VER = 3.2.*
   endif
   ifeq ($(ARCH), ia64)
     # ia64
     REQUIRED_CC_VER = 3.2
     REQUIRED_GCC_VER = 2.9[56789].*
   endif
+  ifeq ($(ARCH), mips)
+    # mips
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
   endif
+  ifneq ("$(findstring ppc,$(ARCH))", "")
+    # ppc or ppc64
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
+  endif
+  ifneq ("$(findstring s390,$(ARCH))", "")
+    # s390 or s390x
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
+  endif
+  ifneq ("$(findstring sparc,$(ARCH))", "")
+    # sparc or sparcv9
+    REQUIRED_CC_VER = 4.0
+    REQUIRED_GCC_VER = 4.0.*
   endif
   # Option used to create a shared library
   SHARED_LIBRARY_FLAG = -shared -mimpure-text
diff -ru openjdk.orig/jdk/make/common/shared/Platform.gmk openjdk/jdk/make/common/shared/Platform.gmk
--- openjdk.orig/jdk/make/common/shared/Platform.gmk	2007-11-08 11:34:54.000000000 +0000
+++ openjdk/jdk/make/common/shared/Platform.gmk	2007-11-08 11:36:15.000000000 +0000
@@ -225,9 +225,9 @@
       ARCH=sparcv9
     endif
   else
-    # i586 is 32-bit, amd64 is 64-bit
+    # i586 and ppc are 32-bit, amd64 and ppc64 are 64-bit
     ifndef ARCH_DATA_MODEL
-      ifeq ($(ARCH), i586)
+      ifneq ($(filter i586 ppc,$(ARCH)),)
         ARCH_DATA_MODEL=32
       else
         ARCH_DATA_MODEL=64
diff -r 37a05a11f281 make/javax/sound/SoundDefs.gmk
--- a/jdk/make/javax/sound/SoundDefs.gmk    Sat Dec 01 00:00:00 2007 +0000
+++ openjdk/jdk/make/javax/sound/SoundDefs.gmk    Wed Jan 23 14:39:17 2008 +0100
@@ -55,9 +55,45 @@ ifeq ($(PLATFORM), solaris)
 
 endif # PLATFORM solaris
 
+ifeq ($(ARCH), alpha)
+  CPPFLAGS += -DX_ARCH=X_ALPHA
+endif # ARCH alpha
+
+ifeq ($(ARCH), amd64)
+  CPPFLAGS += -DX_ARCH=X_AMD64
+endif # ARCH amd64
+
+ifeq ($(ARCH), arm)
+  CPPFLAGS += -DX_ARCH=X_ARM
+endif # ARCH arm
+
 ifeq ($(ARCH), i586)
   CPPFLAGS += -DX_ARCH=X_I586
 endif # ARCH i586
+
+ifeq ($(ARCH), ia64)
+  CPPFLAGS += -DX_ARCH=X_IA64
+endif # ARCH ia64
+
+ifeq ($(ARCH), mips)
+  CPPFLAGS += -DX_ARCH=X_MIPS
+endif # ARCH mips
+
+ifeq ($(ARCH), ppc)
+  CPPFLAGS += -DX_ARCH=X_PPC
+endif # ARCH ppc
+
+ifeq ($(ARCH), ppc64)
+  CPPFLAGS += -DX_ARCH=X_PPC64
+endif # ARCH ppc64
+
+ifeq ($(ARCH), s390)
+  CPPFLAGS += -DX_ARCH=X_S390
+endif # ARCH s390
+
+ifeq ($(ARCH), s390x)
+  CPPFLAGS += -DX_ARCH=X_S390X
+endif # ARCH s390x
 
 ifeq ($(ARCH), sparc)
   CPPFLAGS += -DX_ARCH=X_SPARC
@@ -66,10 +102,6 @@ ifeq ($(ARCH), sparcv9)
 ifeq ($(ARCH), sparcv9)
   CPPFLAGS += -DX_ARCH=X_SPARCV9
 endif # ARCH sparcv9
-
-ifeq ($(ARCH), amd64)
-  CPPFLAGS += -DX_ARCH=X_AMD64
-endif # ARCH amd64
 
 
 # files needed for MIDI i/o
diff -r 37a05a11f281 src/share/native/com/sun/media/sound/SoundDefs.h
--- a/jdk/src/share/native/com/sun/media/sound/SoundDefs.h  Sat Dec 01 00:00:00 2007 +0000
+++ openjdk/jdk/src/share/native/com/sun/media/sound/SoundDefs.h  Wed Jan 23 14:39:17 2008 +0100
@@ -33,11 +33,18 @@
 #define X_LINUX         3
 
 // types for X_ARCH
-#define X_I586          1
-#define X_SPARC         2
-#define X_SPARCV9       3
-#define X_IA64          4
-#define X_AMD64         5
+#define X_ALPHA         1
+#define X_AMD64         2
+#define X_ARM           3
+#define X_I586          4
+#define X_IA64          5
+#define X_MIPS          6
+#define X_PPC           7
+#define X_PPC64         8
+#define X_S390          9
+#define X_S390X         10
+#define X_SPARC         11
+#define X_SPARCV9       12
 
 // **********************************
 // Make sure you set X_PLATFORM and X_ARCH defines correctly.
diff -ru openjdk.orig/corba/make/common/Defs-linux.gmk openjdk-ecj/corba/make/common/Defs-linux.gmk
--- openjdk.orig/corba/make/common/Defs-linux.gmk	2007-11-09 14:12:27.000000000 +0000
+++ openjdk/corba/make/common/Defs-linux.gmk	2007-11-09 14:13:45.000000000 +0000
@@ -99,6 +99,10 @@
 LDFLAGS_COMMON_sparcv9  += -m64 -mcpu=v9
 CFLAGS_REQUIRED_sparc   += -m32 -mcpu=v9
 LDFLAGS_COMMON_sparc    += -m32 -mcpu=v9
+CFLAGS_REQUIRED_ppc     += -m32
+LDFLAGS_COMMON_ppc      += -m32
+CFLAGS_REQUIRED_ppc64   += -m64
+LDFLAGS_COMMON_ppc64    += -m64
 CFLAGS_REQUIRED         =  $(CFLAGS_REQUIRED_$(ARCH))
 LDFLAGS_COMMON          += $(LDFLAGS_COMMON_$(ARCH))
 
diff -ru openjdk.orig/corba/make/common/shared/Compiler-gcc.gmk openjdk-ecj/corba/make/common/shared/Compiler-gcc.gmk
--- openjdk.orig/corba/make/common/shared/Compiler-gcc.gmk	2007-11-09 14:12:27.000000000 +0000
+++ openjdk/corba/make/common/shared/Compiler-gcc.gmk	2007-11-09 14:13:45.000000000 +0000
@@ -74,13 +74,13 @@
     # sparc or sparcv9
     REQUIRED_CC_VER = 4.0
     REQUIRED_GCC_VER = 4.0.*
-  else
-  ifeq ($(ARCH_DATA_MODEL), 32)
+  endif
+  ifeq ($(ARCH), i586)
     # i586
     REQUIRED_CC_VER = 3.2
     REQUIRED_GCC_VER = 3.2.1*
     REQUIRED_GCC_VER_INT = 3.2.1-7a
-  else
+  endif
   ifeq ($(ARCH), amd64)
     # amd64
     REQUIRED_CC_VER = 3.2
@@ -91,7 +91,10 @@
     REQUIRED_CC_VER = 3.2
     REQUIRED_GCC_VER = 2.9[56789].*
   endif
-  endif
+  ifneq ("$(findstring ppc,$(ARCH))", "")
+    # ppc or ppc64
+    REQUIRED_CC_VER = 3.2
+    REQUIRED_GCC_VER = 3.2.*
   endif
   # Option used to create a shared library
   SHARED_LIBRARY_FLAG = -shared -mimpure-text
diff -ru openjdk.orig/corba/make/common/shared/Platform.gmk openjdk-ecj/corba/make/common/shared/Platform.gmk
--- openjdk.orig/corba/make/common/shared/Platform.gmk	2007-11-09 14:12:27.000000000 +0000
+++ openjdk/corba/make/common/shared/Platform.gmk	2007-11-09 14:13:45.000000000 +0000
@@ -214,9 +214,9 @@
       ARCH=sparcv9
     endif
   else
-    # i586 is 32-bit, amd64 is 64-bit
+    # i586 and ppc are 32-bit, amd64 and ppc64 are 64-bit
     ifndef ARCH_DATA_MODEL
-      ifeq ($(ARCH), i586)
+      ifneq ($(filter i586 ppc,$(ARCH)),)
         ARCH_DATA_MODEL=32
       else
         ARCH_DATA_MODEL=64
