diff -Nru openjdk.orig/hotspot/src/share/vm/interpreter/bytecodeInterpreterWithChecks.xsl openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreterWithChecks.xsl
--- openjdk.orig/hotspot/src/share/vm/interpreter/bytecodeInterpreterWithChecks.xsl	2008-05-26 23:57:49.000000000 +0100
+++ openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreterWithChecks.xsl	2008-05-26 23:58:26.000000000 +0100
@@ -5,6 +5,7 @@
 -->
 
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
+<xsl:output method="text" indent="no" omit-xml-declaration="yes"/>
 
 <xsl:template match="processcode">
 <xsl:text>
@@ -15,7 +16,6 @@
 
 </xsl:text>
 
-<xsl:output method="text" indent="no" omit-xml-declaration="yes"/>
 </xsl:template>
 
 </xsl:stylesheet>
