diff -Nru openjdk.orig/jaxp/make/Makefile openjdk/jaxp/make/Makefile
--- openjdk.orig/jaxp/make/Makefile	2008-09-25 08:40:52.000000000 +0100
+++ openjdk/jaxp/make/Makefile	2008-09-26 20:58:44.000000000 +0100
@@ -99,16 +99,9 @@
   ANT_OPTIONS += -Dfindbugs.home=$(FINDBUGS_HOME)
 endif
 
-ifdef ANT_HOME
-  ANT = $(ANT_HOME)/bin/ant
-  ifneq ($(shell test -x $(ANT); echo $$?), 0)
-    $(error "$(ANT) not found; please update ANT_HOME")
-  endif
-else
-  ANT = ant
-  ifneq ($(shell which $(ANT) > /dev/null; echo $$?), 0)
-    $(error "\"ant\" not found; please set ANT_HOME or put \"ant\" on your PATH")
-  endif
+ANT = ant
+ifneq ($(shell which $(ANT) > /dev/null; echo $$?), 0)
+  $(error "\"ant\" not found; please set ANT_HOME or put \"ant\" on your PATH")
 endif
 
 # Default target and expected 'do everything' target
diff -Nru openjdk.orig/jaxws/make/build.xml openjdk/jaxws/make/build.xml
--- openjdk.orig/jaxws/make/build.xml	2008-09-25 08:41:11.000000000 +0100
+++ openjdk/jaxws/make/build.xml	2008-09-26 20:58:44.000000000 +0100
@@ -100,6 +100,7 @@
              memoryMaximumSize="${javac.memoryMaximumSize}"
              target="${javac.target}"
              excludes="com/sun/tools/internal/txw2/**">
+	 <compilerarg line="-bootclasspath ${build.dir}/../../jaxp/build/classes:${build.classes.dir}:../../../generated:${plugs}/jre/lib/tools.jar:${plugs}/jre/lib/rt-closed.jar"/>
          <compilerarg value="-J-Xbootclasspath/p:${bootstrap.dir}/lib/javac.jar"/>
          <compilerarg line="${javac.version.opt}"/>
          <compilerarg line="${javac.no.jdk.warnings}"/>
diff -Nru openjdk.orig/jaxws/make/Makefile openjdk/jaxws/make/Makefile
--- openjdk.orig/jaxws/make/Makefile	2008-09-25 08:41:11.000000000 +0100
+++ openjdk/jaxws/make/Makefile	2008-09-26 20:58:44.000000000 +0100
@@ -80,6 +80,7 @@
 endif
 
 ifdef ALT_BOOTDIR
+  ANT_OPTIONS += -Dplugs=$(ALT_BINARY_PLUGS_PATH)
   ANT_JAVA_HOME = JAVA_HOME=$(ALT_BOOTDIR)
 endif
 
@@ -99,16 +100,9 @@
   ANT_OPTIONS += -Dfindbugs.home=$(FINDBUGS_HOME)
 endif
 
-ifdef ANT_HOME
-  ANT = $(ANT_HOME)/bin/ant
-  ifneq ($(shell test -x $(ANT); echo $$?), 0)
-    $(error "$(ANT) not found; please update ANT_HOME")
-  endif
-else
-  ANT = ant
-  ifneq ($(shell which $(ANT) > /dev/null; echo $$?), 0)
-    $(error "\"ant\" not found; please set ANT_HOME or put \"ant\" on your PATH")
-  endif
+ANT = ant
+ifneq ($(shell which $(ANT) > /dev/null; echo $$?), 0)
+  $(error "\"ant\" not found; please set ANT_HOME or put \"ant\" on your PATH")
 endif
 
 # Default target and expected 'do everything' target
diff -Nru openjdk.orig/langtools/make/Makefile openjdk/langtools/make/Makefile
--- openjdk.orig/langtools/make/Makefile	2008-09-25 08:45:02.000000000 +0100
+++ openjdk/langtools/make/Makefile	2008-09-26 20:58:44.000000000 +0100
@@ -133,16 +133,9 @@
   ANT_OPTIONS += -Dfindbugs.home=$(FINDBUGS_HOME)
 endif
 
-ifdef ANT_HOME
-  ANT = $(ANT_HOME)/bin/ant
-  ifneq ($(shell test -x $(ANT) && echo OK), OK)
-    $(error $(ANT) not found -- please update ANT_HOME)
-  endif
-else
-  ANT = ant
-  ifneq ($(shell test -x "`which $(ANT)`" && echo OK), OK)
-    $(error 'ant' not found -- please set ANT_HOME or put 'ant' on your PATH)
-  endif
+ANT = ant
+ifneq ($(shell which $(ANT) > /dev/null; echo $$?), 0)
+  $(error "\"ant\" not found; please put \"ant\" on your PATH")
 endif
 
 # Default target and expected 'do everything' target
