--- lang.orig	2007-10-02 11:57:53.000000000 -0400
+++ openjdk/langtools/make/Makefile	2007-10-02 11:58:08.000000000 -0400
@@ -129,16 +129,9 @@
   ANT_OPTIONS += -Dfindbugs.home=$(FINDBUGS_HOME)
 endif
 
-ifdef ANT_HOME
-  ANT = $(ANT_HOME)/bin/ant
-  ifneq ($(shell test -x $(ANT) && echo OK), OK)
-    $(error $(ANT) not found -- please update ANT_HOME)
-  endif
-else
-  ANT = ant
-  ifneq ($(shell test -x "`which $(ANT)`" && echo OK), OK)
-    $(error 'ant' not found -- please set ANT_HOME or put 'ant' on your PATH)
-  endif
+ANT = ant
+ifneq ($(shell test -x "`which $(ANT)`" && echo OK), OK)
+  $(error 'ant' not found -- please put 'ant' on your PATH)
 endif
 
 # Default target and expected 'do everything' target
