diff -Nru openjdk.orig/jdk/make/java/redist/Makefile openjdk/jdk/make/java/redist/Makefile
--- openjdk.orig/jdk/make/java/redist/Makefile	2009-10-26 21:42:20.000000000 +0000
+++ openjdk/jdk/make/java/redist/Makefile	2009-10-29 11:53:21.000000000 +0000
@@ -96,11 +96,13 @@
 
 # Hotspot client is only available on 32-bit non-Zero builds
 ifneq ($(ZERO_BUILD), true)
+ifndef ICEDTEA_SHARK_BUILD
 ifeq ($(ARCH_DATA_MODEL), 32)
   IMPORT_LIST += $(LIB_LOCATION)/$(CLIENT_LOCATION)/$(JVM_NAME) \
                  $(LIB_LOCATION)/$(CLIENT_LOCATION)/Xusage.txt
 endif
 endif
+endif
 
 ifeq ($(PLATFORM), windows)
 #  Windows     vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv  Windows
@@ -174,6 +176,7 @@
 endif 
 
 ifneq ($(ZERO_BUILD), true)
+ifndef ICEDTEA_SHARK_BUILD
 ifeq ($(ARCH_DATA_MODEL), 32)
 
 IMPORT_LIST += $(LIB_LOCATION)/$(CLIENT_LOCATION)/$(LIBJSIG_NAME)
@@ -204,6 +207,8 @@
 
 endif # 32bit
 
+endif # ICEDTEA_SHARK_BUILD
+
 endif # ZERO_BUILD
 
 #  NOT Windows ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ NOT Windows
diff -Nru openjdk.orig/make/hotspot-rules.gmk openjdk/make/hotspot-rules.gmk
--- openjdk.orig/make/hotspot-rules.gmk	2009-10-29 11:51:56.000000000 +0000
+++ openjdk/make/hotspot-rules.gmk	2009-10-29 11:52:56.000000000 +0000
@@ -74,6 +74,10 @@
 
 ifeq ($(ZERO_BUILD), true)
   HOTSPOT_TARGET := $(HOTSPOT_TARGET)zero
+else
+ifdef ICEDTEA_SHARK_BUILD                                                                                                
+  HOTSPOT_TARGET := $(HOTSPOT_TARGET)shark                                                                               
+endif      
 endif
 
 HOTSPOT_BUILD_ARGUMENTS += $(COMMON_BUILD_ARGUMENTS)
