diff -Nru openjdk.orig/jdk/make/com/sun/java/pack/Makefile openjdk/jdk/make/com/sun/java/pack/Makefile
--- openjdk.orig/jdk/make/com/sun/java/pack/Makefile	2011-06-13 15:19:20.000000000 +0100
+++ openjdk/jdk/make/com/sun/java/pack/Makefile	2011-06-13 15:21:25.281602536 +0100
@@ -70,12 +70,12 @@
 	     $(ZIPOBJDIR)/inftrees.$(OBJECT_SUFFIX) \
 	     $(ZIPOBJDIR)/inffast.$(OBJECT_SUFFIX)
 
-  OTHER_LDLIBS += -lz
 else
   OTHER_CXXFLAGS += -DNO_ZLIB -DUNPACK_JNI
-  OTHER_LDLIBS += -lz $(JVMLIB)
+  OTHER_LDLIBS += $(JVMLIB)
 endif
 
+OTHER_LDLIBS += -lz
 CXXFLAGS_DBG += -DFULL
 CXXFLAGS_OPT += -DPRODUCT
 CXXFLAGS_COMMON += -DFULL
@@ -92,12 +92,11 @@
   RES = $(OBJDIR)/$(PGRM).res
 else
   LDOUTPUT = -o #Have a space 
-  LDDFLAGS += -lz -lc
-  OTHER_LDLIBS  += $(LIBCXX)
+  OTHER_LDLIBS += $(LIBCXX) -lc
 # setup the list of libraries to link in...
 ifeq ($(PLATFORM), linux)
 ifeq ("$(CC_VER_MAJOR)", "3")
-  OTHER_LDLIBS  += -lz -Wl,-Bstatic -lgcc_eh -Wl,-Bdynamic
+  OTHER_LDLIBS  += -Wl,-Bstatic -lgcc_eh -Wl,-Bdynamic
 endif
 endif #LINUX
 endif #PLATFORM
@@ -150,7 +149,7 @@
 	$(prep-target)
 	$(RM) $(TEMPDIR)/mapfile-vers
 	$(CP) mapfile-vers-unpack200 $(TEMPDIR)/mapfile-vers
-	$(LINKER)  $(LDDFLAGS) $(UNPACK_EXE_FILES_o) $(RES) $(LIBCXX) $(LDOUTPUT)$(TEMPDIR)/unpack200$(EXE_SUFFIX)
+	$(LINKER)  $(LDDFLAGS) $(UNPACK_EXE_FILES_o) $(RES) $(OTHER_LDLIBS) $(LIBCXX) $(LDOUTPUT)$(TEMPDIR)/unpack200$(EXE_SUFFIX)
 ifdef MT
 	$(MT) /manifest $(OBJDIR)/unpack200$(EXE_SUFFIX).manifest /outputresource:$(TEMPDIR)/unpack200$(EXE_SUFFIX);#1
 endif
diff -Nru openjdk.orig/jdk/make/javax/sound/jsoundalsa/Makefile openjdk/jdk/make/javax/sound/jsoundalsa/Makefile
--- openjdk.orig/jdk/make/javax/sound/jsoundalsa/Makefile	2011-06-11 00:38:06.000000000 +0100
+++ openjdk/jdk/make/javax/sound/jsoundalsa/Makefile	2011-06-13 15:20:50.441058241 +0100
@@ -65,7 +65,7 @@
 	$(MIDIFILES_export) \
 	$(PORTFILES_export)
 
-LDFLAGS += -lasound
+OTHER_LDLIBS += -lasound
 
 CPPFLAGS += \
 	-DUSE_DAUDIO=TRUE \
