--- openjdk/hotspot/make/linux/makefiles/gcc.make~	2009-10-30 11:54:26.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2009-10-30 11:55:34.000000000 +0100
@@ -34,13 +34,14 @@
 CC_VER_MAJOR := $(shell $(CC) -dumpversion | sed 's/egcs-//' | cut -d'.' -f1)
 CC_VER_MINOR := $(shell $(CC) -dumpversion | sed 's/egcs-//' | cut -d'.' -f2)
 
+ifeq (,$(DISABLE_PRECOMPILED_HEADER))
 # check for precompiled headers support
 ifneq "$(shell expr \( $(CC_VER_MAJOR) \> 3 \) \| \( \( $(CC_VER_MAJOR) = 3 \) \& \( $(CC_VER_MINOR) \>= 4 \) \))" "0"
 USE_PRECOMPILED_HEADER=1
 PRECOMPILED_HEADER_DIR=.
 PRECOMPILED_HEADER=$(PRECOMPILED_HEADER_DIR)/incls/_precompiled.incl.gch
 endif
-
+endif
 
 #------------------------------------------------------------------------
 # Compiler flags
--- openjdk/hotspot/make/solaris/makefiles/gcc.make~	2009-10-30 11:54:26.000000000 +0100
+++ openjdk/hotspot/make/solaris/makefiles/gcc.make	2009-10-30 11:56:05.000000000 +0100
@@ -45,12 +45,14 @@
 $(shell $(CC) -dumpversion | sed 's/egcs-//' | cut -d'.' -f2)
 
 
+ifeq (,$(DISABLE_PRECOMPILED_HEADER))
 # check for precompiled headers support
 ifneq "$(shell expr \( $(CC_VER_MAJOR) \> 3 \) \| \( \( $(CC_VER_MAJOR) = 3 \) \& \( $(CC_VER_MINOR) \>= 4 \) \))" "0"
 USE_PRECOMPILED_HEADER=1
 PRECOMPILED_HEADER_DIR=.
 PRECOMPILED_HEADER=$(PRECOMPILED_HEADER_DIR)/incls/_precompiled.incl.gch
 endif
+endif
 
 
 #------------------------------------------------------------------------
