diff --git a/src/share/lib/security/java.security-linux b/src/share/lib/security/java.security-linux
--- openjdk/jdk/src/share/lib/security/java.security-linux
+++ openjdk/jdk/src/share/lib/security/java.security-linux
@@ -55,7 +55,7 @@
 # the NSS security provider was not enabled for this build; it can be enabled
 # if NSS (libnss3) is available on the machine. The nss.cfg file may need
 # editing to reflect the location of the NSS installation.
-#security.provider.10=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
+security.provider.10=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
 
 #
 # Select the source of seed data for SecureRandom. By default an
diff --git a/src/share/lib/security/java.security-macosx b/src/share/lib/security/java.security-macosx
--- openjdk/jdk/src/share/lib/security/java.security-macosx
+++ openjdk/jdk/src/share/lib/security/java.security-macosx
@@ -56,7 +56,7 @@
 # the NSS security provider was not enabled for this build; it can be enabled
 # if NSS (libnss3) is available on the machine. The nss.cfg file may need
 # editing to reflect the location of the NSS installation.
-#security.provider.11=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
+security.provider.11=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
 
 #
 # Select the source of seed data for SecureRandom. By default an
diff --git a/src/share/lib/security/java.security-solaris b/src/share/lib/security/java.security-solaris
--- openjdk/jdk/src/share/lib/security/java.security-solaris
+++ openjdk/jdk/src/share/lib/security/java.security-solaris
@@ -57,7 +57,7 @@
 # the NSS security provider was not enabled for this build; it can be enabled
 # if NSS (libnss3) is available on the machine. The nss.cfg file may need
 # editing to reflect the location of the NSS installation.
-#security.provider.12=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
+security.provider.12=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
 
 #
 # Select the source of seed data for SecureRandom. By default an
diff --git a/src/share/lib/security/java.security-windows b/src/share/lib/security/java.security-windows
--- openjdk/jdk/src/share/lib/security/java.security-windows
+++ openjdk/jdk/src/share/lib/security/java.security-windows
@@ -56,7 +56,7 @@
 # the NSS security provider was not enabled for this build; it can be enabled
 # if NSS (libnss3) is available on the machine. The nss.cfg file may need
 # editing to reflect the location of the NSS installation.
-#security.provider.11=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
+security.provider.11=sun.security.pkcs11.SunPKCS11 ${java.home}/lib/security/nss.cfg
 
 #
 # Select the source of seed data for SecureRandom. By default an
