diff -r bb14c5df7088 src/cpu/zero/vm/vm_version_zero.hpp
--- openjdk/hotspot/src/cpu/zero/vm/vm_version_zero.hpp	Thu Jul 14 04:02:40 2016 +0100
+++ openjdk/hotspot/src/cpu/zero/vm/vm_version_zero.hpp	Fri Jul 15 17:17:17 2016 +0100
@@ -1,6 +1,6 @@
 /*
  * Copyright (c) 2003, 2010, Oracle and/or its affiliates. All rights reserved.
- * Copyright 2007 Red Hat, Inc.
+ * Copyright 2015 Red Hat, Inc.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -34,6 +34,7 @@
   static const char* cpu_features() {
     return "";
   }
+  static void initialize();
 };
 
 #endif // CPU_ZERO_VM_VM_VERSION_ZERO_HPP
