diff -Nru openjdk.orig/corba/make/common/shared/Platform.gmk openjdk/corba/make/common/shared/Platform.gmk
--- openjdk.orig/corba/make/common/shared/Platform.gmk	2011-04-21 15:22:49.000000000 +0100
+++ openjdk/corba/make/common/shared/Platform.gmk	2011-04-21 19:57:25.486492074 +0100
@@ -172,6 +172,9 @@
                 arm*) \
                     echo arm \
                     ;; \
+                sh*) \
+                    echo sh \
+		    ;; \
                 *) \
                     echo $(mach) \
                     ;; \
@@ -199,6 +202,9 @@
       else
         ARCH_DATA_MODEL=64
       endif
+      ifeq ($(ARCH), sh)
+        ARCH_DATA_MODEL=32
+      endif
     endif
   endif
 
diff -Nru openjdk.orig/hotspot/src/os/linux/vm/os_linux.cpp openjdk/hotspot/src/os/linux/vm/os_linux.cpp
--- openjdk.orig/hotspot/src/os/linux/vm/os_linux.cpp	2011-04-21 15:22:49.000000000 +0100
+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2011-04-21 19:57:25.490492144 +0100
@@ -1864,7 +1864,8 @@
     {EM_MIPS_RS3_LE, EM_MIPS_RS3_LE, ELFCLASS32, ELFDATA2LSB, (char*)"MIPSel"},
     {EM_MIPS,        EM_MIPS,    ELFCLASS32, ELFDATA2MSB, (char*)"MIPS"},
     {EM_PARISC,      EM_PARISC,  ELFCLASS32, ELFDATA2MSB, (char*)"PARISC"},
-    {EM_68K,         EM_68K,     ELFCLASS32, ELFDATA2MSB, (char*)"M68k"}
+    {EM_68K,         EM_68K,     ELFCLASS32, ELFDATA2MSB, (char*)"M68k"},
+    {EM_SH,          EM_SH,      ELFCLASS32, ELFDATA2LSB, (char*)"SH"} /* Support little endian only*/
   };
 
   #if  (defined IA32)
@@ -1895,9 +1896,11 @@
     static  Elf32_Half running_arch_code=EM_MIPS;
   #elif  (defined M68K)
     static  Elf32_Half running_arch_code=EM_68K;
+  #elif  (defined SH)
+    static  Elf32_Half running_arch_code=EM_SH;
   #else
     #error Method os::dll_load requires that one of following is defined:\
-         IA32, AMD64, IA64, __sparc, __powerpc__, ARM, S390, ALPHA, MIPS, MIPSEL, PARISC, M68K
+         IA32, AMD64, IA64, __sparc, __powerpc__, ARM, S390, ALPHA, MIPS, MIPSEL, PARISC, M68K, SH
   #endif
 
   // Identify compatability class for VM's architecture and library's architecture
diff -Nru openjdk.orig/jdk/make/common/Defs-linux.gmk openjdk/jdk/make/common/Defs-linux.gmk
--- openjdk.orig/jdk/make/common/Defs-linux.gmk	2011-04-21 15:22:49.000000000 +0100
+++ openjdk/jdk/make/common/Defs-linux.gmk	2011-04-21 19:58:06.615219532 +0100
@@ -119,6 +119,7 @@
 LDFLAGS_COMMON_sparc    += -m32 -mcpu=v9
 CFLAGS_REQUIRED_arm     += -fsigned-char -D_LITTLE_ENDIAN
 CFLAGS_REQUIRED_ppc     += -fsigned-char -D_BIG_ENDIAN
+CFLAGS_REQUIRED_sh      += -mieee
 ifeq ($(ZERO_BUILD), true)
   CFLAGS_REQUIRED       =  $(ZERO_ARCHFLAG)
   ifeq ($(ZERO_ENDIANNESS), little)
@@ -197,11 +198,15 @@
 
 CPP_ARCH_FLAGS = -DARCH='"$(ARCH)"'
 
-# Alpha arch does not like "alpha" defined (potential general arch cleanup issue here)
-ifneq ($(ARCH),alpha)
-  CPP_ARCH_FLAGS += -D$(ARCH)
+# Alpha and sh archs do not like "alpha" or "sh" defined (potential general arch cleanup issue here)
+ifeq ($(ARCH),alpha)
+  CPP_ARCH_FLAGS += -D_$(ARCH)_
 else
+ifeq ($(ARCH),sh)
   CPP_ARCH_FLAGS += -D_$(ARCH)_
+else
+  CPP_ARCH_FLAGS += -D$(ARCH)
+endif
 endif
 
 CPPFLAGS_COMMON = $(CPP_ARCH_FLAGS) -DLINUX $(VERSION_DEFINES) \
diff -Nru openjdk.orig/jdk/make/common/shared/Compiler-gcc.gmk openjdk/jdk/make/common/shared/Compiler-gcc.gmk
--- openjdk.orig/jdk/make/common/shared/Compiler-gcc.gmk	2011-04-21 15:22:50.000000000 +0100
+++ openjdk/jdk/make/common/shared/Compiler-gcc.gmk	2011-04-21 19:57:25.490492144 +0100
@@ -119,6 +119,11 @@
     REQUIRED_CC_VER = 3.2
     REQUIRED_GCC_VER = 3.2.*
   endif
+  ifneq ("$(findstring sh,$(ARCH))", "")
+    # sh4*, sh3*
+    REQUIRED_CC_VER = 4.0
+    REQUIRED_GCC_VER = 4.0.*
+  endif
   # Option used to create a shared library
   SHARED_LIBRARY_FLAG = -shared -mimpure-text
   SUN_COMP_VER := $(shell $(CC) --verbose 2>&1 )
diff -Nru openjdk.orig/jdk/make/common/shared/Platform.gmk openjdk/jdk/make/common/shared/Platform.gmk
--- openjdk.orig/jdk/make/common/shared/Platform.gmk	2011-04-21 15:22:49.000000000 +0100
+++ openjdk/jdk/make/common/shared/Platform.gmk	2011-04-21 19:57:25.490492144 +0100
@@ -179,6 +179,9 @@
                 arm*) \
                     echo arm \
                     ;; \
+                sh*) \
+                    echo sh \
+                    ;; \
                 *) \
                     echo $(mach) \
                     ;; \
@@ -208,6 +211,9 @@
       ifeq ($(ARCH), ia64)
         ARCH_DATA_MODEL=64
       endif
+      ifeq ($(ARCH), sh)
+        ARCH_DATA_MODEL=32
+      endif
     endif
   endif
 
diff -Nru openjdk.orig/jdk/make/javax/sound/SoundDefs.gmk openjdk/jdk/make/javax/sound/SoundDefs.gmk
--- openjdk.orig/jdk/make/javax/sound/SoundDefs.gmk	2011-04-21 15:22:49.000000000 +0100
+++ openjdk/jdk/make/javax/sound/SoundDefs.gmk	2011-04-21 19:58:38.367780866 +0100
@@ -126,6 +126,10 @@
     CPPFLAGS += -DX_ARCH=X_PPC
   endif # ARCH ppc
 
+  ifeq ($(ARCH), sh)
+    CPPFLAGS += -DX_ARCH=X_SH
+  endif # ARCH Renesas SuperH(sh)
+
 endif
 
 
diff -Nru openjdk.orig/jdk/make/jdk_generic_profile.sh openjdk/jdk/make/jdk_generic_profile.sh
--- openjdk.orig/jdk/make/jdk_generic_profile.sh	2011-04-11 12:34:30.000000000 +0100
+++ openjdk/jdk/make/jdk_generic_profile.sh	2011-04-21 19:57:25.498492286 +0100
@@ -258,13 +258,14 @@
     i?86)    ZERO_LIBARCH=i386      ;;
     sparc64) ZERO_LIBARCH=sparcv9   ;;
     arm*)    ZERO_LIBARCH=arm       ;;
+    sh*)     ZERO_LIBARCH=sh        ;;
     *)       ZERO_LIBARCH="$(arch)"
   esac
   export ZERO_LIBARCH
 
   # ARCH_DATA_MODEL is the number of bits in a pointer
   case "${ZERO_LIBARCH}" in
-    i386|ppc|s390|sparc|arm)
+    i386|ppc|s390|sparc|arm|sh)
       ARCH_DATA_MODEL=32
       ;;
     amd64|ppc64|s390x|sparcv9|ia64|alpha)
diff -Nru openjdk.orig/jdk/src/share/native/com/sun/media/sound/SoundDefs.h openjdk/jdk/src/share/native/com/sun/media/sound/SoundDefs.h
--- openjdk.orig/jdk/src/share/native/com/sun/media/sound/SoundDefs.h	2011-04-21 15:22:49.000000000 +0100
+++ openjdk/jdk/src/share/native/com/sun/media/sound/SoundDefs.h	2011-04-21 19:59:08.392311421 +0100
@@ -48,7 +48,7 @@
 #define X_PPC64         13
 #define X_S390          14
 #define X_S390X         15
-
+#define X_SH            16
 
 // **********************************
 // Make sure you set X_PLATFORM and X_ARCH defines correctly.
