diff -Nru openjdk.orig/hotspot/src/share/vm/runtime/sharedRuntime.cpp openjdk/hotspot/src/share/vm/runtime/sharedRuntime.cpp
--- openjdk.orig/hotspot/src/share/vm/runtime/sharedRuntime.cpp	2011-04-11 12:33:06.000000000 +0100
+++ openjdk/hotspot/src/share/vm/runtime/sharedRuntime.cpp	2011-04-11 16:40:57.825703429 +0100
@@ -890,7 +890,7 @@
   int size = o->size();
   Symbol* name = klass->name();
   HS_DTRACE_PROBE4(hotspot, object__alloc, get_java_tid(thread),
-                   name->bytes(), name->utf8_length(), size * HeapWordSize);
+                   name->bytes(), name->utf8_length(), size);
   return 0;
 }
 
