diff --git a/make/com/sun/nio/sctp/Makefile b/make/com/sun/nio/sctp/Makefile
--- openjdk/jdk/make/com/sun/nio/sctp/Makefile
+++ openjdk/jdk/make/com/sun/nio/sctp/Makefile
@@ -60,7 +60,6 @@
   -I$(CLASSHDRDIR)/../../../../java/java.nio/nio/CClassHeaders
 
 ifeq ($(PLATFORM), linux)
-COMPILER_WARNINGS_FATAL=true
 #OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -ljava -lnet -lpthread -ldl
 OTHER_LDLIBS += -L$(LIBDIR)/$(LIBARCH) -lnio -lnet -lpthread -ldl
 endif
diff --git a/make/java/nio/Makefile b/make/java/nio/Makefile
--- openjdk/jdk/make/java/nio/Makefile
+++ openjdk/jdk/make/java/nio/Makefile
@@ -31,7 +31,7 @@
 PACKAGE = java.nio
 LIBRARY = nio
 PRODUCT = java
-OTHER_JAVACFLAGS += -Xmaxwarns 1000 -Xlint:serial -Werror
+OTHER_JAVACFLAGS += -Xmaxwarns 1000 -Xlint:serial
 include $(BUILDDIR)/common/Defs.gmk
 
 NIO_SRC = $(SHARE_SRC)/classes/java/nio
diff --git a/make/java/sun_nio/Makefile b/make/java/sun_nio/Makefile
--- openjdk/jdk/make/java/sun_nio/Makefile
+++ openjdk/jdk/make/java/sun_nio/Makefile
@@ -31,7 +31,7 @@
 PACKAGE = sun.nio
 PRODUCT = sun
 
-OTHER_JAVACFLAGS += -Xlint:serial,-deprecation -Werror
+OTHER_JAVACFLAGS += -Xlint:serial,-deprecation
 include $(BUILDDIR)/common/Defs.gmk
 
 #
diff --git a/make/sun/native2ascii/Makefile b/make/sun/native2ascii/Makefile
--- openjdk/jdk/make/sun/native2ascii/Makefile
+++ openjdk/jdk/make/sun/native2ascii/Makefile
@@ -30,7 +30,7 @@
 BUILDDIR = ../..
 PACKAGE = sun.tools.native2ascii
 PRODUCT = sun
-OTHER_JAVACFLAGS += -Xlint:serial -Werror
+OTHER_JAVACFLAGS += -Xlint:serial
 include $(BUILDDIR)/common/Defs.gmk
 
 # This program must contain a manifest that defines the execution level
diff --git a/make/sun/nio/cs/Makefile b/make/sun/nio/cs/Makefile
--- openjdk/jdk/make/sun/nio/cs/Makefile
+++ openjdk/jdk/make/sun/nio/cs/Makefile
@@ -36,7 +36,7 @@
 # This re-directs all the class files to a separate location
 CLASSDESTDIR = $(TEMPDIR)/classes
 
-OTHER_JAVACFLAGS += -Xlint:serial,-deprecation -Werror
+OTHER_JAVACFLAGS += -Xlint:serial,-deprecation
 include $(BUILDDIR)/common/Defs.gmk
 
 #
diff --git a/make/tools/Makefile b/make/tools/Makefile
--- openjdk/jdk/make/tools/Makefile
+++ openjdk/jdk/make/tools/Makefile
@@ -29,7 +29,14 @@
 
 BUILDDIR = ..
 
-SUBDIRS_MAKEFLAGS += JAVAC_MAX_WARNINGS=true JAVAC_WARNINGS_FATAL=true
+ifndef JAVAC_MAX_WARNINGS
+  SUBDIRS_MAKEFLAGS += JAVAC_MAX_WARNINGS=true
+endif
+
+ifndef JAVAC_WARNINGS_FATAL
+  SUBDIRS_MAKEFLAGS += JAVAC_WARNINGS_FATAL=true
+endif
+
 include $(BUILDDIR)/common/Defs.gmk
 
 # Note: freetypecheck is built by Sanity.gmk if needed
