diff --git a/make/linux/makefiles/gcc.make b/make/linux/makefiles/gcc.make
--- openjdk/hotspot/make/linux/makefiles/gcc.make
+++ openjdk/hotspot/make/linux/makefiles/gcc.make
@@ -75,9 +75,6 @@
 VM_PICFLAG/AOUT   =
 VM_PICFLAG        = $(VM_PICFLAG/$(LINK_INTO))
 
-ifeq ($(JVM_VARIANT_ZERO), true)
-CFLAGS += $(LIBFFI_CFLAGS)
-endif
 ifeq ($(JVM_VARIANT_ZEROSHARK), true)
 CFLAGS += $(LIBFFI_CFLAGS)
 CFLAGS += $(LLVM_CFLAGS)
diff --git a/make/linux/makefiles/vm.make b/make/linux/makefiles/vm.make
--- openjdk/hotspot/make/linux/makefiles/vm.make
+++ openjdk/hotspot/make/linux/makefiles/vm.make
@@ -284,9 +284,6 @@
 
   LIBS_VM                  += $(LIBS)
 endif
-ifeq ($(JVM_VARIANT_ZERO), true)
-  LIBS_VM += $(LIBFFI_LIBS)
-endif
 ifeq ($(JVM_VARIANT_ZEROSHARK), true)
   LIBS_VM   += $(LIBFFI_LIBS) $(LLVM_LIBS)
   LFLAGS_VM += $(LLVM_LDFLAGS)
diff --git a/make/linux/makefiles/zero.make b/make/linux/makefiles/zero.make
--- openjdk/hotspot/make/linux/makefiles/zero.make
+++ openjdk/hotspot/make/linux/makefiles/zero.make
@@ -30,3 +30,7 @@
 
 # Install libjvm.so, etc in in server directory.
 VM_SUBDIR = server
+
+# Make sure libffi is included
+CFLAGS += $(LIBFFI_CFLAGS)
+LIBS_VM += $(LIBFFI_LIBS)
