diff -ru openjdk/j2se/make/common/Release.gmk openjdk/j2se/make/common/Release.gmk
--- openjdk/j2se/make/common/Release.gmk	2007-05-25 15:13:23.000000000 -0400
+++ openjdk/j2se/make/common/Release.gmk	2007-05-30 12:06:47.000000000 -0400
@@ -273,6 +273,7 @@
 
 
 CLOSED_CLASSES =  \
+gnu \
 sun/dc \
 java/awt/color/CMMException.class \
 java/awt/color/ColorSpace.class \
@@ -309,28 +310,18 @@
 com/sun/jmx/snmp/SnmpDataTypeEnums.class \
 com/sun/jmx/snmp/SnmpDefinitions.class \
 com/sun/jmx/snmp/SnmpOid.class \
-com/sun/jmx/snmp/SnmpOidDatabase.class \
-com/sun/jmx/snmp/SnmpOidDatabaseSupport.class \
 com/sun/jmx/snmp/SnmpOidRecord.class \
 com/sun/jmx/snmp/SnmpOidTable.class \
 com/sun/jmx/snmp/SnmpOidTableSupport.class \
 com/sun/jmx/snmp/SnmpParameters.class \
 com/sun/jmx/snmp/SnmpPduPacket.class \
 com/sun/jmx/snmp/SnmpPeer.class \
+com/sun/jmx/snmp/SnmpSession.class \
 com/sun/jmx/snmp/SnmpTimeticks.class \
 com/sun/jmx/snmp/SnmpVarBind.class \
 com/sun/jmx/snmp/SnmpVarBindList.class \
-com/sun/jmx/snmp/Timestamp.class \
 com/sun/jmx/snmp/daemon/SendQ.class \
 com/sun/jmx/snmp/daemon/SnmpInformRequest.class \
-com/sun/jmx/snmp/daemon/SnmpQManager.class \
-com/sun/jmx/snmp/daemon/SnmpRequestCounter.class \
-com/sun/jmx/snmp/daemon/SnmpResponseHandler.class \
-com/sun/jmx/snmp/daemon/SnmpSendServer.class \
-com/sun/jmx/snmp/daemon/SnmpSession.class \
-com/sun/jmx/snmp/daemon/SnmpSocket.class \
-com/sun/jmx/snmp/daemon/SnmpTimerServer.class \
-com/sun/jmx/snmp/daemon/WaitQ.class \
 com/sun/media/sound
 
 ABSJAR = $(ABS_BINDIR)/jar
diff -ru openjdk/j2se/make/java/awt/Makefile openjdk/j2se/make/java/awt/Makefile
--- openjdk/j2se/make/java/awt/Makefile	2007-05-25 15:13:24.000000000 -0400
+++ openjdk/j2se/make/java/awt/Makefile	2007-05-25 15:23:16.000000000 -0400
@@ -35,6 +35,7 @@
 
 ifdef OPENJDK
 CLOSED_AWT_CLASSES =  \
+gnu \
 java/awt/color/CMMException.class \
 java/awt/color/ColorSpace.class \
 java/awt/color/ICC_ColorSpace.class \
diff -ru openjdk/j2se/make/java/redist/Makefile openjdk/j2se/make/java/redist/Makefile
--- openjdk/j2se/make/java/redist/Makefile	2007-05-25 15:13:24.000000000 -0400
+++ openjdk/j2se/make/java/redist/Makefile	2007-05-30 12:07:39.000000000 -0400
@@ -308,6 +308,7 @@
 # that a local clobber and remake works.
 CLOSED_CLASSES =  \
 sun/dc \
+gnu \
 java/awt/color/CMMException.class \
 java/awt/color/ColorSpace.class \
 java/awt/color/ICC_ColorSpace.class \
@@ -343,28 +344,18 @@
 com/sun/jmx/snmp/SnmpDataTypeEnums.class \
 com/sun/jmx/snmp/SnmpDefinitions.class \
 com/sun/jmx/snmp/SnmpOid.class \
-com/sun/jmx/snmp/SnmpOidDatabase.class \
-com/sun/jmx/snmp/SnmpOidDatabaseSupport.class \
 com/sun/jmx/snmp/SnmpOidRecord.class \
 com/sun/jmx/snmp/SnmpOidTable.class \
 com/sun/jmx/snmp/SnmpOidTableSupport.class \
 com/sun/jmx/snmp/SnmpParameters.class \
 com/sun/jmx/snmp/SnmpPduPacket.class \
 com/sun/jmx/snmp/SnmpPeer.class \
+com/sun/jmx/snmp/SnmpSession.class \
 com/sun/jmx/snmp/SnmpTimeticks.class \
 com/sun/jmx/snmp/SnmpVarBind.class \
 com/sun/jmx/snmp/SnmpVarBindList.class \
-com/sun/jmx/snmp/Timestamp.class \
 com/sun/jmx/snmp/daemon/SendQ.class \
 com/sun/jmx/snmp/daemon/SnmpInformRequest.class \
-com/sun/jmx/snmp/daemon/SnmpQManager.class \
-com/sun/jmx/snmp/daemon/SnmpRequestCounter.class \
-com/sun/jmx/snmp/daemon/SnmpResponseHandler.class \
-com/sun/jmx/snmp/daemon/SnmpSendServer.class \
-com/sun/jmx/snmp/daemon/SnmpSession.class \
-com/sun/jmx/snmp/daemon/SnmpSocket.class \
-com/sun/jmx/snmp/daemon/SnmpTimerServer.class \
-com/sun/jmx/snmp/daemon/WaitQ.class \
 com/sun/media/sound
 build: openjdkredist
 openjdkredist:
diff -ru openjdk/j2se/make/tools/sharing/classlist.linux openjdk/j2se/make/tools/sharing/classlist.linux
--- openjdk/j2se/make/tools/sharing/classlist.linux	2007-05-25 15:13:23.000000000 -0400
+++ openjdk/j2se/make/tools/sharing/classlist.linux	2007-05-25 15:24:08.000000000 -0400
@@ -638,7 +638,6 @@
 sun/awt/color/ProfileDeferralInfo
 sun/awt/color/ProfileDeferralMgr
 java/awt/color/ICC_ProfileRGB
-java/awt/color/ICC_Profile$1
 sun/awt/color/ProfileActivator
 java/awt/color/ICC_ColorSpace
 sun/java2d/pipe/NullPipe
diff -ru openjdk/j2se/make/tools/sharing/classlist.solaris openjdk/j2se/make/tools/sharing/classlist.solaris
--- openjdk/j2se/make/tools/sharing/classlist.solaris	2007-05-25 15:13:23.000000000 -0400
+++ openjdk/j2se/make/tools/sharing/classlist.solaris	2007-05-25 15:24:17.000000000 -0400
@@ -512,7 +512,6 @@
 sun/awt/color/ProfileDeferralInfo
 sun/awt/color/ProfileDeferralMgr
 java/awt/color/ICC_ProfileRGB
-java/awt/color/ICC_Profile$1
 sun/awt/color/ProfileActivator
 java/awt/color/ICC_ColorSpace
 sun/java2d/pipe/NullPipe
diff -ru openjdk/j2se/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java openjdk/j2se/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java
--- openjdk/j2se/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java	2007-05-25 15:13:00.000000000 -0400
+++ openjdk/j2se/src/share/classes/com/sun/jmx/mbeanserver/OpenConverter.java	2007-05-30 11:04:58.000000000 -0400
@@ -1152,7 +1152,7 @@
             Set<BitSet> getterIndexSets = newSet();
             for (Constructor constr : annotatedConstrList) {
                 String[] propertyNames =
-                    constr.getAnnotation(propertyNamesClass).value();
+                    ((ConstructorProperties)constr.getAnnotation(propertyNamesClass)).value();
 
                 Type[] paramTypes = constr.getGenericParameterTypes();
                 if (paramTypes.length != propertyNames.length) {
diff -ru openjdk/j2se/src/share/classes/com/sun/jmx/snmp/daemon/SnmpAdaptorServer.java openjdk/j2se/src/share/classes/com/sun/jmx/snmp/daemon/SnmpAdaptorServer.java
--- openjdk/j2se/src/share/classes/com/sun/jmx/snmp/daemon/SnmpAdaptorServer.java	2007-05-25 15:13:00.000000000 -0400
+++ openjdk/j2se/src/share/classes/com/sun/jmx/snmp/daemon/SnmpAdaptorServer.java	2007-05-29 16:46:51.000000000 -0400
@@ -65,6 +65,7 @@
 import com.sun.jmx.snmp.InetAddressAcl;
 import com.sun.jmx.snmp.SnmpPeer;
 import com.sun.jmx.snmp.SnmpParameters;
+import com.sun.jmx.snmp.SnmpSession;
 // SNMP Runtime imports
 //
 import com.sun.jmx.snmp.SnmpPduFactoryBER;
diff -ru openjdk/j2se/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java openjdk/j2se/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java
--- openjdk/j2se/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java	2007-05-25 15:13:00.000000000 -0400
+++ openjdk/j2se/src/share/classes/com/sun/jmx/snmp/SnmpPduTrap.java	2007-05-30 10:58:09.000000000 -0400
@@ -78,6 +78,9 @@
      */
     public long            timeStamp ;
 
+    // TODO: IcedTea: I am a stub.
+    static public int trapAuthenticationFailure = 0;
+
 
 
     /**
diff -ru openjdk/j2se/src/share/classes/com/sun/script/javascript/JavaAdapter.java openjdk/j2se/src/share/classes/com/sun/script/javascript/JavaAdapter.java
--- openjdk/j2se/src/share/classes/com/sun/script/javascript/JavaAdapter.java	2007-05-25 15:13:00.000000000 -0400
+++ openjdk/j2se/src/share/classes/com/sun/script/javascript/JavaAdapter.java	2007-05-29 16:44:40.000000000 -0400
@@ -27,6 +27,7 @@
 
 import javax.script.Invocable;
 import sun.org.mozilla.javascript.internal.*;
+import javax.script.Wrapper;
 
 /**
  * This class implements Rhino-like JavaAdapter to help implement a Java
