--- openjdk/hotspot/make/Makefile.orig	2008-11-25 09:11:57.000000000 +0000
+++ openjdk/hotspot/make/Makefile	2008-12-05 16:54:54.000000000 +0000
@@ -91,6 +91,15 @@
 all_debug:     jvmg jvmg1 jvmgkernel docs export_debug
 all_optimized: optimized optimized1 optimizedkernel docs export_optimized
 
+# Core (non-compiler) targets made available with this Makefile
+CORE_VM_TARGETS=productcore fastdebugcore optimizedcore jvmgcore
+
+allcore:           all_productcore all_fastdebugcore
+all_productcore:   productcore docs export_product
+all_fastdebugcore: fastdebugcore docs export_fastdebug
+all_debugcore:     jvmgcore docs export_debug
+all_optimizedcore: optimizedcore docs export_optimized
+
 # Do everything
 world:         all create_jdk
 
@@ -113,6 +122,10 @@
 	$(CD) $(GAMMADIR)/make; \
 	$(MAKE) VM_TARGET=$@ generic_build2 $(ALT_OUT)
 
+$(CORE_VM_TARGETS):
+	$(CD) $(GAMMADIR)/make; \
+	$(MAKE) VM_TARGET=$@ generic_buildcore $(ALT_OUT)
+
 $(KERNEL_VM_TARGETS):
 	$(CD) $(GAMMADIR)/make; \
 	$(MAKE) VM_TARGET=$@ generic_buildkernel $(ALT_OUT)
@@ -159,6 +172,12 @@
 		      $(MAKE_ARGS) $(VM_TARGET)
 endif
 
+generic_buildcore:
+	$(MKDIR) -p $(OUTPUTDIR)
+	$(CD) $(OUTPUTDIR); \
+		$(MAKE) -f $(ABS_OS_MAKEFILE) \
+			$(MAKE_ARGS) $(VM_TARGET)
+
 generic_buildkernel:
 	$(MKDIR) -p $(OUTPUTDIR)
 ifeq ($(OSNAME),windows)
@@ -205,13 +224,21 @@
 XUSAGE=$(HS_SRC_DIR)/share/vm/Xusage.txt
 DOCS_DIR=$(OUTPUTDIR)/$(VM_PLATFORM)_docs
 C1_BASE_DIR=$(OUTPUTDIR)/$(VM_PLATFORM)_compiler1
+ifdef ICEDTEA_CORE_BUILD
+C2_BASE_DIR=$(OUTPUTDIR)/$(VM_PLATFORM)_core
+else
 C2_BASE_DIR=$(OUTPUTDIR)/$(VM_PLATFORM)_compiler2
+endif
 KERNEL_BASE_DIR=$(OUTPUTDIR)/$(VM_PLATFORM)_kernel
 C1_DIR=$(C1_BASE_DIR)/$(VM_SUBDIR)
 C2_DIR=$(C2_BASE_DIR)/$(VM_SUBDIR)
 KERNEL_DIR=$(KERNEL_BASE_DIR)/$(VM_SUBDIR)
 
 # Misc files and generated files need to come from C1 or C2 area
+ifdef ICEDTEA_CORE_BUILD
+  MISC_DIR=$(C2_DIR)
+  GEN_DIR=$(C2_BASE_DIR)/generated
+else
 ifeq ($(ARCH_DATA_MODEL), 32)
   MISC_DIR=$(C1_DIR)
   GEN_DIR=$(C1_BASE_DIR)/generated
@@ -219,6 +246,7 @@
   MISC_DIR=$(C2_DIR)
   GEN_DIR=$(C2_BASE_DIR)/generated
 endif
+endif
 
 # Bin files (windows)
 ifeq ($(OSNAME),windows)
--- openjdk/hotspot/src/share/vm/runtime/vm_version.cpp.orig	2008-12-05 16:54:13.000000000 +0000
+++ openjdk/hotspot/src/share/vm/runtime/vm_version.cpp	2008-12-05 16:54:54.000000000 +0000
@@ -89,8 +89,12 @@
 #ifdef TIERED
   #define VMTYPE "Server"
 #else
-  #define VMTYPE COMPILER1_PRESENT("Client")   \
-                 COMPILER2_PRESENT("Server")   
+#if defined(COMPILER1) || defined(COMPILER2)
+   #define VMTYPE COMPILER1_PRESENT("Client")   \
+                  COMPILER2_PRESENT("Server")
+#else
+  #define VMTYPE "Core"
+#endif // COMPILER1 || COMPILER2
 #endif // TIERED
 
 #ifndef HOTSPOT_VM_DISTRO
--- openjdk/hotspot/build/linux/makefiles/defs.make.orig	2008-11-25 09:11:51.000000000 +0000
+++ openjdk/hotspot/build/linux/makefiles/defs.make	2008-12-05 16:54:54.000000000 +0000
@@ -97,6 +97,7 @@
 EXPORT_LIST += $(EXPORT_SERVER_DIR)/Xusage.txt
 EXPORT_LIST += $(EXPORT_SERVER_DIR)/libjsig.so
 EXPORT_LIST += $(EXPORT_SERVER_DIR)/libjvm.so
+ifndef ICEDTEA_CORE_BUILD
 ifeq ($(ARCH_DATA_MODEL), 32)
   EXPORT_CLIENT_DIR = $(EXPORT_JRE_LIB_ARCH_DIR)/client
   EXPORT_LIST += $(EXPORT_CLIENT_DIR)/Xusage.txt
@@ -111,3 +112,4 @@
       EXPORT_LIST += $(EXPORT_LIB_DIR)/sa-jdi.jar
     endif
 endif
+endif
