diff -Nru openjdk.orig/corba/make/common/shared/Platform.gmk openjdk/corba/make/common/shared/Platform.gmk
--- openjdk.orig/corba/make/common/shared/Platform.gmk	2011-03-14 22:08:07.000000000 +0000
+++ openjdk/corba/make/common/shared/Platform.gmk	2011-04-21 10:35:01.629157957 +0100
@@ -153,6 +153,9 @@
   OS_VERSION := $(shell uname -r)
   # Arch and OS name/version
   mach := $(shell uname -m)
+  ifneq (,$(wildcard /usr/bin/dpkg-architecture))
+    mach := $(shell (dpkg-architecture -qDEB_BUILD_ARCH_CPU 2>/dev/null || echo $(mach)) | sed 's/powerpc$$/ppc/;s/hppa/parisc/')
+  endif
   archExpr = case "$(mach)" in \
                 i[3-9]86) \
                     echo i586 \
diff -Nru openjdk.orig/jdk/make/common/shared/Platform.gmk openjdk/jdk/make/common/shared/Platform.gmk
--- openjdk.orig/jdk/make/common/shared/Platform.gmk	2011-04-20 04:40:20.000000000 +0100
+++ openjdk/jdk/make/common/shared/Platform.gmk	2011-04-21 10:35:35.653711575 +0100
@@ -160,6 +160,9 @@
   else
     mach := $(shell uname -m)
   endif
+  ifneq (,$(wildcard /usr/bin/dpkg-architecture))
+    mach := $(shell (dpkg-architecture -qDEB_BUILD_ARCH_CPU 2>/dev/null || echo $(mach)) | sed 's/powerpc$$/ppc/;s/hppa/parisc/')
+  endif
   archExpr = case "$(mach)" in \
                 i[3-9]86) \
                     echo i586 \
