diff -Nru openjdk.orig/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c openjdk/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c
--- openjdk-ecj.orig/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c	2009-03-18 16:25:52.000000000 +0000
+++ openjdk-ecj/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c	2009-03-18 16:27:03.000000000 +0000
@@ -36,6 +36,14 @@
 #include "nio.h"
 #include "nio_util.h"
 
+#undef sun_nio_ch_FileDispatcher_NO_LOCK
+#define sun_nio_ch_FileDispatcher_NO_LOCK -1L
+#undef sun_nio_ch_FileDispatcher_LOCKED
+#define sun_nio_ch_FileDispatcher_LOCKED 0L
+#undef sun_nio_ch_FileDispatcher_RET_EX_LOCK
+#define sun_nio_ch_FileDispatcher_RET_EX_LOCK 1L
+#undef sun_nio_ch_FileDispatcher_INTERRUPTED
+#define sun_nio_ch_FileDispatcher_INTERRUPTED 2L
 
 static int preCloseFD = -1;     /* File descriptor to which we dup other fd's
                                    before closing them for real */
