diff -Nru openjdk.orig/hotspot/src/share/vm/gc_implementation/includeDB_gc_parallelScavenge openjdk/hotspot/src/share/vm/gc_implementation/includeDB_gc_parallelScavenge
--- openjdk.orig/hotspot/src/share/vm/gc_implementation/includeDB_gc_parallelScavenge	2009-02-19 08:40:26.000000000 +0000
+++ openjdk/hotspot/src/share/vm/gc_implementation/includeDB_gc_parallelScavenge	2009-02-23 21:09:49.000000000 +0000
@@ -308,6 +308,8 @@
 psPermGen.cpp                           psMarkSweepDecorator.hpp
 psPermGen.cpp                           psParallelCompact.hpp
 psPermGen.cpp                           psPermGen.hpp
+psPermGen.cpp                           markOop.inline.hpp
+psPermGen.cpp                           markSweep.inline.hpp
 
 psPermGen.hpp                           psOldGen.hpp
 
