diff -Nru openjdk.orig/hotspot/agent/src/os/linux/Makefile openjdk/hotspot/agent/src/os/linux/Makefile
--- openjdk.orig/hotspot/agent/src/os/linux/Makefile	2010-07-29 21:54:38.000000000 +0100
+++ openjdk/hotspot/agent/src/os/linux/Makefile	2011-04-21 00:04:10.990086029 +0100
@@ -23,7 +23,7 @@
 #
 
 ARCH := $(shell if ([ `uname -m` = "ia64" ])  ; then echo ia64 ; elif ([ `uname -m` = "x86_64" ]) ; then echo amd64; elif ([ `uname -m` = "sparc64" ]) ; then echo sparc; else echo i386 ; fi )
-GCC      = gcc
+GCC      = gcc$(GCC_SUFFIX)
 
 JAVAH    = ${JAVA_HOME}/bin/javah
 
diff -Nru openjdk.orig/hotspot/make/linux/makefiles/gcc.make openjdk/hotspot/make/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/make/linux/makefiles/gcc.make	2011-04-20 12:06:06.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2011-04-21 01:00:54.300899166 +0100
@@ -28,13 +28,13 @@
 # When cross-compiling the ALT_COMPILER_PATH points
 # to the cross-compilation toolset
 ifdef CROSS_COMPILE_ARCH
-CPP = $(ALT_COMPILER_PATH)/g++
-CC  = $(ALT_COMPILER_PATH)/gcc
-HOSTCPP = g++
-HOSTCC  = gcc
+CPP = $(ALT_COMPILER_PATH)/g++$(GCC_SUFFIX)
+CC  = $(ALT_COMPILER_PATH)/gcc$(GCC_SUFFIX)
+HOSTCPP = g++$(GCC_SUFFIX)
+HOSTCC  = gcc$(GCC_SUFFIX)
 else
-CPP = g++
-CC  = gcc
+CPP = g++$(GCC_SUFFIX)
+CC  = gcc$(GCC_SUFFIX)
 HOSTCPP = $(CPP)
 HOSTCC  = $(CC)
 endif
