diff -Nru openjdk.orig/hotspot/make/linux/makefiles/gcc.make openjdk/hotspot/make/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/make/linux/makefiles/gcc.make	2008-12-01 12:06:51.000000000 +0000
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2008-12-01 12:27:08.000000000 +0000
@@ -98,7 +98,7 @@
 endif
 
 # Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror
+# WARNINGS_ARE_ERRORS = -Werror
 
 # Except for a few acceptable ones
 # Since GCC 4.3, -Wconversion has changed its meanings to warn these implicit
diff -Nru openjdk.orig/hotspot/make/solaris/makefiles/gcc.make openjdk/hotspot/make/solaris/makefiles/gcc.make
--- openjdk.orig/hotspot/make/solaris/makefiles/gcc.make	2008-11-22 00:11:18.000000000 +0000
+++ openjdk/hotspot/make/solaris/makefiles/gcc.make	2008-12-01 12:25:10.000000000 +0000
@@ -109,7 +109,8 @@
 
 
 # Compiler warnings are treated as errors 
-WARNINGS_ARE_ERRORS = -Werror 
+# Commented out for now because of gcc 4.3 warnings OpenJDK isn't ready for
+# WARNINGS_ARE_ERRORS = -Werror 
 # Enable these warnings. See 'info gcc' about details on these options
 ADDITIONAL_WARNINGS = -Wpointer-arith -Wconversion -Wsign-compare 
 CFLAGS_WARN/DEFAULT = $(WARNINGS_ARE_ERRORS) $(ADDITIONAL_WARNINGS) 
diff -Nru openjdk.orig/hotspot/src/share/vm/adlc/filebuff.cpp openjdk/hotspot/src/share/vm/adlc/filebuff.cpp
--- openjdk.orig/hotspot/src/share/vm/adlc/filebuff.cpp	2008-11-22 00:11:18.000000000 +0000
+++ openjdk/hotspot/src/share/vm/adlc/filebuff.cpp	2008-12-01 12:25:10.000000000 +0000
@@ -25,6 +25,8 @@
 // FILEBUFF.CPP - Routines for handling a parser file buffer
 #include "adlc.hpp"
 
+using namespace std;
+
 //------------------------------FileBuff---------------------------------------
 // Create a new parsing buffer
 FileBuff::FileBuff( BufferedFile *fptr, ArchDesc& archDesc) : _fp(fptr), _AD(archDesc) {
diff -Nru openjdk.orig/hotspot/src/share/vm/adlc/filebuff.hpp openjdk/hotspot/src/share/vm/adlc/filebuff.hpp
--- openjdk.orig/hotspot/src/share/vm/adlc/filebuff.hpp	2008-11-22 00:11:18.000000000 +0000
+++ openjdk/hotspot/src/share/vm/adlc/filebuff.hpp	2008-12-01 12:25:10.000000000 +0000
@@ -94,8 +94,11 @@
   FileBuffRegion *copy();                   // Deep copy
   FileBuffRegion *merge(FileBuffRegion*); // Merge 2 regions; delete input
 
-//  void print(std::ostream&);
-//  friend std::ostream& operator<< (std::ostream&, FileBuffRegion&);
+#if defined(__GNUC__) && __GNUC__ >= 3
+  void print(std::ostream&);
+  friend std::ostream& operator<< (std::ostream&, FileBuffRegion&);
+#else
   void print(ostream&);
   friend ostream& operator<< (ostream&, FileBuffRegion&);
+#endif
 };
