diff -Nru cacao.orig/cacao-8948a434c10d/contrib/mapfile-vers-product cacao/cacao-8948a434c10d/contrib/mapfile-vers-product
--- cacao.orig/cacao-8948a434c10d/contrib/mapfile-vers-product	2010-05-19 12:14:46.000000000 +0100
+++ cacao/cacao-8948a434c10d/contrib/mapfile-vers-product	2010-06-21 20:08:28.000000000 +0100
@@ -86,6 +86,7 @@
                 JVM_EnableCompiler;
                 JVM_Exit;
                 JVM_FillInStackTrace;
+		JVM_FindClassFromBootLoader;
                 JVM_FindClassFromClass;
                 JVM_FindClassFromClassLoader;
                 JVM_FindLibraryEntry;
diff -Nru cacao.orig/cacao-8948a434c10d/src/native/vm/openjdk/jvm.cpp cacao/cacao-8948a434c10d/src/native/vm/openjdk/jvm.cpp
--- cacao.orig/cacao-8948a434c10d/src/native/vm/openjdk/jvm.cpp	2010-05-19 12:14:46.000000000 +0100
+++ cacao/cacao-8948a434c10d/src/native/vm/openjdk/jvm.cpp	2010-06-21 20:08:28.000000000 +0100
@@ -668,6 +668,9 @@
 
 	assert(throwError == false);
 
+	if (name == NULL)
+		return NULL;
+
 	u  = utf_new_char(name);
 	cl = loader_hashtable_classloader_add((java_handle_t *) loader);
 
@@ -685,6 +688,15 @@
 }
 
 
+/* JVM_FindClassFromBootLoader */
+
+jclass JVM_FindClassFromBootLoader(JNIEnv* env, const char* name)
+{
+	TRACEJVMCALLS(("JVM_FindClassFromBootLoader(name=%s)", name));
+	return JVM_FindClassFromClassLoader(env, name, JNI_FALSE,
+										(jobject)NULL, false);
+}
+
 /* JVM_FindClassFromClass */
 
 jclass JVM_FindClassFromClass(JNIEnv *env, const char *name, jboolean init, jclass from)
