--- openjdk/jdk/test/java/util/concurrent/BlockingQueue/ProducerConsumerLoops.java~	2008-08-28 10:20:49.000000000 +0200
+++ openjdk/jdk/test/java/util/concurrent/BlockingQueue/ProducerConsumerLoops.java	2008-10-19 17:02:41.000000000 +0200
@@ -37,6 +37,7 @@
  * @compile -source 1.5 ProducerConsumerLoops.java
  * @run main/timeout=3600 ProducerConsumerLoops
  * @summary  multiple producers and consumers using blocking queues
+ * @ignore cacao test hog, ignore for the sake of buildds
  */
 
 import java.util.concurrent.*;
