Index: cacao/cacao/src/vm/vm.cpp
===================================================================
--- cacao.orig/cacao/src/vm/vm.cpp	2011-03-04 00:20:24.000000000 +0100
+++ cacao/cacao/src/vm/vm.cpp	2011-03-04 01:12:45.000000000 +0100
@@ -1239,9 +1239,10 @@
 
 #endif
 		default:
-			printf("Unknown option: %s\n",
+			fprintf(stderr,"Unknown option: %s\n",
 				   vm_args->options[opt_index].optionString);
-			usage();
+			opt_index++;
+			/* Ignore */
 		}
 	}
 
