diff -Nru cacao.orig/Makefile.am cacao/Makefile.am
--- cacao.orig/cacao/src/cacao/Makefile.am	2008-08-04 17:51:28.000000000 +0100
+++ cacao/cacao/src/cacao/Makefile.am	2010-05-11 10:29:35.000000000 +0100
@@ -96,12 +96,12 @@
 	$(mkdir_p) $(prefix)/jre/lib/$(JAVA_ARCH)/server
 	$(LN_S) -f $(libdir)/libjvm.so $(prefix)/jre/lib/$(JAVA_ARCH)/server
 	$(ECHO) $(ECHO_N) > $(prefix)/jre/lib/$(JAVA_ARCH)/server/Xusage.txt
-	$(ECHO) $(ECHO_N) > $(prefix)/jre/lib/$(JAVA_ARCH)/server/libjsig.so
+	$(ECHO) $(ECHO_N) > $(prefix)/jre/lib/$(JAVA_ARCH)/libjsig.so
 
 uninstall-local:
 	rm -f $(prefix)/jre/lib/$(JAVA_ARCH)/server/libjvm.so
 	rm -f $(prefix)/jre/lib/$(JAVA_ARCH)/server/Xusage.txt
-	rm -f $(prefix)/jre/lib/$(JAVA_ARCH)/server/libjsig.so
+	rm -f $(prefix)/jre/lib/$(JAVA_ARCH)/libjsig.so
 endif
 
 
