diff -Nru openjdk.orig/jdk/make/docs/Makefile openjdk/jdk/make/docs/Makefile
--- openjdk.orig/jdk/make/docs/Makefile	2009-03-19 16:10:56.000000000 +0000
+++ openjdk/jdk/make/docs/Makefile	2009-03-19 16:12:59.000000000 +0000
@@ -36,7 +36,7 @@
 ifeq ($(ARCH_DATA_MODEL),64)
   MAX_VM_MEMORY = 1024
 else
-  MAX_VM_MEMORY = 512
+  MAX_VM_MEMORY = 768
 endif
 
 #
diff -Nru openjdk.orig/jdk/src/share/bin/java.c openjdk/jdk/src/share/bin/java.c
--- openjdk.orig/jdk/src/share/bin/java.c	2009-03-19 16:10:56.000000000 +0000
+++ openjdk/jdk/src/share/bin/java.c	2009-03-19 16:18:16.000000000 +0000
@@ -1873,7 +1873,8 @@
       args.classname = classname;
       args.ifn = *ifn;
 
-      rslt = ContinueInNewThread0(JavaMain, threadStackSize, (void*)&args);
+      /*rslt = ContinueInNewThread0(JavaMain, threadStackSize, (void*)&args);*/
+      rslt = JavaMain((void*)&args);
       /* If the caller has deemed there is an error we
        * simply return that, otherwise we return the value of
        * the callee
