--- cacao/cacao/configure.ac~	2009-03-16 12:42:56.000000000 +0100
+++ cacao/cacao/configure.ac	2009-06-23 16:19:20.000000000 +0200
@@ -37,7 +37,7 @@
 
 dnl set optimization and debugging for all architectures and systems
 if test x"$CFLAGS" = "x"; then
-    OPT_CFLAGS="-g -O2"
+    OPT_CFLAGS="-g -O2 -fno-strict-aliasing"
 else
     OPT_CFLAGS=$CFLAGS
 fi
--- cacao.orig/cacao/configure		2009-03-16 12:44:25.000000000 +0100
+++ cacao/cacao/configure		2009-06-05 17:58:48.000000000 +0200
@@ -2825,7 +2825,7 @@
 
 
 if test x"$CFLAGS" = "x"; then
-    OPT_CFLAGS="-g -O2"
+    OPT_CFLAGS="-g -O2 -fno-strict-aliasing"
 else
     OPT_CFLAGS=$CFLAGS
 fi
