--- cacao/cacao/m4/version.m4~	2010-06-23 17:18:58.000000000 +0200
+++ cacao/cacao/m4/version.m4	2010-06-23 21:00:36.739828757 +0200
@@ -24,7 +24,7 @@
 dnl define detailed version numbers
 
 AC_DEFUN([AC_VERSION_DETAIL],[
-version="$PACKAGE_VERSION"
+version=$(echo "$PACKAGE_VERSION" | sed 's/pre.*//')
 if test x`echo "$version" | $SED -e 's/[[0-9a-z+]]*//g'` = "x..";
 then
     major=`echo "$version" | $SED -e 's/\.[[0-9a-z.+]]*$//'`
