--- cacao/cacao/m4/version.m4~	2010-06-23 16:18:58.000000000 +0100
+++ cacao/cacao/m4/version.m4	2010-06-24 19:11:36.000000000 +0100
@@ -29,7 +29,7 @@
 then
     major=`echo "$version" | $SED -e 's/\.[[0-9a-z.+]]*$//'`
     minor=`echo "$version" | $SED -e 's/^[[0-9]]*\.//' -e 's/\.[[0-9a-z.+]]*$//'`
-    micro=`echo "$version" | $SED -e 's/^[[0-9]]*\.[[0-9]]*\.//' -e 's/[[a-z.+]]*$//'`
+    micro=`echo "$version" | $SED -r -e 's/^[[0-9]]*\.[[0-9]]*\.([[0-9]]*).*/\1/'`
     extra=`echo "$version" | $SED -e 's/^[[0-9]]*\.[[0-9]]*\.[[0-9]]*//'`
 else
     major=`echo "$version" | $SED -e 's/\.[[0-9a-z.+]]*$//'`
