diff -Nru openjdk.orig/jdk/make/sun/awt/FILES_c_unix.gmk openjdk/jdk/make/sun/awt/FILES_c_unix.gmk
--- openjdk.orig/jdk/make/sun/awt/FILES_c_unix.gmk	2009-05-12 15:54:23.000000000 +0100
+++ openjdk/jdk/make/sun/awt/FILES_c_unix.gmk	2009-10-23 15:48:51.000000000 +0100
@@ -211,6 +211,7 @@
 	X11SurfaceData.c \
 	X11FontScaler_md.c \
 	X11TextRenderer_md.c \
+        XRTextRenderer_md.c \
         OGLBlitLoops.c \
         OGLBufImgOps.c \
         OGLContext.c \
