diff -Nru openjdk.orig/jdk/make/sun/awt/mapfile-mawt-vers openjdk/jdk/make/sun/awt/mapfile-mawt-vers
--- openjdk.orig/jdk/make/sun/awt/mapfile-mawt-vers	2009-10-23 15:37:31.000000000 +0100
+++ openjdk/jdk/make/sun/awt/mapfile-mawt-vers	2009-10-23 15:48:18.000000000 +0100
@@ -446,7 +446,6 @@
         	Java_sun_java2d_xr_XRRenderer_devCopyArea;
 		Java_sun_java2d_xr_XRMaskFill_maskFill;
 		Java_sun_java2d_xr_XRMaskBlit_maskBlit;
- 		XRT_DrawGlyphList;
 
                 Java_sun_java2d_opengl_OGLContext_getOGLIdString;
                 Java_sun_java2d_opengl_OGLMaskFill_maskFill;
@@ -544,6 +543,7 @@
                 AWTCharAscent;
                 AWTCharDescent;
                 AWTDrawGlyphList;
+ 		XRT_DrawGlyphList;
                 AccelGlyphCache_RemoveAllCellInfos;
 
 	local:
diff -Nru openjdk.orig/jdk/make/sun/awt/mapfile-vers-linux openjdk/jdk/make/sun/awt/mapfile-vers-linux
--- openjdk.orig/jdk/make/sun/awt/mapfile-vers-linux	2009-10-21 16:42:03.000000000 +0100
+++ openjdk/jdk/make/sun/awt/mapfile-vers-linux	2009-10-23 15:48:18.000000000 +0100
@@ -568,6 +568,7 @@
                 AWTCharAscent;
                 AWTCharDescent;
                 AWTDrawGlyphList;
+		XRT_DrawGlyphList;
                 AccelGlyphCache_RemoveAllCellInfos;
 
 	local:
diff -Nru openjdk.orig/jdk/make/sun/headless/mapfile-vers openjdk/jdk/make/sun/headless/mapfile-vers
--- openjdk.orig/jdk/make/sun/headless/mapfile-vers	2009-10-23 15:37:31.000000000 +0100
+++ openjdk/jdk/make/sun/headless/mapfile-vers	2009-10-23 15:48:18.000000000 +0100
@@ -57,8 +57,6 @@
                 Java_sun_java2d_x11_X11SurfaceData_XSetXorMode;
                 Java_sun_java2d_x11_X11SurfaceData_XSetForeground;
 
-		XRT_DrawGlyphList;
-
 		X11SurfaceData_GetOps;
 		Java_java_awt_Font_initIDs;
                 Java_sun_font_FontConfigManager_getFontConfig;
@@ -99,6 +97,7 @@
                 AWTCharAscent;
                 AWTCharDescent;
                 AWTDrawGlyphList;
+		XRT_DrawGlyphList;
                 AccelGlyphCache_RemoveAllCellInfos;
 
 
diff -Nru openjdk.orig/jdk/src/solaris/native/sun/java2d/x11/XRPMBlitLoops.c openjdk/jdk/src/solaris/native/sun/java2d/x11/XRPMBlitLoops.c
--- openjdk.orig/jdk/src/solaris/native/sun/java2d/x11/XRPMBlitLoops.c	2009-10-23 15:37:31.000000000 +0100
+++ openjdk/jdk/src/solaris/native/sun/java2d/x11/XRPMBlitLoops.c	2009-10-23 15:48:18.000000000 +0100
@@ -67,6 +67,9 @@
 
 
 #ifndef HEADLESS
+/**
+ * Prepares a mask-picture according to the mask and the properties currently set, which is used for composition later.
+ */
 static Picture prepareMaskPM(JNIEnv *env, jbyteArray maskArray, jint maskoff, jint maskscan, jint w, jint h)
 {
     int line, pix;
@@ -107,6 +110,7 @@
             }
           }
 
+          /*Upload Image to X-Server*/
           XPutImage(awt_display, maskBuffer->maskPixmap, maskBuffer->maskGC, img, 0, 0, 0, 0, w, h);
           (*env)->ReleasePrimitiveArrayCritical(env, maskArray, mask, JNI_ABORT);
 
@@ -124,6 +128,7 @@
         }
     }else
     {
+      /*If we haven't got a mask (full coverage), we have to return the EA mask, if EA is used*/
       if(useEA) {
         return xrSrcData.alphaMask;
       }
