diff -Nru openjdk.orig/jdk/make/sun/awt/mapfile-mawt-vers openjdk/jdk/make/sun/awt/mapfile-mawt-vers
--- openjdk.orig/jdk/make/sun/awt/mapfile-mawt-vers	2008-11-20 08:44:12.000000000 +0000
+++ openjdk/jdk/make/sun/awt/mapfile-mawt-vers	2008-12-09 19:24:36.000000000 +0000
@@ -417,6 +417,35 @@
                 Java_sun_java2d_x11_X11SurfaceData_XSetForeground;
                 Java_sun_java2d_x11_X11SurfaceData_XSetGraphicsExposures;
 
+		Java_sun_java2d_xr_XRSurfaceData_flushNativeSurface;
+		Java_sun_java2d_xr_XRSurfaceData_XRIsDrawableValid;
+		Java_sun_java2d_xr_XRSurfaceData_setInvalid;
+		Java_sun_java2d_xr_XRSurfaceData_XRInitXRender;
+		Java_sun_java2d_xr_XRSurfaceData_initOps;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetForeground;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetComposite;
+		Java_sun_java2d_xr_XRSurfaceData_initIDs;
+		Java_sun_java2d_xr_XRSurfaceData_XRInitSurface;
+		Java_sun_java2d_xr_XRSurfaceData_XRResetClip;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetClip;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetTransformNative;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetFilter;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetTexturePaint;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetGradientPaint;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetLinearGradientPaint;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetRadialGradientPaint;
+		Java_sun_java2d_xr_XRSurfaceData_XRResetPaint;
+		Java_sun_java2d_xr_XRSurfaceData_XRSetRepeat;
+		Java_sun_java2d_xr_XRPMBlitLoops_nativeRenderBlit;
+		Java_sun_java2d_xr_XRRenderer_XRFillRect;
+		Java_sun_java2d_xr_XRRenderer_XRFillSpans;
+		Java_sun_java2d_xr_XRRenderer_XRDoPath;
+		Java_sun_java2d_xr_XRRenderer_XRDrawLine;
+        	Java_sun_java2d_xr_XRRenderer_devCopyArea;
+		Java_sun_java2d_xr_XRMaskFill_maskFill;
+		Java_sun_java2d_xr_XRMaskBlit_maskBlit;
+ 		XRT_DrawGlyphList;
+
                 Java_sun_java2d_opengl_OGLContext_getOGLIdString;
                 Java_sun_java2d_opengl_OGLMaskFill_maskFill;
                 Java_sun_java2d_opengl_OGLRenderer_drawPoly;
diff -Nru openjdk.orig/jdk/make/sun/headless/mapfile-vers openjdk/jdk/make/sun/headless/mapfile-vers
--- openjdk.orig/jdk/make/sun/headless/mapfile-vers	2008-11-20 08:44:12.000000000 +0000
+++ openjdk/jdk/make/sun/headless/mapfile-vers	2008-12-09 19:23:34.000000000 +0000
@@ -57,6 +57,8 @@
                 Java_sun_java2d_x11_X11SurfaceData_XSetXorMode;
                 Java_sun_java2d_x11_X11SurfaceData_XSetForeground;
 
+		XRT_DrawGlyphList;
+
 		X11SurfaceData_GetOps;
 		Java_java_awt_Font_initIDs;
                 Java_sun_font_FontManager_getFontConfig;
diff -Nru openjdk.orig/jdk/make/sun/xawt/mapfile-vers openjdk/jdk/make/sun/xawt/mapfile-vers
--- openjdk.orig/jdk/make/sun/xawt/mapfile-vers	2008-12-09 19:19:29.000000000 +0000
+++ openjdk/jdk/make/sun/xawt/mapfile-vers	2008-12-09 19:23:34.000000000 +0000
@@ -358,6 +358,7 @@
         Java_sun_java2d_xr_XRSurfaceData_XRSetComposite;
         Java_sun_java2d_xr_XRSurfaceData_initIDs;
         Java_sun_java2d_xr_XRSurfaceData_XRInitSurface;
+	Java_sun_java2d_xr_XRSurfaceData_XRSetClip;
 	Java_sun_java2d_xr_XRSurfaceData_XRResetClip;
 	Java_sun_java2d_xr_XRSurfaceData_XRSetTransformNative;
         Java_sun_java2d_xr_XRSurfaceData_XRSetFilter;
@@ -372,6 +373,7 @@
 	Java_sun_java2d_xr_XRRenderer_XRFillSpans;
         Java_sun_java2d_xr_XRRenderer_XRDoPath;
 	Java_sun_java2d_xr_XRRenderer_XRDrawLine;
+        Java_sun_java2d_xr_XRRenderer_devCopyArea;
         Java_sun_java2d_xr_XRMaskFill_maskFill;
 	Java_sun_java2d_xr_XRMaskBlit_maskBlit;
 	XRT_DrawGlyphList;
