diff --git a/make/common/Rules.gmk b/make/common/Rules.gmk
--- openjdk-boot/corba/make/common/Rules.gmk
+++ openjdk-boot/corba/make/common/Rules.gmk
@@ -184,7 +184,9 @@
 	@$(MKDIR) -p $(CLASSDESTDIR)
 	if [ -s $(JAVA_SOURCE_LIST) ] ; then \
 	  $(CAT) $(JAVA_SOURCE_LIST); \
-	  $(JAVAC_CMD) $(JAVAC_PREFER_SOURCE) -sourcepath "$(SOURCEPATH)" -d $(CLASSDESTDIR) @$(JAVA_SOURCE_LIST); \
+	  $(JAVAC_CMD) $(JAVAC_PREFER_SOURCE) \
+	    -sourcepath "$(SOURCEPATH):$(JDK_TOPDIR)/src/share/classes:$(JDK_TOPDIR)/src/solaris/classes:$(LANGTOOLS_TOPDIR)/src/share/classes" \
+	    -d $(CLASSDESTDIR) @$(JAVA_SOURCE_LIST); \
 	fi
 	@$(java-vm-cleanup)
 
diff --git a/make/Defs-internal.gmk b/make/Defs-internal.gmk
--- openjdk-boot.orig/make/Defs-internal.gmk
+++ openjdk-boot/make/Defs-internal.gmk
@@ -305,6 +305,7 @@
 
 # Common make arguments (supplied to all component builds)
 COMMON_BUILD_ARGUMENTS = \
+    LANGTOOLS_TOPDIR=$(ABS_LANGTOOLS_TOPDIR) \
     JDK_TOPDIR=$(ABS_JDK_TOPDIR) \
     JDK_MAKE_SHARED_DIR=$(ABS_JDK_TOPDIR)/make/common/shared \
     EXTERNALSANITYCONTROL=true \
