diff -Nru openjdk-boot.orig/corba/make/common/Defs-linux.gmk openjdk-boot/corba/make/common/Defs-linux.gmk
--- openjdk-boot.orig/corba/make/common/Defs-linux.gmk	2009-12-13 23:10:09.000000000 +0000
+++ openjdk-boot/corba/make/common/Defs-linux.gmk	2009-12-14 15:54:17.000000000 +0000
@@ -271,7 +271,7 @@
 #
 VM_NAME         = server
 JVMLIB          = -L$(BOOTDIR)/jre/lib/$(LIBARCH)/$(VM_NAME) -ljvm
-JAVALIB         = -L$(BOOTDIR)/jre/lib/$(LIBARCH) -ljava $(JVMLIB)
+JAVALIB         = -L$(BOOTDIR)/jre/lib/$(LIBARCH) -L$(dir $(shell $(if $(GCJ),$(GCJ),$(subst gcc,gcj,$(CC))) --print-file-name=libgcj.so)) -lgcj $(JVMLIB)
 
 #
 # We want to privatize JVM symbols on Solaris. This is so the user can
