diff -Nru ../openjdk.orig/openjdk-boot/corba/make/common/Defs.gmk openjdk-boot/corba/make/common/Defs.gmk
--- ../openjdk.orig/openjdk-boot/corba/make/common/Defs.gmk	2009-10-03 00:02:43.000000000 +0100
+++ openjdk-boot/corba/make/common/Defs.gmk	2009-10-30 16:59:16.000000000 +0000
@@ -53,6 +53,13 @@
 
 _OUTPUTDIR=$(TOPDIR)/build/$(PLATFORM)-$(ARCH)
 
+
+ifdef ALT_LANGTOOLS_DIST
+  LANGTOOLS_DIST :=$(call FullPath,$(ALT_LANGTOOLS_DIST))
+else
+  LANGTOOLS_DIST =
+endif 
+
 #
 # Get platform definitions
 #
@@ -129,8 +136,6 @@
 # for build tool jar files
 BUILDTOOLJARDIR     = $(OUTPUTDIR)/btjars
 ABS_BUILDTOOLJARDIR = $(ABS_OUTPUTDIR)/btjars
-# for generated java source files
-GENSRCDIR           = $(OUTPUTDIR)/gensrc
 # for generated C source files (not javah)
 GENNATIVESRCDIR     = $(OUTPUTDIR)/gennativesrc
 # for imported source files
@@ -231,7 +236,7 @@
 # We define an intermediate variable for Java files because
 # we use its value later to help define $SOURCEPATH
 
-VPATH0.java = $(GENSRCDIR)$(CLASSPATH_SEPARATOR)$(PLATFORM_SRC)/classes$(CLASSPATH_SEPARATOR)$(SHARE_SRC)/classes
+VPATH0.java = $(GENSRCDIR)$(CLASSPATH_SEPARATOR)$(SHARE_SRC)/classes
 VPATH.java = $(VPATH0.java)
 vpath %.java $(VPATH.java)
 vpath %.class $(CLASSBINDIR)
