diff -Nru openjdk-boot.orig/corba/make/common/Defs.gmk openjdk-boot/corba/make/common/Defs.gmk
--- openjdk-boot.orig/corba/make/common/Defs.gmk	2011-03-08 18:49:48.566665532 +0000
+++ openjdk-boot/corba/make/common/Defs.gmk	2011-03-08 18:47:51.734115218 +0000
@@ -50,6 +50,9 @@
 ifndef SHARE_SRC
   SHARE_SRC    = $(TOPDIR)/src/share
 endif
+ifndef JDK_SRC
+  JDK_SRC = $(JDK_TOPDIR)/src/share
+endif
 
 _OUTPUTDIR=$(TOPDIR)/build/$(PLATFORM)-$(ARCH)
 
@@ -172,7 +175,7 @@
 # We define an intermediate variable for Java files because
 # we use its value later to help define $SOURCEPATH
 
-VPATH0.java = $(GENSRCDIR)$(CLASSPATH_SEPARATOR)$(SHARE_SRC)/classes
+VPATH0.java = $(GENSRCDIR)$(CLASSPATH_SEPARATOR)$(SHARE_SRC)/classes$(CLASSPATH_SEPARATOR)$(JDK_SRC)/classes
 VPATH.java = $(VPATH0.java)
 vpath %.java $(VPATH.java)
 vpath %.class $(CLASSBINDIR)
diff -Nru openjdk-boot.orig/corba/make/common/Rules.gmk openjdk-boot/corba/make/common/Rules.gmk
--- openjdk-boot.orig/corba/make/common/Rules.gmk	2011-03-08 18:48:19.953282134 +0000
+++ openjdk-boot/corba/make/common/Rules.gmk	2011-03-08 18:48:05.453710200 +0000
@@ -184,7 +184,12 @@
 	@$(MKDIR) -p $(CLASSDESTDIR)
 	if [ -s $(JAVA_SOURCE_LIST) ] ; then \
 	  $(CAT) $(JAVA_SOURCE_LIST); \
-	  $(JAVAC_CMD) $(JAVAC_PREFER_SOURCE) -sourcepath "$(SOURCEPATH)" -d $(CLASSDESTDIR) @$(JAVA_SOURCE_LIST); \
+ 	  $(ECHO) $(JAVAC_CMD) -classpath $(ABS_OUTPUTDIR)/corba/classes \
+	  	 -classpath $(LANGTOOLS_DIST)/lib/classes.jar \
+ 		 -sourcepath "$(SOURCEPATH)" -d $(CLASSDESTDIR) @$(JAVA_SOURCE_LIST); \
+ 	  $(JAVAC_CMD) -classpath $(ABS_OUTPUTDIR)/corba/classes \
+	  	 -classpath $(LANGTOOLS_DIST)/lib/classes.jar \
+ 		 -sourcepath "$(SOURCEPATH)" -d $(CLASSDESTDIR) @$(JAVA_SOURCE_LIST); \
 	fi
 	@$(java-vm-cleanup)
 
