diff -Nru openjdk.orig/jdk/make/common/Release.gmk openjdk/jdk/make/common/Release.gmk
--- openjdk-boot.orig/jdk/make/common/Release.gmk	2011-03-16 20:59:31.000000000 +0000
+++ openjdk-boot/jdk/make/common/Release.gmk	2011-03-16 21:29:40.269191529 +0000
@@ -1071,16 +1071,6 @@
 	$(CP) $(PLATFORM_SRC)/bin/java_md.h $(JDK_IMAGE_DIR)/src/launcher
 	$(CD) $(JDK_IMAGE_DIR)/src && $(ZIPEXE) -qr ../src.zip *
 	$(RM) -r $(JDK_IMAGE_DIR)/src
-	@#
-	@# demo, include
-	@#
-	$(CP) -r -f $(DEMODIR) $(JDK_IMAGE_DIR)
-	$(CP) -r -f $(SAMPLEDIR) $(JDK_IMAGE_DIR)
-	$(CP) -r $(INCLUDEDIR) $(JDK_IMAGE_DIR)
-	@#
-	@# Swing BeanInfo generation
-	@#
-	$(CD) javax/swing/beaninfo && $(MAKE) swing-1.2-beans
 ifneq ($(PLATFORM), windows)
 	$(call copy-man-pages,$(JDK_IMAGE_DIR),$(JDK_MAN_PAGES))
 endif # !windows
diff -Nru openjdk.orig/jdk/make/Makefile openjdk/jdk/make/Makefile
--- openjdk-boot.orig/jdk/make/Makefile	2011-03-14 22:10:30.000000000 +0000
+++ openjdk-boot/jdk/make/Makefile	2011-03-16 21:29:40.269191529 +0000
@@ -238,7 +238,7 @@
 
 SUBDIRS       = tools java javax sun com
 SUBDIRS_tools = launchers
-SUBDIRS_misc  = org sunw jpda mkdemo mksample
+SUBDIRS_misc  = org sunw jpda mksample
 
 # Alternate classes implementation
 ifndef OPENJDK
