diff -Nru openjdk-boot.orig/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java openjdk-boot/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java
--- openjdk-boot.orig/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java	2021-07-30 02:02:25.719553968 +0100
+++ openjdk-boot/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java	2021-07-30 20:03:08.296144222 +0100
@@ -495,8 +495,8 @@
                             // available
                             ldr.wait(readTimeout - elapsedMilli);
                             elapsedNano += (System.nanoTime() - beginNano);
-                            elapsedMilli += elapsedNano / 1000_000;
-                            elapsedNano %= 1000_000;
+                            elapsedMilli += elapsedNano / 1000000;
+                            elapsedNano %= 1000000;
 
                         } else {
                             // no timeout is set so we wait infinitely until
diff -Nru openjdk-boot.orig/jdk/src/share/classes/sun/net/ftp/impl/FtpClient.java openjdk-boot/jdk/src/share/classes/sun/net/ftp/impl/FtpClient.java
--- openjdk-boot.orig/jdk/src/share/classes/sun/net/ftp/impl/FtpClient.java	2021-07-28 00:49:02.000000000 +0100
+++ openjdk-boot/jdk/src/share/classes/sun/net/ftp/impl/FtpClient.java	2021-07-30 20:03:51.393203284 +0100
@@ -140,8 +140,8 @@
 
                     public Object run() {
                         acceptPasvAddress[0] = System.getProperty("jdk.net.ftp.trustPasvAddress", "false");
-                        vals[0] = Integer.getInteger("sun.net.client.defaultReadTimeout", 300_000).intValue();
-                        vals[1] = Integer.getInteger("sun.net.client.defaultConnectTimeout", 300_000).intValue();
+                        vals[0] = Integer.getInteger("sun.net.client.defaultReadTimeout", 300000).intValue();
+                        vals[1] = Integer.getInteger("sun.net.client.defaultConnectTimeout", 300000).intValue();
                         encs[0] = System.getProperty("file.encoding", "ISO8859_1");
                         return null;
                     }
diff -Nru openjdk-boot.orig/jdk/src/share/classes/sun/security/util/Parsing.java openjdk-boot/jdk/src/share/classes/sun/security/util/Parsing.java
--- openjdk-boot.orig/jdk/src/share/classes/sun/security/util/Parsing.java	2021-07-28 00:49:02.000000000 +0100
+++ openjdk-boot/jdk/src/share/classes/sun/security/util/Parsing.java	2021-07-30 20:03:08.296144222 +0100
@@ -93,7 +93,7 @@
                     return Integer.parseInt(s, radix);
                 } else {
                     long ell = Long.parseLong(s, radix);
-                    if ((ell & 0xffff_ffff_0000_0000L) == 0) {
+                    if ((ell & 0xffffffff00000000L) == 0) {
                         return (int) ell;
                     } else {
                         throw new
