diff -Nru openjdk-boot.orig/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java openjdk-boot/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java
--- openjdk-boot.orig/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java	2017-02-13 03:45:45.000000000 +0000
+++ openjdk-boot/jdk/src/share/classes/com/sun/jndi/ldap/Connection.java	2017-02-13 04:19:50.459572606 +0000
@@ -466,8 +466,8 @@
                             // available
                             ldr.wait(readTimeout - elapsedMilli);
                             elapsedNano += (System.nanoTime() - beginNano);
-                            elapsedMilli += elapsedNano / 1000_000;
-                            elapsedNano %= 1000_000;
+                            elapsedMilli += elapsedNano / 1000000;
+                            elapsedNano %= 1000000;
 
                         } else {
                             // no timeout is set so we wait infinitely until
diff -Nru openjdk-boot.orig/jdk/src/share/classes/sun/security/util/Parsing.java openjdk-boot/jdk/src/share/classes/sun/security/util/Parsing.java
--- openjdk-boot.orig/jdk/src/share/classes/sun/security/util/Parsing.java	2017-02-13 04:21:58.181446482 +0000
+++ openjdk-boot/jdk/src/share/classes/sun/security/util/Parsing.java	2017-02-13 04:22:10.617239537 +0000
@@ -93,7 +93,7 @@
                     return Integer.parseInt(s, radix);
                 } else {
                     long ell = Long.parseLong(s, radix);
-                    if ((ell & 0xffff_ffff_0000_0000L) == 0) {
+                    if ((ell & 0xffffffff00000000L) == 0) {
                         return (int) ell;
                     } else {
                         throw new
--- openjdk-boot.orig/jdk/src/share/classes/sun/net/ftp/impl/FtpClient.java	2017-12-05 04:44:13.171649999 +0000
+++ openjdk-boot/jdk/src/share/classes/sun/net/ftp/impl/FtpClient.java	2017-12-05 04:44:27.047426865 +0000
@@ -117,8 +117,8 @@
                 new PrivilegedAction<Object>() {
 
                     public Object run() {
-                        vals[0] = Integer.getInteger("sun.net.client.defaultReadTimeout", 300_000).intValue();
-                        vals[1] = Integer.getInteger("sun.net.client.defaultConnectTimeout", 300_000).intValue();
+                        vals[0] = Integer.getInteger("sun.net.client.defaultReadTimeout", 300000).intValue();
+                        vals[1] = Integer.getInteger("sun.net.client.defaultConnectTimeout", 300000).intValue();
                         encs[0] = System.getProperty("file.encoding", "ISO8859_1");
                         return null;
                     }
