diff -Nru openjdk-boot.orig/jdk/make/sun/awt/Makefile openjdk-boot/jdk/make/sun/awt/Makefile
--- openjdk-boot.orig/jdk/make/sun/awt/Makefile	2011-04-20 04:40:21.000000000 +0100
+++ openjdk-boot/jdk/make/sun/awt/Makefile	2011-04-21 23:16:13.427471369 +0100
@@ -400,10 +400,9 @@
 
 COMPILEFONTCONFIG_JARFILE = $(BUILDTOOLJARDIR)/compilefontconfig.jar
 
-$(LIBDIR)/%.bfc: $(FONTCONFIGS_SRC)/$(FONTCONFIGS_SRC_PREFIX)%.properties \
-    $(COMPILEFONTCONFIG_JARFILE)
+$(LIBDIR)/%.bfc: $(GENSRCDIR)/fontconfig/%.bfc
 	$(prep-target)
-	$(BOOT_JAVA_CMD) -jar $(COMPILEFONTCONFIG_JARFILE) $(COMPILEFONTCONFIG_FLAGS) $< $@
+	$(install-file)
 	$(call chmod-file, 444)
 	@$(java-vm-cleanup)
 
