diff -Nru ../openjdk.orig/openjdk-boot/jdk/make/sun/awt/Makefile openjdk-boot/jdk/make/sun/awt/Makefile
--- ../openjdk.orig/openjdk-boot/jdk/make/sun/awt/Makefile	2009-05-08 16:10:44.000000000 +0100
+++ openjdk-boot/jdk/make/sun/awt/Makefile	2009-10-30 16:59:16.000000000 +0000
@@ -401,10 +401,9 @@
 COMPILEFONTCONFIG_JARFILE = $(BUILDTOOLJARDIR)/compilefontconfig.jar
 
 $(LIBDIR)/%.bfc: $(FONTCONFIGS_SRC)/$(FONTCONFIGS_SRC_PREFIX)%.properties \
-    $(COMPILEFONTCONFIG_JARFILE)
-	$(prep-target)
-	$(BOOT_JAVA_CMD) -jar $(COMPILEFONTCONFIG_JARFILE) $< $@
-	$(CHMOD) 444 $(@)
+	$(COMPILEFONTCONFIG_JARFILE)
+	$(CP) -f $(GENSRCDIR)/fontconfig/* $(LIBDIR)
+	$(CHMOD) 444 $(LIBDIR)/*.bfc
 	@$(java-vm-cleanup)
 
 fontconfigs.clean :
