diff -Nru openjdk-boot.orig/jdk/make/sun/awt/Makefile openjdk-boot/jdk/make/sun/awt/Makefile
--- openjdk-boot.orig/jdk/make/sun/awt/Makefile	2011-04-11 12:34:30.000000000 +0100
+++ openjdk-boot/jdk/make/sun/awt/Makefile	2011-04-11 16:51:37.423839968 +0100
@@ -401,11 +401,9 @@
 
 COMPILEFONTCONFIG_JARFILE = $(BUILDTOOLJARDIR)/compilefontconfig.jar
 
-$(LIBDIR)/%.bfc: $(FONTCONFIGS_SRC)/$(FONTCONFIGS_SRC_PREFIX)%.properties \
-    $(COMPILEFONTCONFIG_JARFILE)
+$(LIBDIR)/%.bfc: $(GENSRCDIR)/fontconfig/%.bfc
 	$(prep-target)
-	$(BOOT_JAVA_CMD) -jar $(COMPILEFONTCONFIG_JARFILE) $(COMPILEFONTCONFIG_FLAGS) $< $@
-	$(install-module-file)
+	$(install-file)
 	$(call chmod-file, 444)
 	@$(java-vm-cleanup)
 
