diff -Nru openjdk-boot.orig/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c openjdk-boot/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c
--- openjdk-boot.orig/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c	2009-03-30 17:23:08.000000000 +0100
+++ openjdk-boot/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c	2009-05-18 19:58:36.000000000 +0100
@@ -192,9 +192,9 @@
     lockResult = fcntl(fd, cmd, &fl);
     if (lockResult < 0) {
         if ((cmd == F_SETLK64) && (errno == EAGAIN))
-            return sun_nio_ch_FileDispatcherImpl_NO_LOCK;
+            return sun_nio_ch_FileDispatcher_NO_LOCK;
         if (errno == EINTR)
-            return sun_nio_ch_FileDispatcherImpl_INTERRUPTED;
+            return sun_nio_ch_FileDispatcher_INTERRUPTED;
         JNU_ThrowIOExceptionWithLastError(env, "Lock failed");
     }
     return 0;
