--- openjdk-boot.orig/jdk/make/tools/src/build/tools/charsetmapping/EUC_TW.java	2010-05-05 18:45:07.000000000 +0100
+++ openjdk-boot/jdk/make/tools/src/build/tools/charsetmapping/EUC_TW.java	2010-05-05 21:26:31.000000000 +0100
@@ -83,8 +83,8 @@
     static void genClass(String args[]) throws Exception
     {
         InputStream is = new FileInputStream(new File(args[0], "euc_tw.map"));
-        PrintStream ps = new PrintStream(new File(args[1], "EUC_TWMapping.java"),
-                                         "ISO-8859-1");
+        FileOutputStream fos = new FileOutputStream(new File(args[1], "EUC_TWMapping.java"));
+        PrintStream ps = new PrintStream(fos, false, "ISO-8859-1");
         String copyright = getCopyright(new File(args[3]));
 
 
