diff -Nru openjdk-boot.orig/jdk/make/java/Makefile openjdk-boot/jdk/make/java/Makefile
--- openjdk-boot.orig/jdk/make/java/Makefile	2010-04-28 22:05:26.000000000 +0100
+++ openjdk-boot/jdk/make/java/Makefile	2010-05-04 14:02:16.000000000 +0100
@@ -44,7 +44,7 @@
 SUBDIRS_desktop    = awt applet beans
 SUBDIRS_management = management
 SUBDIRS_misc       = npt java_crw_demo java_hprof_demo \
-                     logging instrument dyn sql rmi
+                     logging instrument sql rmi
 
 
 ifeq ($(PLATFORM), solaris)
