diff -Nru openjdk-boot.orig/jdk/make/common/Release.gmk openjdk-boot/jdk/make/common/Release.gmk
--- openjdk-boot.orig/jdk/make/common/Release.gmk	2011-06-13 16:08:46.000000000 +0100
+++ openjdk-boot/jdk/make/common/Release.gmk	2011-06-13 17:06:49.515488800 +0100
@@ -627,9 +627,19 @@
 RESOURCES_JAR=$(ABS_TEMPDIR)/resources-orig.jar
 $(RESOURCES_JAR): $(RES_JAR_FILELIST) $(OTHER_JAR_MANIFEST_FILE)
 	$(prep-target)
+ifneq (,$(JAR_KNOWS_ATFILE))
 	$(CD) $(CLASSBINDIR) && \
 	    $(BOOT_JAR_CMD) $(CREATE_JAR_OPTS) $(OTHER_JAR_MANIFEST_FILE) $@ \
 	        @$(RES_JAR_FILELIST) $(BOOT_JAR_JFLAGS)
+else ifneq (,$(JAR_ACCEPTS_STDIN_LIST))
+	(cd $(CLASSBINDIR) && $(CAT) $(RES_JAR_FILELIST) | \
+	$(BOOT_JAR_CMD) $(CREATE_JAR_OPTS)@ $(OTHER_JAR_MANIFEST_FILE) $@ \
+	    $(BOOT_JAR_JFLAGS))
+else
+	(cd $(CLASSBINDIR) && \
+	$(BOOT_JAR_CMD) $(CREATE_JAR_OPTS) $(OTHER_JAR_MANIFEST_FILE) $@ \
+	    $(BOOT_JAR_JFLAGS) `cat $(RES_JAR_FILELIST)`)
+endif
 	@$(CD) $(CLASSBINDIR) && $(java-vm-cleanup)
 
 # Create jsse.jar containing SunJSSE implementation classes
@@ -683,9 +693,19 @@
 RT_JAR=$(ABS_TEMPDIR)/rt-orig.jar
 $(RT_JAR): $(RT_JAR_FILELIST) $(RT_JAR_MANIFEST_FILE)
 	$(prep-target)
+ifneq (,$(JAR_KNOWS_ATFILE))
 	$(CD) $(CLASSBINDIR) && \
 	    $(BOOT_JAR_CMD) $(CREATE_JAR_OPTS) $(RT_JAR_MANIFEST_FILE) $@ \
 	        @$(RT_JAR_FILELIST) $(BOOT_JAR_JFLAGS)
+else ifneq (,$(JAR_ACCEPTS_STDIN_LIST))
+	$(CD) $(CLASSBINDIR) && $(CAT) $(RT_JAR_FILELIST) | \
+	    $(BOOT_JAR_CMD) $(CREATE_JAR_OPTS)@ $(RT_JAR_MANIFEST_FILE) $@ \
+	        $(BOOT_JAR_JFLAGS)
+else
+	$(CD) $(CLASSBINDIR) && \
+	    $(BOOT_JAR_CMD) $(CREATE_JAR_OPTS) $(RT_JAR_MANIFEST_FILE) $@ \
+	        $(BOOT_JAR_JFLAGS) `cat $(RT_JAR_FILELIST)`
+endif
 	@$(CD) $(CLASSBINDIR) && $(java-vm-cleanup)
 
 # Meta-index construction to make core class loaders lazier
