diff -Nru ../openjdk.orig/openjdk-boot/jdk/make/common/Release.gmk openjdk-boot/jdk/make/common/Release.gmk
--- ../openjdk.orig/openjdk-boot/jdk/make/common/Release.gmk	2009-10-30 16:49:37.000000000 +0000
+++ openjdk-boot/jdk/make/common/Release.gmk	2009-10-30 16:59:16.000000000 +0000
@@ -650,10 +650,21 @@
 RESOURCES_JAR=$(ABS_TEMPDIR)/resources-orig.jar
 $(RESOURCES_JAR): $(RES_JAR_FILELIST) $(JAR_MANIFEST_FILE)
 	$(prep-target)
+ifneq (,$(JAR_KNOWS_ATFILE))
 	$(CD) $(CLASSBINDIR) && \
 	    $(BOOT_JAR_CMD) c0mf $(JAR_MANIFEST_FILE) $@ \
 	        @$(RES_JAR_FILELIST) $(BOOT_JAR_JFLAGS)
-	@$(CD) $(CLASSBINDIR) && $(java-vm-cleanup)
+	@$(CD) $(CLASSBINDIR)
+else ifneq (,$(JAR_ACCEPTS_STDIN_LIST))
+	(cd $(CLASSBINDIR) && $(CAT) $(RES_JAR_FILELIST) | \
+	$(BOOT_JAR_CMD) c0mf@ $(JAR_MANIFEST_FILE) $@ \
+	    $(BOOT_JAR_JFLAGS))
+else
+	(cd $(CLASSBINDIR) && \
+	$(BOOT_JAR_CMD) c0mf $(JAR_MANIFEST_FILE) $@ \
+	    $(BOOT_JAR_JFLAGS) `cat $(RES_JAR_FILELIST)`)
+endif
+	$(java-vm-cleanup)
 
 # Create jsse.jar containing SunJSSE implementation classes
 JSSE_JAR=$(ABS_TEMPDIR)/jsse-orig.jar
@@ -711,9 +722,19 @@
 RT_JAR=$(ABS_TEMPDIR)/rt-orig.jar
 $(RT_JAR): $(RT_JAR_FILELIST) $(JAR_MANIFEST_FILE)
 	$(prep-target)
+ifneq (,$(JAR_KNOWS_ATFILE))
 	$(CD) $(CLASSBINDIR) && \
 	    $(BOOT_JAR_CMD) c0mf $(JAR_MANIFEST_FILE) $@ \
 	        @$(RT_JAR_FILELIST) $(BOOT_JAR_JFLAGS)
+else ifneq (,$(JAR_ACCEPTS_STDIN_LIST))
+	$(CD) $(CLASSBINDIR) && $(CAT) $(RT_JAR_FILELIST) | \
+	    $(BOOT_JAR_CMD) c0mf@ $(JAR_MANIFEST_FILE) $@ \
+	        $(BOOT_JAR_JFLAGS)
+else
+	$(CD) $(CLASSBINDIR) && \
+	    $(BOOT_JAR_CMD) c0mf $(JAR_MANIFEST_FILE) $@ \
+	        $(BOOT_JAR_JFLAGS) `cat $(RT_JAR_FILELIST)`
+endif
 	@$(CD) $(CLASSBINDIR) && $(java-vm-cleanup)
 
 # Meta-index construction to make core class loaders lazier
--- openjdk-boot/jdk/make/modules/Makefile.orig	2010-06-13 10:37:13.000000000 +0200
+++ openjdk-boot/jdk/make/modules/Makefile	2010-06-13 10:42:37.000000000 +0200
@@ -111,10 +111,21 @@
            fi ; \
            $(MKDIR) -p $(ABS_MODULES_DIR)/$$m/lib; \
 	   $(CD) $(MODULE_CLASSES) && \
+ifneq (,$(JAR_KNOWS_ATFILE))
                $(BOOT_JAR_CMD) c0mf $(MODULE_JAR_MANIFEST_FILE) \
 		   $(ABS_MODULES_DIR)/$$m/lib/$$m.jar \
 	           @$(TMP)/tmp.cf \
                    $(BOOT_JAR_JFLAGS); \
+else ifneq (,$(JAR_ACCEPTS_STDIN_LIST))
+               $(CAT) $(TMP)/tmp.cf | \
+               $(BOOT_JAR_CMD) c0mf@ $(MODULE_JAR_MANIFEST_FILE) \
+		   $(ABS_MODULES_DIR)/$$m/lib/$$m.jar \
+                   $(BOOT_JAR_JFLAGS); \
+else
+               $(BOOT_JAR_CMD) c0mf $(MODULE_JAR_MANIFEST_FILE) \
+		   $(ABS_MODULES_DIR)/$$m/lib/$$m.jar \
+                   $(BOOT_JAR_JFLAGS) `cat $(TMP)/tmp.cf`; \
+endif
 	   for s in `$(GREP) "^$$m" $(MODULES_LIST)` ; do \
                if [ -d $(TMP)/$$s ] ; then \
                	   $(CP) -rf $(TMP)/$$s/*  $(ABS_MODULES_DIR)/$$m; \
