diff -Nru openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_encoding.jmk openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_encoding.jmk
--- openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_encoding.jmk	2010-07-29 21:54:09.000000000 +0100
+++ openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_encoding.jmk	2011-03-16 20:45:33.914471077 +0000
@@ -62,7 +62,6 @@
 	com/sun/corba/se/impl/encoding/TypeCodeInputStream.java \
 	com/sun/corba/se/impl/encoding/TypeCodeOutputStream.java \
 	com/sun/corba/se/impl/encoding/TypeCodeReader.java \
-	com/sun/corba/se/impl/encoding/WrapperInputStream.java
-
-
-
+	com/sun/corba/se/impl/encoding/WrapperInputStream.java \
+	com/sun/corba/se/impl/encoding/IDLJavaSerializationInputStream.java \
+	com/sun/corba/se/impl/encoding/IDLJavaSerializationOutputStream.java
diff -Nru openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_ior.jmk openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_ior.jmk
--- openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_ior.jmk	2010-07-29 21:54:09.000000000 +0100
+++ openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_ior.jmk	2011-03-16 20:45:33.922471207 +0000
@@ -66,4 +66,5 @@
 	com/sun/corba/se/impl/ior/iiop/JavaSerializationComponent.java \
 	com/sun/corba/se/impl/ior/iiop/MaxStreamFormatVersionComponentImpl.java \
 	com/sun/corba/se/impl/ior/iiop/RequestPartitioningComponentImpl.java \
-	com/sun/corba/se/impl/ior/iiop/ORBTypeComponentImpl.java
+	com/sun/corba/se/impl/ior/iiop/ORBTypeComponentImpl.java \
+	com/sun/corba/se/impl/ior/iiop/JavaSerializationComponent.java
diff -Nru openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_orbutil.jmk openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_orbutil.jmk
--- openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_orbutil.jmk	2010-07-29 21:54:09.000000000 +0100
+++ openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_orbutil.jmk	2011-03-16 20:45:33.922471207 +0000
@@ -81,4 +81,5 @@
 	com/sun/corba/se/impl/orbutil/graph/Node.java \
 	com/sun/corba/se/impl/orbutil/graph/NodeData.java \
 	com/sun/corba/se/impl/orbutil/graph/Graph.java \
-	com/sun/corba/se/impl/orbutil/graph/GraphImpl.java 
+	com/sun/corba/se/impl/orbutil/graph/GraphImpl.java \
+	com/sun/corba/se/impl/orbutil/GetPropertyAction.java
diff -Nru openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_protocol.jmk openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_protocol.jmk
--- openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_impl_protocol.jmk	2010-07-29 21:54:09.000000000 +0100
+++ openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_impl_protocol.jmk	2011-03-16 20:45:33.922471207 +0000
@@ -81,5 +81,5 @@
 	com/sun/corba/se/impl/protocol/giopmsgheaders/RequestMessage_1_1.java \
 	com/sun/corba/se/impl/protocol/giopmsgheaders/RequestMessage_1_2.java \
 	com/sun/corba/se/impl/protocol/giopmsgheaders/TargetAddress.java \
-	com/sun/corba/se/impl/protocol/giopmsgheaders/TargetAddressHelper.java
-
+	com/sun/corba/se/impl/protocol/giopmsgheaders/TargetAddressHelper.java \
+	com/sun/corba/se/impl/protocol/giopmsgheaders/LocateReplyOrReplyMessage.java
diff -Nru openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_spi_monitoring.jmk openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_spi_monitoring.jmk
--- openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_spi_monitoring.jmk	2010-07-29 21:54:09.000000000 +0100
+++ openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_spi_monitoring.jmk	2011-03-16 20:45:33.922471207 +0000
@@ -30,6 +30,7 @@
 	com/sun/corba/se/spi/monitoring/StringMonitoredAttributeBase.java \
 	com/sun/corba/se/spi/monitoring/LongMonitoredAttributeBase.java \
 	com/sun/corba/se/spi/monitoring/MonitoringFactories.java \
+	com/sun/corba/se/spi/monitoring/MonitoringConstants.java \
 	com/sun/corba/se/spi/monitoring/MonitoredAttributeInfo.java \
 	com/sun/corba/se/spi/monitoring/MonitoredObject.java \
 	com/sun/corba/se/spi/monitoring/MonitoredObjectFactory.java \
diff -Nru openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_spi_presentation_rmi.jmk openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_spi_presentation_rmi.jmk
--- openjdk-boot.origcorba/make/com/sun/corba/minclude/com_sun_corba_se_spi_presentation_rmi.jmk	2010-07-29 21:54:09.000000000 +0100
+++ openjdk-boot/corba/make/com/sun/corba/minclude/com_sun_corba_se_spi_presentation_rmi.jmk	2011-03-16 20:45:33.922471207 +0000
@@ -29,5 +29,6 @@
 	com/sun/corba/se/spi/presentation/rmi/IDLNameTranslator.java \
 	com/sun/corba/se/spi/presentation/rmi/PresentationDefaults.java \
 	com/sun/corba/se/spi/presentation/rmi/PresentationManager.java \
+	com/sun/corba/se/spi/presentation/rmi/PresentationDefaults.java \
 	com/sun/corba/se/spi/presentation/rmi/StubAdapter.java
 
diff -Nru openjdk-boot.origjdk/make/java/java/FILES_java.gmk openjdk-boot/jdk/make/java/java/FILES_java.gmk
--- openjdk-boot.origjdk/make/java/java/FILES_java.gmk	2011-03-14 22:10:30.000000000 +0000
+++ openjdk-boot/jdk/make/java/java/FILES_java.gmk	2011-03-16 20:45:33.922471207 +0000
@@ -29,6 +29,80 @@
 # will generate header files
 #
 JAVA_JAVA_java = \
+    java/lang/reflect/AccessibleObject.java \
+    java/lang/reflect/AnnotatedElement.java \
+    java/lang/reflect/Array.java \
+    java/lang/reflect/Constructor.java \
+    java/lang/reflect/Field.java \
+    java/lang/reflect/GenericArrayType.java \
+    java/lang/reflect/GenericDeclaration.java \
+    java/lang/reflect/GenericSignatureFormatError.java \
+    java/lang/reflect/InvocationHandler.java \
+    java/lang/reflect/InvocationTargetException.java \
+    java/lang/reflect/MalformedParameterizedTypeException.java \
+    java/lang/reflect/Member.java \
+    java/lang/reflect/Method.java \
+    java/lang/reflect/Modifier.java \
+    java/lang/reflect/package-info.java \
+    java/lang/reflect/ParameterizedType.java \
+    java/lang/reflect/Proxy.java \
+    java/lang/reflect/ReflectAccess.java \
+    java/lang/reflect/ReflectPermission.java \
+    java/lang/reflect/Type.java \
+    java/lang/reflect/TypeVariable.java \
+    java/lang/reflect/UndeclaredThrowableException.java \
+    java/lang/reflect/WildcardType.java \
+    java/lang/ref/Finalizer.java \
+    java/lang/ref/FinalReference.java \
+    java/lang/ref/PhantomReference.java \
+    java/lang/ref/Reference.java \
+    java/lang/ref/ReferenceQueue.java \
+    java/lang/ref/SoftReference.java \
+    java/lang/ref/WeakReference.java \
+    java/lang/management/ClassLoadingMXBean.java \
+    java/lang/management/CompilationMXBean.java \
+    java/lang/management/GarbageCollectorMXBean.java \
+    java/lang/management/LockInfo.java \
+    java/lang/management/ManagementFactory.java \
+    java/lang/management/ManagementPermission.java \
+    java/lang/management/MemoryManagerMXBean.java \
+    java/lang/management/MemoryMXBean.java \
+    java/lang/management/MemoryNotificationInfo.java \
+    java/lang/management/MemoryPoolMXBean.java \
+    java/lang/management/MemoryType.java \
+    java/lang/management/MemoryUsage.java \
+    java/lang/management/MonitorInfo.java \
+    java/lang/management/OperatingSystemMXBean.java \
+    java/lang/management/RuntimeMXBean.java \
+    java/lang/management/ThreadInfo.java \
+    java/lang/management/ThreadMXBean.java \
+    java/lang/instrument/ClassDefinition.java \
+    java/lang/instrument/ClassFileTransformer.java \
+    java/lang/instrument/IllegalClassFormatException.java \
+    java/lang/instrument/Instrumentation.java \
+    java/lang/instrument/UnmodifiableClassException.java \
+    java/lang/annotation/AnnotationFormatError.java \
+    java/lang/annotation/Annotation.java \
+    java/lang/annotation/AnnotationTypeMismatchException.java \
+    java/lang/annotation/Documented.java \
+    java/lang/annotation/ElementType.java \
+    java/lang/annotation/IncompleteAnnotationException.java \
+    java/lang/annotation/Inherited.java \
+    java/lang/annotation/package-info.java \
+    java/lang/annotation/Retention.java \
+    java/lang/annotation/RetentionPolicy.java \
+    java/lang/annotation/Target.java \
+    java/io/IOError.java \
+    java/lang/Deprecated.java \
+    java/lang/Iterable.java \
+    java/util/RandomAccess.java \
+    java/lang/CharacterData00.java \
+    java/lang/CharacterData01.java \
+    java/lang/CharacterData02.java \
+    java/lang/CharacterData0E.java \
+    java/lang/CharacterDataLatin1.java \
+    java/lang/CharacterDataPrivateUse.java \
+    java/lang/CharacterDataUndefined.java \
     java/lang/Object.java \
     java/lang/AutoCloseable.java \
     java/lang/Class.java \
@@ -139,13 +213,6 @@
     java/lang/Override.java \
     java/lang/SafeVarargs.java \
     java/lang/SuppressWarnings.java \
-    java/lang/ref/Reference.java \
-        java/lang/ref/SoftReference.java \
-	java/lang/ref/WeakReference.java \
-	java/lang/ref/FinalReference.java \
-	java/lang/ref/PhantomReference.java \
-    java/lang/ref/ReferenceQueue.java \
-    java/lang/ref/Finalizer.java \
     java/util/BitSet.java \
     java/util/Calendar.java \
         java/util/GregorianCalendar.java \
