--- openjdk-boot.orig/jaxws/src/share/jaxws_classes/com/sun/xml/internal/bind/v2/runtime/unmarshaller/StAXEventConnector.java	2009-11-17 16:37:06.000000000 +0000
+++ openjdk-boot/jaxws/src/share/jaxws_classes/com/sun/xml/internal/bind/v2/runtime/unmarshaller/StAXEventConnector.java	2009-11-17 16:38:00.000000000 +0000
@@ -276,7 +276,7 @@
                 qName = localName;
             else
                 qName = prefix + ':' + localName;
-            String type = staxAttr.getDTDType();
+            String type = staxAttr.getDTDType().toString();
             String value = staxAttr.getValue();
 
             attrs.addAttribute(uri, localName, qName, type, value);
--- openjdk-boot.orig/jaxws/src/share/jaxws_classes/com/sun/xml/internal/fastinfoset/stax/events/AttributeBase.java.old	2009-11-17 19:46:11.000000000 +0000
+++ openjdk-boot/jaxws/src/share/jaxws_classes/com/sun/xml/internal/fastinfoset/stax/events/AttributeBase.java	2009-11-17 19:46:35.000000000 +0000
@@ -108,11 +108,10 @@
    * the String "CDATA"
    * @return the type as a String, default is "CDATA"
    */
-    public String getDTDType() {
-        return _attributeType;
+    public QName getDTDType() {
+        return new QName(_attributeType);
     }
 
-
   /**
    * A flag indicating whether this attribute was actually
    * specified in the start-tag of its element, or was defaulted from the schema.
