--- openjdk-boot.orig/jdk/make/tools/src/build/tools/charsetmapping/EUC_TW.java	2010-05-05 18:45:07.000000000 +0100
+++ openjdk-boot/jdk/make/tools/src/build/tools/charsetmapping/EUC_TW.java	2010-05-05 21:26:31.000000000 +0100
@@ -83,8 +83,8 @@
     static void genClass(String args[]) throws Exception
     {
         InputStream is = new FileInputStream(new File(args[0], "euc_tw.map"));
-        PrintStream ps = new PrintStream(new File(args[1], "EUC_TWMapping.java"),
-                                         "ISO-8859-1");
+        FileOutputStream fos = new FileOutputStream(new File(args[1], "EUC_TWMapping.java"));
+        PrintStream ps = new PrintStream(fos, false, "ISO-8859-1");
         String copyright = getCopyright(new File(args[3]));
 
 
diff -r c1cee45daf4d make/tools/src/build/tools/charsetmapping/HKSCS.java
--- openjdk-boot.orig/jdk/make/tools/src/build/tools/charsetmapping/HKSCS.java	Wed Apr 28 22:15:05 2010 +0100
+++ openjdk-boot/jdk/make/tools/src/build/tools/charsetmapping/HKSCS.java	Thu May 06 14:17:11 2010 +0100
@@ -43,29 +43,28 @@
         Pattern.compile("(?:0x)?+(\\p{XDigit}++)\\s++(?:0x|U\\+)?+(\\p{XDigit}++)?\\s*+(?:0x|U\\+)?(\\p{XDigit}++)?\\s*+.*");
 
     static void genClass(String args[]) throws Exception {
-
+        FileOutputStream fos = new FileOutputStream(new File(args[1], "HKSCSMapping.java"));
         // hkscs2008
         genClass0(new FileInputStream(new File(args[0], "HKSCS2008.map")),
                   new FileInputStream(new File(args[0], "HKSCS2008.c2b")),
-                  new PrintStream(new File(args[1], "HKSCSMapping.java"),
-                                  "ISO-8859-1"),
+                  new PrintStream(fos, false, "ISO-8859-1"),
                   "HKSCSMapping",
                   getCopyright(new File(args[3])));
 
 
         // xp2001
+        fos = new FileOutputStream(new File(args[1], "HKSCS_XPMapping.java"));
         genClass0(new FileInputStream(new File(args[0], "HKSCS_XP.map")),
                   null,
-                  new PrintStream(new File(args[1], "HKSCS_XPMapping.java"),
-                                  "ISO-8859-1"),
+                  new PrintStream(fos, false, "ISO-8859-1"),
                   "HKSCS_XPMapping",
                   getCopyright(new File(args[3])));
 
         // hkscs2001
+        fos = new FileOutputStream(new File(args[1], "HKSCS2001Mapping.java"));
         genClass0(new FileInputStream(new File(args[0], "HKSCS2001.map")),
                   new FileInputStream(new File(args[0], "HKSCS2001.c2b")),
-                  new PrintStream(new File(args[1], "HKSCS2001Mapping.java"),
-                                  "ISO-8859-1"),
+                  new PrintStream(fos, false, "ISO-8859-1"),
                   "HKSCS2001Mapping",
                   getCopyright(new File(args[3])));
     }
