diff -Nru openjdk-boot.orig/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c openjdk-boot/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c
--- openjdk-boot.orig/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c	2012-08-29 16:19:35.860554102 +0100
+++ openjdk-boot/jdk/src/solaris/native/sun/nio/ch/FileDispatcherImpl.c	2012-08-29 16:23:37.688539537 +0100
@@ -209,9 +209,9 @@
     lockResult = fcntl(fd, cmd, &fl);
     if (lockResult < 0) {
         if ((cmd == F_SETLK64) && (errno == EAGAIN || errno == EACCES))
-            return sun_nio_ch_FileDispatcherImpl_NO_LOCK;
+            return sun_nio_ch_FileDispatcher_NO_LOCK;
         if (errno == EINTR)
-            return sun_nio_ch_FileDispatcherImpl_INTERRUPTED;
+            return sun_nio_ch_FileDispatcher_INTERRUPTED;
         JNU_ThrowIOExceptionWithLastError(env, "Lock failed");
     }
     return 0;
