diff -Nru openjdk.orig/jdk/make/common/shared/Defs-versions.gmk openjdk/jdk/make/common/shared/Defs-versions.gmk
--- openjdk-boot.orig/jdk/make/common/shared/Defs-versions.gmk	2011-03-16 22:23:11.000000000 +0000
+++ openjdk-boot/jdk/make/common/shared/Defs-versions.gmk	2011-03-16 22:29:36.710722785 +0000
@@ -206,7 +206,7 @@
 
 # Generic
 REQUIRED_ANT_VER          = 1.6.3
-REQUIRED_BOOT_VER         = 1.6
+REQUIRED_BOOT_VER         = 1.5
 REQUIRED_FREETYPE_VERSION = 2.2.1
 REQUIRED_MAKE_VER         = 3.81
 REQUIRED_UNZIP_VER        = 5.12
