diff -Nru openjdk-boot.orig/jdk/src/share/classes/sun/util/calendar/ZoneInfoFile.java openjdk-boot/jdk/src/share/classes/sun/util/calendar/ZoneInfoFile.java
--- openjdk-boot.orig/jdk/src/share/classes/sun/util/calendar/ZoneInfoFile.java	2011-06-11 00:38:11.000000000 +0100
+++ openjdk-boot/jdk/src/share/classes/sun/util/calendar/ZoneInfoFile.java	2011-06-14 01:44:11.082438342 +0100
@@ -486,7 +486,7 @@
                     String otherDir = getZoneInfoDir(homeDir);
                     if (otherDir != null)
                         zi = otherDir;
-                    zi = FileSystems.getDefault().getPath(zi).toRealPath().toString();
+                    zi = new File(zi).getCanonicalPath();
                 } catch(Exception e) {
                 }
                 return zi;
