diff -Nru openjdk.orig/hotspot/src/share/vm/interpreter/bytecodeInterpreter.hpp openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreter.hpp
--- openjdk.orig/hotspot/src/share/vm/interpreter/bytecodeInterpreter.hpp	2008-10-23 08:41:04.000000000 +0100
+++ openjdk/hotspot/src/share/vm/interpreter/bytecodeInterpreter.hpp	2008-10-30 15:48:48.000000000 +0000
@@ -60,7 +60,6 @@
 };
 
 class BytecodeInterpreter : StackObj {
-friend class SharedRuntime;
 friend class AbstractInterpreterGenerator;
 friend class CppInterpreterGenerator;
 friend class InterpreterGenerator;
