diff -Nru openjdk.orig/jdk/src/share/bin/java.c openjdk/jdk/src/share/bin/java.c
--- openjdk.orig/jdk/src/share/bin/java.c	2009-05-12 23:35:16.000000000 +0100
+++ openjdk/jdk/src/share/bin/java.c	2009-05-12 23:43:56.000000000 +0100
@@ -885,8 +885,14 @@
      * "Valid" returns (other than unrecoverable errors) follow.  Set
      * main_class as a side-effect of this routine.
      */
-    if (info.main_class != NULL)
+    if (info.main_class != NULL) {
+      if (strlen(info.main_class) <= MAXNAMELEN) {
         *main_class = JLI_StringDup(info.main_class);
+      } else {
+        JLI_ReportErrorMessage("Error: main-class: attribute exceeds system limits\n", JNI_TRUE);
+	exit(1);
+      }
+    }
 
     /*
      * If no version selection information is found either on the command
