Index: jamvm/jamvm/src/classlib/openjdk/management.c
===================================================================
--- jamvm.orig/jamvm/src/classlib/openjdk/management.c	2011-08-09 23:39:21.067466386 +0200
+++ jamvm/jamvm/src/classlib/openjdk/management.c	2011-08-09 23:43:21.987466384 +0200
@@ -146,6 +146,9 @@
         case JMM_JVM_INIT_DONE_TIME_MS:
             result = 0;
             break;
+        case JMM_OS_PROCESS_ID:
+            result = getpid();
+            break;
 
         case JMM_CLASS_LOADED_COUNT:
         case JMM_CLASS_UNLOADED_COUNT:
@@ -153,7 +156,6 @@
         case JMM_THREAD_LIVE_COUNT:
         case JMM_THREAD_PEAK_COUNT:
         case JMM_THREAD_DAEMON_COUNT:
-        case JMM_OS_PROCESS_ID:
 
         default:
             UNIMPLEMENTED("jmm_GetLongAttribute: Unknown attribute %d", att);
