Index: jamvm/jamvm/configure.ac
===================================================================
--- jamvm.orig/jamvm/configure.ac	2011-03-09 13:13:12.781424034 +0100
+++ jamvm/jamvm/configure.ac	2011-03-09 13:13:47.752903998 +0100
@@ -40,6 +40,7 @@
 x86_64-*-linux*) host_os=linux ;;
 hppa*-*-linux*) host_cpu=parisc host_os=linux ;;
 mipsel-*-linux*) host_cpu=mips host_os=linux ;;
+mips64el-*-linux*) host_cpu=mips host_os=linux ;;
 x86_64-*-openbsd*) host_os=bsd libdl_needed=no ;;
 x86_64-*-freebsd*) host_os=bsd libdl_needed=no ;;
 arm*-*-linux*) host_cpu=arm host_os=linux ;;
